// @wix/auto_sdk_stores_brands-v-3/build/cjs/src/stores-catalog-v3-brand-brands-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.RequestedFields = exports.onBrandUpdated = exports.onBrandDeleted = exports.onBrandCreated = exports.bulkDeleteBrands = exports.bulkGetOrCreateBrands = exports.getOrCreateBrand = exports.bulkUpdateBrands = exports.bulkCreateBrands = exports.queryBrands = exports.deleteBrand = exports.updateBrand = exports.getBrand = exports.createBrand = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_brand_brands_v_3_universal_js_1 = require("./stores-catalog-v3-brand-brands-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createBrand(httpClient) {
    return (brand) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.createBrand)(brand, 
    // @ts-ignore
    { httpClient });
}
exports.createBrand = createBrand;
function getBrand(httpClient) {
    return (brandId, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.getBrand)(brandId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getBrand = getBrand;
function updateBrand(httpClient) {
    return (_id, brand, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.updateBrand)(_id, brand, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateBrand = updateBrand;
function deleteBrand(httpClient) {
    return (brandId) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.deleteBrand)(brandId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteBrand = deleteBrand;
function queryBrands(httpClient) {
    return (options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.queryBrands)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryBrands = queryBrands;
function bulkCreateBrands(httpClient) {
    return (brands, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.bulkCreateBrands)(brands, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateBrands = bulkCreateBrands;
function bulkUpdateBrands(httpClient) {
    return (brands, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.bulkUpdateBrands)(brands, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateBrands = bulkUpdateBrands;
function getOrCreateBrand(httpClient) {
    return (brandName, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.getOrCreateBrand)(brandName, options, 
    // @ts-ignore
    { httpClient });
}
exports.getOrCreateBrand = getOrCreateBrand;
function bulkGetOrCreateBrands(httpClient) {
    return (brandNames, options) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.bulkGetOrCreateBrands)(brandNames, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkGetOrCreateBrands = bulkGetOrCreateBrands;
function bulkDeleteBrands(httpClient) {
    return (brandIds) => (0, stores_catalog_v3_brand_brands_v_3_universal_js_1.bulkDeleteBrands)(brandIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteBrands = bulkDeleteBrands;
exports.onBrandCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.brand_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBrandDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.brand_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBrandUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.brand_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var stores_catalog_v3_brand_brands_v_3_universal_js_2 = require("./stores-catalog-v3-brand-brands-v-3.universal.js");
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_brand_brands_v_3_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_brand_brands_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_brand_brands_v_3_universal_js_2.WebhookIdentityType; } });
