// @wix/auto_sdk_stores_inventory-items-v-3/build/internal/cjs/src/stores-catalog-v3-inventory-item-inventory-items-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkSetOrAdjustInventoryItemsByFilter = exports.bulkIncrementInventoryItemsByVariantAndLocation = exports.bulkDecrementInventoryItemsByVariantAndLocation = exports.bulkIncrementInventoryItems = exports.bulkDecrementInventoryItems = exports.searchInventoryItems = exports.queryInventoryItems = exports.bulkDeleteInventoryItems = exports.deleteInventoryItem = exports.bulkUpdateInventoryItemsByFilter = exports.bulkUpdateInventoryItems = exports.updateInventoryItem = exports.getInventoryItem = exports.bulkCreateInventoryItems = exports.createInventoryItem = void 0;
const ambassadorWixStoresCatalogV3InventoryItem = __importStar(require("./stores-catalog-v3-inventory-item-inventory-items-v-3.http.js"));
function createInventoryItem() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.createInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createInventoryItem = createInventoryItem;
function bulkCreateInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkCreateInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateInventoryItems = bulkCreateInventoryItems;
function getInventoryItem() {
    const payload = { inventoryItemId: ':inventoryItemId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.getInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/inventory-items/{inventoryItemId}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getInventoryItem = getInventoryItem;
function updateInventoryItem() {
    const payload = { inventoryItem: { id: ':inventoryItemId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.updateInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/inventory-items/{inventoryItem.id}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateInventoryItem = updateInventoryItem;
function bulkUpdateInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkUpdateInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateInventoryItems = bulkUpdateInventoryItems;
function bulkUpdateInventoryItemsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkUpdateInventoryItemsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateInventoryItemsByFilter = bulkUpdateInventoryItemsByFilter;
function deleteInventoryItem() {
    const payload = { inventoryItemId: ':inventoryItemId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.deleteInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/inventory-items/{inventoryItemId}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteInventoryItem = deleteInventoryItem;
function bulkDeleteInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDeleteInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteInventoryItems = bulkDeleteInventoryItems;
function queryInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.queryInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryInventoryItems = queryInventoryItems;
function searchInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.searchInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchInventoryItems = searchInventoryItems;
function bulkDecrementInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDecrementInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/decrement',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDecrementInventoryItems = bulkDecrementInventoryItems;
function bulkIncrementInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkIncrementInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/increment',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkIncrementInventoryItems = bulkIncrementInventoryItems;
function bulkDecrementInventoryItemsByVariantAndLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDecrementInventoryItemsByVariantAndLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/decrement-by-variant-and-location',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDecrementInventoryItemsByVariantAndLocation = bulkDecrementInventoryItemsByVariantAndLocation;
function bulkIncrementInventoryItemsByVariantAndLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkIncrementInventoryItemsByVariantAndLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/increment-by-variant-and-location',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkIncrementInventoryItemsByVariantAndLocation = bulkIncrementInventoryItemsByVariantAndLocation;
function bulkSetOrAdjustInventoryItemsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkSetOrAdjustInventoryItemsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/set-or-adjust-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkSetOrAdjustInventoryItemsByFilter = bulkSetOrAdjustInventoryItemsByFilter;
