// @wix/auto_sdk_stores_products-v-3/build/internal/cjs/src/stores-catalog-v3-product-products-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkRemoveProductsFromCategoriesByFilter = exports.bulkAddProductsToCategoriesByFilter = exports.bulkRemoveInfoSectionsFromProducts = exports.bulkRemoveInfoSectionsFromProductsByFilter = exports.bulkAddInfoSectionsToProducts = exports.bulkAddInfoSectionsToProductsByFilter = exports.bulkAdjustProductVariantsByFilter = exports.bulkUpdateProductVariantsByFilter = exports.countProducts = exports.queryProducts = exports.searchProducts = exports.getProductBySlug = exports.getProduct = exports.bulkDeleteProductsByFilter = exports.bulkDeleteProducts = exports.deleteProduct = exports.updateExtendedFields = exports.bulkUpdateProductsByFilter = exports.bulkUpdateProductsWithInventory = exports.bulkUpdateProducts = exports.bulkCreateProductsWithInventory = exports.bulkCreateProducts = exports.updateProductWithInventory = exports.updateProduct = exports.createProductWithInventory = exports.createProduct = void 0;
const ambassadorWixStoresCatalogV3Product = __importStar(require("./stores-catalog-v3-product-products-v-3.http.js"));
function createProduct() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.createProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createProduct = createProduct;
function createProductWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.createProductWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products-with-inventory',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createProductWithInventory = createProductWithInventory;
function updateProduct() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/products/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateProduct = updateProduct;
function updateProductWithInventory() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateProductWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/products-with-inventory/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateProductWithInventory = updateProductWithInventory;
function bulkCreateProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkCreateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateProducts = bulkCreateProducts;
function bulkCreateProductsWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkCreateProductsWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products-with-inventory/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateProductsWithInventory = bulkCreateProductsWithInventory;
function bulkUpdateProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateProducts = bulkUpdateProducts;
function bulkUpdateProductsWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductsWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products-with-inventory/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateProductsWithInventory = bulkUpdateProductsWithInventory;
function bulkUpdateProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateProductsByFilter = bulkUpdateProductsByFilter;
function updateExtendedFields() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/{productId}/update-extended-fields',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
function deleteProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.deleteProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/products/{productId}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteProduct = deleteProduct;
function bulkDeleteProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkDeleteProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteProducts = bulkDeleteProducts;
function bulkDeleteProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkDeleteProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/delete-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteProductsByFilter = bulkDeleteProductsByFilter;
function getProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.getProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/products/{productId}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getProduct = getProduct;
function getProductBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.getProductBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/products/slug/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getProductBySlug = getProductBySlug;
function searchProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.searchProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchProducts = searchProducts;
function queryProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.queryProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryProducts = queryProducts;
function countProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.countProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countProducts = countProducts;
function bulkUpdateProductVariantsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductVariantsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update-variants-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateProductVariantsByFilter = bulkUpdateProductVariantsByFilter;
function bulkAdjustProductVariantsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAdjustProductVariantsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/adjust-variants-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAdjustProductVariantsByFilter = bulkAdjustProductVariantsByFilter;
function bulkAddInfoSectionsToProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddInfoSectionsToProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-info-sections-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAddInfoSectionsToProductsByFilter = bulkAddInfoSectionsToProductsByFilter;
function bulkAddInfoSectionsToProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddInfoSectionsToProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAddInfoSectionsToProducts = bulkAddInfoSectionsToProducts;
function bulkRemoveInfoSectionsFromProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveInfoSectionsFromProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-info-sections-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkRemoveInfoSectionsFromProductsByFilter = bulkRemoveInfoSectionsFromProductsByFilter;
function bulkRemoveInfoSectionsFromProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveInfoSectionsFromProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkRemoveInfoSectionsFromProducts = bulkRemoveInfoSectionsFromProducts;
function bulkAddProductsToCategoriesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddProductsToCategoriesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-to-categories-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAddProductsToCategoriesByFilter = bulkAddProductsToCategoriesByFilter;
function bulkRemoveProductsFromCategoriesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveProductsFromCategoriesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-from-categories-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkRemoveProductsFromCategoriesByFilter = bulkRemoveProductsFromCategoriesByFilter;
