// @wix/auto_sdk_stores_products-v-3/build/internal/cjs/src/stores-catalog-v3-product-products-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Direction = exports.DecorationType = exports.DateHistogramAggregationInterval = exports.Crop = exports.CommonSortOrder = exports.CommonSearchDetailsMode = exports.CommonScalarType = exports.CommonAggregationType = exports.ChoiceType = exports.BulkAdjustProductVariantsByFilterRequestRoundingStrategy = exports.BackgroundType = exports.AvailabilityStatus = exports.AppType = exports.Alignment = exports.AggregationValueAggregationSortType = exports.AggregationValueAggregationSortDirection = exports.AggregationValueAggregationMissingValues = exports.AggregationType = exports.AggregationNestedAggregationNestedAggregationType = exports.AggregationDateHistogramAggregationInterval = exports.onProductUpdated = exports.onProductDeleted = exports.onProductCreated = exports.bulkRemoveProductsFromCategoriesByFilter = exports.bulkAddProductsToCategoriesByFilter = exports.bulkRemoveInfoSectionsFromProducts = exports.bulkRemoveInfoSectionsFromProductsByFilter = exports.bulkAddInfoSectionsToProducts = exports.bulkAddInfoSectionsToProductsByFilter = exports.bulkAdjustProductVariantsByFilter = exports.bulkUpdateProductVariantsByFilter = exports.countProducts = exports.queryProducts = exports.searchProducts = exports.getProductBySlug = exports.getProduct = exports.bulkDeleteProductsByFilter = exports.bulkDeleteProducts = exports.deleteProduct = exports.updateExtendedFields = exports.bulkUpdateProductsByFilter = exports.bulkUpdateProductsWithInventory = exports.bulkUpdateProducts = exports.bulkCreateProductsWithInventory = exports.bulkCreateProducts = exports.updateProductWithInventory = exports.updateProduct = exports.createProductWithInventory = exports.createProduct = exports.__metadata = void 0;
exports.ViewRole = exports.ViewMode = exports.VerticalAlignment = exports.VariantsOpsRequestedFields = exports.ValueAggregationSortType = exports.ValueAggregationSortDirection = exports.ValueAggregationMissingValues = exports.Type = exports.ThumbnailsAlignment = exports.TextAlignment = exports.Target = exports.SubscriptionFrequency = exports.State = exports.Source = exports.SortType = exports.SortOrder = exports.SortDirection = exports.SingleEntityOpsRequestedFields = exports.SearchDetailsMode = exports.ScalarType = exports.RoundingStrategy = exports.RequestedFields = exports.ProductType = exports.ProductPreorderAvailability = exports.ProductOptionRenderType = exports.PreorderStatus = exports.PollLayoutType = exports.PollLayoutDirection = exports.PluginContainerDataAlignment = exports.Orientation = exports.NullValue = exports.NodeType = exports.NestedAggregationType = exports.NestedAggregationNestedAggregationType = exports.ModifierRenderType = exports.Mode = exports.MissingValues = exports.MediaType = exports.MeasurementUnit = exports.MapType = exports.LineStyle = exports.LayoutType = exports.InventoryAvailabilityStatus = exports.Interval = exports.InitialExpandedItems = exports.GIFType = exports.FontType = exports.FileType = exports.Enum = exports.DiscountType = void 0;
exports.WixCommonSortOrder = exports.WixCommonScalarType = exports.WixCommonAggregationType = exports.WidthType = exports.Width = exports.WeightUnit = exports.WebhookIdentityType = exports.VoteRole = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const video_v2_1 = require("@wix/sdk-runtime/transformations/video-v2");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_product_products_v_3_universal_js_1 = require("./stores-catalog-v3-product-products-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createProduct(httpClient) {
    return (product, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.createProduct)(product, options, 
    // @ts-ignore
    { httpClient });
}
exports.createProduct = createProduct;
function createProductWithInventory(httpClient) {
    return (product, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.createProductWithInventory)(product, options, 
    // @ts-ignore
    { httpClient });
}
exports.createProductWithInventory = createProductWithInventory;
function updateProduct(httpClient) {
    return (_id, product, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.updateProduct)(_id, product, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateProduct = updateProduct;
function updateProductWithInventory(httpClient) {
    return (_id, product, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.updateProductWithInventory)(_id, product, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateProductWithInventory = updateProductWithInventory;
function bulkCreateProducts(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkCreateProducts)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateProducts = bulkCreateProducts;
function bulkCreateProductsWithInventory(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkCreateProductsWithInventory)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateProductsWithInventory = bulkCreateProductsWithInventory;
function bulkUpdateProducts(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkUpdateProducts)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateProducts = bulkUpdateProducts;
function bulkUpdateProductsWithInventory(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkUpdateProductsWithInventory)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateProductsWithInventory = bulkUpdateProductsWithInventory;
function bulkUpdateProductsByFilter(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkUpdateProductsByFilter)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateProductsByFilter = bulkUpdateProductsByFilter;
function updateExtendedFields(httpClient) {
    return (productId, namespace, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.updateExtendedFields)(productId, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
function deleteProduct(httpClient) {
    return (productId) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.deleteProduct)(productId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteProduct = deleteProduct;
function bulkDeleteProducts(httpClient) {
    return (productIds) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkDeleteProducts)(productIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteProducts = bulkDeleteProducts;
function bulkDeleteProductsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkDeleteProductsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteProductsByFilter = bulkDeleteProductsByFilter;
function getProduct(httpClient) {
    return (productId, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.getProduct)(productId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getProduct = getProduct;
function getProductBySlug(httpClient) {
    return (slug, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.getProductBySlug)(slug, options, 
    // @ts-ignore
    { httpClient });
}
exports.getProductBySlug = getProductBySlug;
function searchProducts(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.searchProducts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.searchProducts = searchProducts;
function queryProducts(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.queryProducts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryProducts = queryProducts;
function countProducts(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.countProducts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countProducts = countProducts;
function bulkUpdateProductVariantsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkUpdateProductVariantsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateProductVariantsByFilter = bulkUpdateProductVariantsByFilter;
function bulkAdjustProductVariantsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkAdjustProductVariantsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAdjustProductVariantsByFilter = bulkAdjustProductVariantsByFilter;
function bulkAddInfoSectionsToProductsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkAddInfoSectionsToProductsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAddInfoSectionsToProductsByFilter = bulkAddInfoSectionsToProductsByFilter;
function bulkAddInfoSectionsToProducts(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkAddInfoSectionsToProducts)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAddInfoSectionsToProducts = bulkAddInfoSectionsToProducts;
function bulkRemoveInfoSectionsFromProductsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkRemoveInfoSectionsFromProductsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkRemoveInfoSectionsFromProductsByFilter = bulkRemoveInfoSectionsFromProductsByFilter;
function bulkRemoveInfoSectionsFromProducts(httpClient) {
    return (products, options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkRemoveInfoSectionsFromProducts)(products, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkRemoveInfoSectionsFromProducts = bulkRemoveInfoSectionsFromProducts;
function bulkAddProductsToCategoriesByFilter(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkAddProductsToCategoriesByFilter)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAddProductsToCategoriesByFilter = bulkAddProductsToCategoriesByFilter;
function bulkRemoveProductsFromCategoriesByFilter(httpClient) {
    return (options) => (0, stores_catalog_v3_product_products_v_3_universal_js_1.bulkRemoveProductsFromCategoriesByFilter)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkRemoveProductsFromCategoriesByFilter = bulkRemoveProductsFromCategoriesByFilter;
exports.onProductCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.product_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            {
                path: 'entity.infoSections.description.metadata.createdTimestamp',
            },
            {
                path: 'entity.infoSections.description.metadata.updatedTimestamp',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.url' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.physicalProperties.pricePerUnit.quantity' },
            { path: 'entity.physicalProperties.shippingWeightRange.minValue' },
            { path: 'entity.physicalProperties.shippingWeightRange.maxValue' },
            {
                path: 'entity.variantsInfo.variants.physicalProperties.pricePerUnit.settings.quantity',
            },
            {
                path: 'entity.variantsInfo.variants.revenueDetails.profitMargin',
            },
            {
                path: 'entity.subscriptionDetails.subscriptions.discount.percentOff',
            },
            {
                path: 'entity.minVariantPriceInfo.pricePerUnitData.settings.quantity',
            },
            { path: 'entity.minVariantPriceInfo.revenueDetails.profitMargin' },
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.options.item.ratio',
            },
            {
                path: 'entity.infoSections.description.nodes.imageData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lat',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lng',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.design.poll.background.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.appEmbedData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.video.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.audio.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.coverImage.duration',
            },
            { path: 'entity.variantsInfo.variants.physicalProperties.weight' },
            { path: 'entity.minVariantPriceInfo.weight' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.main.image' },
            { path: 'entity.media.itemsInfo.items.image' },
            { path: 'entity.options.choicesSettings.choices.image' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.modifiers.choicesSettings.choices.image' },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.variantsInfo.variants.media.image' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [
            { path: 'entity.media.main.video' },
            { path: 'entity.media.itemsInfo.items.video' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.video',
            },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.video',
            },
            { path: 'entity.variantsInfo.variants.media.video' },
        ],
    },
])))();
exports.onProductDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.product_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            {
                path: 'entity.infoSections.description.metadata.createdTimestamp',
            },
            {
                path: 'entity.infoSections.description.metadata.updatedTimestamp',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.url' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.physicalProperties.pricePerUnit.quantity' },
            { path: 'entity.physicalProperties.shippingWeightRange.minValue' },
            { path: 'entity.physicalProperties.shippingWeightRange.maxValue' },
            {
                path: 'entity.variantsInfo.variants.physicalProperties.pricePerUnit.settings.quantity',
            },
            {
                path: 'entity.variantsInfo.variants.revenueDetails.profitMargin',
            },
            {
                path: 'entity.subscriptionDetails.subscriptions.discount.percentOff',
            },
            {
                path: 'entity.minVariantPriceInfo.pricePerUnitData.settings.quantity',
            },
            { path: 'entity.minVariantPriceInfo.revenueDetails.profitMargin' },
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.options.item.ratio',
            },
            {
                path: 'entity.infoSections.description.nodes.imageData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lat',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lng',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.design.poll.background.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.appEmbedData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.video.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.audio.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.coverImage.duration',
            },
            { path: 'entity.variantsInfo.variants.physicalProperties.weight' },
            { path: 'entity.minVariantPriceInfo.weight' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.main.image' },
            { path: 'entity.media.itemsInfo.items.image' },
            { path: 'entity.options.choicesSettings.choices.image' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.modifiers.choicesSettings.choices.image' },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.variantsInfo.variants.media.image' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [
            { path: 'entity.media.main.video' },
            { path: 'entity.media.itemsInfo.items.video' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.video',
            },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.video',
            },
            { path: 'entity.variantsInfo.variants.media.video' },
        ],
    },
])))();
exports.onProductUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.product_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            {
                path: 'entity.infoSections.description.metadata.createdTimestamp',
            },
            {
                path: 'entity.infoSections.description.metadata.updatedTimestamp',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.url' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.physicalProperties.pricePerUnit.quantity' },
            { path: 'entity.physicalProperties.shippingWeightRange.minValue' },
            { path: 'entity.physicalProperties.shippingWeightRange.maxValue' },
            {
                path: 'entity.variantsInfo.variants.physicalProperties.pricePerUnit.settings.quantity',
            },
            {
                path: 'entity.variantsInfo.variants.revenueDetails.profitMargin',
            },
            {
                path: 'entity.subscriptionDetails.subscriptions.discount.percentOff',
            },
            {
                path: 'entity.minVariantPriceInfo.pricePerUnitData.settings.quantity',
            },
            { path: 'entity.minVariantPriceInfo.revenueDetails.profitMargin' },
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.galleryData.options.item.ratio',
            },
            {
                path: 'entity.infoSections.description.nodes.imageData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lat',
            },
            {
                path: 'entity.infoSections.description.nodes.mapData.mapSettings.lng',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.pollData.design.poll.background.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.appEmbedData.image.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.video.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.videoData.thumbnail.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.audio.duration',
            },
            {
                path: 'entity.infoSections.description.nodes.audioData.coverImage.duration',
            },
            { path: 'entity.variantsInfo.variants.physicalProperties.weight' },
            { path: 'entity.minVariantPriceInfo.weight' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.main.image' },
            { path: 'entity.media.itemsInfo.items.image' },
            { path: 'entity.options.choicesSettings.choices.image' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.modifiers.choicesSettings.choices.image' },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.image',
            },
            { path: 'entity.variantsInfo.variants.media.image' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [
            { path: 'entity.media.main.video' },
            { path: 'entity.media.itemsInfo.items.video' },
            {
                path: 'entity.options.choicesSettings.choices.linkedMedia.video',
            },
            {
                path: 'entity.modifiers.choicesSettings.choices.linkedMedia.video',
            },
            { path: 'entity.variantsInfo.variants.media.video' },
        ],
    },
])))();
var stores_catalog_v3_product_products_v_3_universal_js_2 = require("./stores-catalog-v3-product-products-v-3.universal.js");
Object.defineProperty(exports, "AggregationDateHistogramAggregationInterval", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationDateHistogramAggregationInterval; } });
Object.defineProperty(exports, "AggregationNestedAggregationNestedAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationNestedAggregationNestedAggregationType; } });
Object.defineProperty(exports, "AggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationType; } });
Object.defineProperty(exports, "AggregationValueAggregationMissingValues", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationValueAggregationMissingValues; } });
Object.defineProperty(exports, "AggregationValueAggregationSortDirection", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationValueAggregationSortDirection; } });
Object.defineProperty(exports, "AggregationValueAggregationSortType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AggregationValueAggregationSortType; } });
Object.defineProperty(exports, "Alignment", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Alignment; } });
Object.defineProperty(exports, "AppType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AppType; } });
Object.defineProperty(exports, "AvailabilityStatus", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.AvailabilityStatus; } });
Object.defineProperty(exports, "BackgroundType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.BackgroundType; } });
Object.defineProperty(exports, "BulkAdjustProductVariantsByFilterRequestRoundingStrategy", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.BulkAdjustProductVariantsByFilterRequestRoundingStrategy; } });
Object.defineProperty(exports, "ChoiceType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ChoiceType; } });
Object.defineProperty(exports, "CommonAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.CommonAggregationType; } });
Object.defineProperty(exports, "CommonScalarType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.CommonScalarType; } });
Object.defineProperty(exports, "CommonSearchDetailsMode", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.CommonSearchDetailsMode; } });
Object.defineProperty(exports, "CommonSortOrder", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.CommonSortOrder; } });
Object.defineProperty(exports, "Crop", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Crop; } });
Object.defineProperty(exports, "DateHistogramAggregationInterval", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.DateHistogramAggregationInterval; } });
Object.defineProperty(exports, "DecorationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.DecorationType; } });
Object.defineProperty(exports, "Direction", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Direction; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "Enum", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Enum; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.FileType; } });
Object.defineProperty(exports, "FontType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.FontType; } });
Object.defineProperty(exports, "GIFType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.GIFType; } });
Object.defineProperty(exports, "InitialExpandedItems", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.InitialExpandedItems; } });
Object.defineProperty(exports, "Interval", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Interval; } });
Object.defineProperty(exports, "InventoryAvailabilityStatus", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.InventoryAvailabilityStatus; } });
Object.defineProperty(exports, "LayoutType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.LayoutType; } });
Object.defineProperty(exports, "LineStyle", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.LineStyle; } });
Object.defineProperty(exports, "MapType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.MapType; } });
Object.defineProperty(exports, "MeasurementUnit", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.MeasurementUnit; } });
Object.defineProperty(exports, "MediaType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.MediaType; } });
Object.defineProperty(exports, "MissingValues", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.MissingValues; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Mode; } });
Object.defineProperty(exports, "ModifierRenderType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ModifierRenderType; } });
Object.defineProperty(exports, "NestedAggregationNestedAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.NestedAggregationNestedAggregationType; } });
Object.defineProperty(exports, "NestedAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.NestedAggregationType; } });
Object.defineProperty(exports, "NodeType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.NodeType; } });
Object.defineProperty(exports, "NullValue", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.NullValue; } });
Object.defineProperty(exports, "Orientation", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Orientation; } });
Object.defineProperty(exports, "PluginContainerDataAlignment", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.PluginContainerDataAlignment; } });
Object.defineProperty(exports, "PollLayoutDirection", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.PollLayoutDirection; } });
Object.defineProperty(exports, "PollLayoutType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.PollLayoutType; } });
Object.defineProperty(exports, "PreorderStatus", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.PreorderStatus; } });
Object.defineProperty(exports, "ProductOptionRenderType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ProductOptionRenderType; } });
Object.defineProperty(exports, "ProductPreorderAvailability", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ProductPreorderAvailability; } });
Object.defineProperty(exports, "ProductType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ProductType; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "RoundingStrategy", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.RoundingStrategy; } });
Object.defineProperty(exports, "ScalarType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ScalarType; } });
Object.defineProperty(exports, "SearchDetailsMode", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SearchDetailsMode; } });
Object.defineProperty(exports, "SingleEntityOpsRequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SingleEntityOpsRequestedFields; } });
Object.defineProperty(exports, "SortDirection", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SortDirection; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SortType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SortType; } });
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Source; } });
Object.defineProperty(exports, "State", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.State; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "Target", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Target; } });
Object.defineProperty(exports, "TextAlignment", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.TextAlignment; } });
Object.defineProperty(exports, "ThumbnailsAlignment", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ThumbnailsAlignment; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Type; } });
Object.defineProperty(exports, "ValueAggregationMissingValues", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ValueAggregationMissingValues; } });
Object.defineProperty(exports, "ValueAggregationSortDirection", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ValueAggregationSortDirection; } });
Object.defineProperty(exports, "ValueAggregationSortType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ValueAggregationSortType; } });
Object.defineProperty(exports, "VariantsOpsRequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.VariantsOpsRequestedFields; } });
Object.defineProperty(exports, "VerticalAlignment", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.VerticalAlignment; } });
Object.defineProperty(exports, "ViewMode", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ViewMode; } });
Object.defineProperty(exports, "ViewRole", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.ViewRole; } });
Object.defineProperty(exports, "VoteRole", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.VoteRole; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WeightUnit; } });
Object.defineProperty(exports, "Width", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.Width; } });
Object.defineProperty(exports, "WidthType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WidthType; } });
Object.defineProperty(exports, "WixCommonAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WixCommonAggregationType; } });
Object.defineProperty(exports, "WixCommonScalarType", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WixCommonScalarType; } });
Object.defineProperty(exports, "WixCommonSortOrder", { enumerable: true, get: function () { return stores_catalog_v3_product_products_v_3_universal_js_2.WixCommonSortOrder; } });
