// @wix/auto_sdk_stores_wishlist/build/es/src/platform-v1-wishlist-wishlist.universal.js
import { transformError as sdkTransformError } from '@wix/sdk-runtime/transform-error';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
import * as ambassadorWixPlatformV1Wishlist from './platform-v1-wishlist-wishlist.http.js';
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
/**
 * Get wishlist by id
 * @param _id - Unique identifier representing requested list
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @permissionId WISHLIST.MANAGE-SITE-WISHLISTS
 * @returns Object containing requested list data
 * @fqn com.wixpress.wishlist.Wishlist.GetWishlistById
 */
export async function getWishlistById(_id, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        limit: options?.limit,
        offset: options?.offset,
        kind: options?.kind,
    });
    const reqOpts = ambassadorWixPlatformV1Wishlist.getWishlistById(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data)?.wishlist;
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                id: '$[0]',
                limit: '$[1].limit',
                offset: '$[1].offset',
                kind: '$[1].kind',
            },
            singleArgumentUnchanged: false,
        }, ['_id', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
