// @wix/auto_sdk_ecom_shipping-rates/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ChargeType: () => ChargeType,
  GenericShippingRatesWixError: () => GenericShippingRatesWixError,
  IdentityType: () => IdentityType,
  InvalidAddressWixError: () => InvalidAddressWixError,
  InvalidPostalCodeWixError: () => InvalidPostalCodeWixError,
  MissingPostalCodeWixError: () => MissingPostalCodeWixError,
  PickupMethod: () => PickupMethod,
  RuleType: () => RuleType,
  SubdivisionType: () => SubdivisionType,
  VatType: () => VatType,
  WeightUnit: () => WeightUnit,
  provideHandlers: () => provideHandlers2,
  publicProvideHandlers: () => provideHandlers
});
module.exports = __toCommonJS(index_exports);

// src/service-plugins-types.ts
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["HANDLING_FEE"] = "HANDLING_FEE";
  return ChargeType2;
})(ChargeType || {});
var RuleType = /* @__PURE__ */ ((RuleType2) => {
  RuleType2["VALIDATION"] = "VALIDATION";
  RuleType2["OTHER"] = "OTHER";
  RuleType2["MAX"] = "MAX";
  RuleType2["MIN"] = "MIN";
  RuleType2["MAX_LENGTH"] = "MAX_LENGTH";
  RuleType2["MIN_LENGTH"] = "MIN_LENGTH";
  RuleType2["MAX_SIZE"] = "MAX_SIZE";
  RuleType2["MIN_SIZE"] = "MIN_SIZE";
  RuleType2["FORMAT"] = "FORMAT";
  RuleType2["DECIMAL_LTE"] = "DECIMAL_LTE";
  RuleType2["DECIMAL_GTE"] = "DECIMAL_GTE";
  RuleType2["DECIMAL_LT"] = "DECIMAL_LT";
  RuleType2["DECIMAL_GT"] = "DECIMAL_GT";
  RuleType2["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
  RuleType2["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
  RuleType2["REQUIRED_FIELD"] = "REQUIRED_FIELD";
  RuleType2["FIELD_NOT_ALLOWED"] = "FIELD_NOT_ALLOWED";
  RuleType2["ONE_OF_ALIGNMENT"] = "ONE_OF_ALIGNMENT";
  RuleType2["EXACT_LENGTH"] = "EXACT_LENGTH";
  RuleType2["EXACT_SIZE"] = "EXACT_SIZE";
  RuleType2["REQUIRED_ONE_OF_FIELD"] = "REQUIRED_ONE_OF_FIELD";
  return RuleType2;
})(RuleType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecom-v1-shipping-rates-provider.public.ts
var import_sdk_types = require("@wix/sdk-types");
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");
var provideHandlers = (0, import_sdk_types.ServicePluginDefinition)("ECOM_SHIPPING_RATES", [
  {
    name: "getShippingRates",
    primaryHttpMappingPath: "/v1/getRates",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = (0, import_transform_paths.transformPaths)(payload, [
          {
            transformFn: import_address.transformSDKAddressToRESTAddress,
            paths: [{ path: "shippingRates.logistics.pickupDetails.address" }]
          },
          {
            transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
            paths: [
              { path: "shippingRates.logistics.deliveryTimeSlot.from" },
              { path: "shippingRates.logistics.deliveryTimeSlot.to" }
            ]
          }
        ]);
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = (0, import_transform_paths.transformPaths)(payload, [
          {
            transformFn: import_address2.transformRESTAddressToSDKAddress,
            paths: [
              { path: "request.shippingDestination" },
              { path: "request.shippingOrigin" }
            ]
          },
          {
            transformFn: import_float.transformRESTFloatToSDKFloat,
            paths: [{ path: "request.lineItems.physicalProperties.weight" }]
          }
        ]);
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecom-v1-shipping-rates-provider.context.ts
var import_service_plugin_modules = require("@wix/sdk-runtime/service-plugin-modules");
var provideHandlers2 = (0, import_service_plugin_modules.createServicePluginModule)(
  provideHandlers
);

// src/service-plugins-error-classes.ts
var MissingPostalCodeWixError = class extends Error {
  constructor(data = {}) {
    super("MissingPostalCode");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "MISSING_POSTAL_CODE";
    this.name = "MissingPostalCode";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.MissingPostalCodeError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "MissingPostalCode",
      applicationCode: "MISSING_POSTAL_CODE"
    };
    this.data = data;
  }
};
/** @hidden */
MissingPostalCodeWixError.__type = "wix_spi_error";
var InvalidPostalCodeWixError = class extends Error {
  constructor(data = {}) {
    super("InvalidPostalCode");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "INVALID_POSTAL_CODE";
    this.name = "InvalidPostalCode";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.InvalidPostalCodeError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "InvalidPostalCode",
      applicationCode: "INVALID_POSTAL_CODE"
    };
    this.data = data;
  }
};
/** @hidden */
InvalidPostalCodeWixError.__type = "wix_spi_error";
var InvalidAddressWixError = class extends Error {
  constructor(data = {}) {
    super("InvalidAddress");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "PARTIAL_ADDRESS";
    this.name = "InvalidAddress";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.InvalidAddressError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "InvalidAddress",
      applicationCode: "PARTIAL_ADDRESS"
    };
    this.data = data;
  }
};
/** @hidden */
InvalidAddressWixError.__type = "wix_spi_error";
var GenericShippingRatesWixError = class extends Error {
  constructor(data = {}) {
    super("GenericShippingRatesError");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "GENERIC_ERROR";
    this.name = "GenericShippingRatesError";
    this.errorSchemaName = "com.wix.ecom.shipping_rates.v1.GenericShippingRatesError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "GenericShippingRatesError",
      applicationCode: "GENERIC_ERROR"
    };
    this.data = data;
  }
};
/** @hidden */
GenericShippingRatesWixError.__type = "wix_spi_error";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChargeType,
  GenericShippingRatesWixError,
  IdentityType,
  InvalidAddressWixError,
  InvalidPostalCodeWixError,
  MissingPostalCodeWixError,
  PickupMethod,
  RuleType,
  SubdivisionType,
  VatType,
  WeightUnit,
  provideHandlers,
  publicProvideHandlers
});
