// @wix/auto_sdk_stores_wishlist/build/internal/es/src/platform-v1-wishlist-wishlist.http.js
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveComWixpressWishlistWishlistUrl(opts) {
    const domainToMappings = {
        'bo._base_domain_': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'ecom._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '/api',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/stores/v1/wishlists',
                destPath: '/v1/wishlists',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_wishlist';
/** Get wishlist by id */
export function getWishlistById(payload) {
    function __getWishlistById({ host }) {
        const metadata = {
            entityFqdn: 'wix.platform.v1.wishlist',
            method: 'POST',
            methodFqn: 'com.wixpress.wishlist.Wishlist.GetWishlistById',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressWishlistWishlistUrl({
                protoPath: '/v1/wishlists/get',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'wishlist.items.dateAdded' }],
                },
            ]),
        };
        return metadata;
    }
    return __getWishlistById;
}
