// @wix/auto_sdk_ecom_discounts/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  CouponDoesNotExistWixError: () => CouponDoesNotExistWixError,
  CouponHasExpiredWixError: () => CouponHasExpiredWixError,
  CouponIsDisabledWixError: () => CouponIsDisabledWixError,
  CouponIsNotActiveYetWixError: () => CouponIsNotActiveYetWixError,
  CouponLimitPerCustomerExceededWixError: () => CouponLimitPerCustomerExceededWixError,
  CouponUsageExceededWixError: () => CouponUsageExceededWixError,
  DiscountSource: () => DiscountSource,
  IdentityType: () => IdentityType,
  TargetType: () => TargetType,
  Type: () => Type
});
module.exports = __toCommonJS(index_typings_exports);

// src/service-plugins-error-classes.ts
var CouponLimitPerCustomerExceededWixError = class extends Error {
  constructor() {
    super("CouponLimitPerCustomerExceeded");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED";
    this.name = "CouponLimitPerCustomerExceeded";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponLimitPerCustomerExceeded",
      applicationCode: "ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED"
    };
  }
};
/** @hidden */
CouponLimitPerCustomerExceededWixError.__type = "wix_spi_error";
var CouponIsDisabledWixError = class extends Error {
  constructor() {
    super("CouponIsDisabled");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_IS_DISABLED";
    this.name = "CouponIsDisabled";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponIsDisabled",
      applicationCode: "ERROR_COUPON_IS_DISABLED"
    };
  }
};
/** @hidden */
CouponIsDisabledWixError.__type = "wix_spi_error";
var CouponUsageExceededWixError = class extends Error {
  constructor() {
    super("CouponUsageExceeded");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_USAGE_EXCEEDED";
    this.name = "CouponUsageExceeded";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponUsageExceeded",
      applicationCode: "ERROR_COUPON_USAGE_EXCEEDED"
    };
  }
};
/** @hidden */
CouponUsageExceededWixError.__type = "wix_spi_error";
var CouponHasExpiredWixError = class extends Error {
  constructor() {
    super("CouponHasExpired");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_HAS_EXPIRED";
    this.name = "CouponHasExpired";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponHasExpired",
      applicationCode: "ERROR_COUPON_HAS_EXPIRED"
    };
  }
};
/** @hidden */
CouponHasExpiredWixError.__type = "wix_spi_error";
var CouponIsNotActiveYetWixError = class extends Error {
  constructor() {
    super("CouponIsNotActiveYet");
    this.httpCode = 428;
    this.statusCode = "FAILED_PRECONDITION";
    this.applicationCode = "ERROR_COUPON_IS_NOT_ACTIVE_YET";
    this.name = "CouponIsNotActiveYet";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponIsNotActiveYet",
      applicationCode: "ERROR_COUPON_IS_NOT_ACTIVE_YET"
    };
  }
};
/** @hidden */
CouponIsNotActiveYetWixError.__type = "wix_spi_error";
var CouponDoesNotExistWixError = class extends Error {
  constructor() {
    super("CouponDoesNotExist");
    this.httpCode = 404;
    this.statusCode = "NOT_FOUND";
    this.applicationCode = "ERROR_COUPON_DOES_NOT_EXIST";
    this.name = "CouponDoesNotExist";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "CouponDoesNotExist",
      applicationCode: "ERROR_COUPON_DOES_NOT_EXIST"
    };
  }
};
/** @hidden */
CouponDoesNotExistWixError.__type = "wix_spi_error";

// src/service-plugins-types.ts
var TargetType = /* @__PURE__ */ ((TargetType2) => {
  TargetType2["UNKNOWN_TARGET"] = "UNKNOWN_TARGET";
  TargetType2["GLOBAL"] = "GLOBAL";
  TargetType2["SHIPPING"] = "SHIPPING";
  TargetType2["LINE_ITEMS"] = "LINE_ITEMS";
  return TargetType2;
})(TargetType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN_RATE"] = "UNKNOWN_RATE";
  Type2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  Type2["PERCENTAGE"] = "PERCENTAGE";
  return Type2;
})(Type || {});
var DiscountSource = /* @__PURE__ */ ((DiscountSource2) => {
  DiscountSource2["UNKNOWN_SOURCE"] = "UNKNOWN_SOURCE";
  DiscountSource2["COUPON"] = "COUPON";
  DiscountSource2["AUTOMATIC"] = "AUTOMATIC";
  DiscountSource2["CUSTOM"] = "CUSTOM";
  return DiscountSource2;
})(DiscountSource || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CouponDoesNotExistWixError,
  CouponHasExpiredWixError,
  CouponIsDisabledWixError,
  CouponIsNotActiveYetWixError,
  CouponLimitPerCustomerExceededWixError,
  CouponUsageExceededWixError,
  DiscountSource,
  IdentityType,
  TargetType,
  Type
});
