// @wix/auto_sdk_stores_abandoned-carts/build/internal/cjs/src/stores-v1-abandoned-cart-abandoned-carts.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ActivityType = exports.Identity = exports.Status = void 0;
var Status;
(function (Status) {
    Status["ABANDONED"] = "ABANDONED";
    Status["RECOVERED"] = "RECOVERED";
})(Status || (exports.Status = Status = {}));
var Identity;
(function (Identity) {
    /** Customer is the site owner */
    Identity["ADMIN"] = "ADMIN";
    /** Customer is logged in */
    Identity["MEMBER"] = "MEMBER";
    /** Customer is not logged in */
    Identity["VISITOR"] = "VISITOR";
    /** Contact was created for the customer */
    Identity["CONTACT"] = "CONTACT";
})(Identity || (exports.Identity = Identity = {}));
var ActivityType;
(function (ActivityType) {
    ActivityType["UNRECOGNIZED_TYPE"] = "UNRECOGNIZED_TYPE";
    ActivityType["SCHEDULED"] = "SCHEDULED";
    ActivityType["EMAIL_SENT"] = "EMAIL_SENT";
    ActivityType["EMAIL_NOT_SENT"] = "EMAIL_NOT_SENT";
    ActivityType["NOTIFICATION_SENT"] = "NOTIFICATION_SENT";
    ActivityType["TASK_CREATED"] = "TASK_CREATED";
    ActivityType["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
})(ActivityType || (exports.ActivityType = ActivityType = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
