// @wix/auto_sdk_stores_customizations-v-3/build/cjs/src/stores-catalog-v3-customization-customizations-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.RequestedFields = exports.CustomizationType = exports.CustomizationRenderType = exports.ChoiceType = exports.onCustomizationUpdated = exports.onCustomizationDeleted = exports.onCustomizationCreated = exports.bulkUpdateCustomizations = exports.bulkAddCustomizationChoices = exports.removeCustomizationChoices = exports.setCustomizationChoices = exports.addCustomizationChoices = exports.bulkCreateCustomizations = exports.queryCustomizations = exports.deleteCustomization = exports.updateCustomization = exports.getCustomization = exports.createCustomization = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_customization_customizations_v_3_universal_js_1 = require("./stores-catalog-v3-customization-customizations-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createCustomization(httpClient) {
    return (customization) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.createCustomization)(customization, 
    // @ts-ignore
    { httpClient });
}
exports.createCustomization = createCustomization;
function getCustomization(httpClient) {
    return (customizationId, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.getCustomization)(customizationId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getCustomization = getCustomization;
function updateCustomization(httpClient) {
    return (_id, customization, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.updateCustomization)(_id, customization, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCustomization = updateCustomization;
function deleteCustomization(httpClient) {
    return (customizationId) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.deleteCustomization)(customizationId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCustomization = deleteCustomization;
function queryCustomizations(httpClient) {
    return (options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.queryCustomizations)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryCustomizations = queryCustomizations;
function bulkCreateCustomizations(httpClient) {
    return (customizations, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.bulkCreateCustomizations)(customizations, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateCustomizations = bulkCreateCustomizations;
function addCustomizationChoices(httpClient) {
    return (customizationId, choices, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.addCustomizationChoices)(customizationId, choices, options, 
    // @ts-ignore
    { httpClient });
}
exports.addCustomizationChoices = addCustomizationChoices;
function setCustomizationChoices(httpClient) {
    return (customizationId, choices, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.setCustomizationChoices)(customizationId, choices, options, 
    // @ts-ignore
    { httpClient });
}
exports.setCustomizationChoices = setCustomizationChoices;
function removeCustomizationChoices(httpClient) {
    return (customizationId, choiceIds, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.removeCustomizationChoices)(customizationId, choiceIds, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeCustomizationChoices = removeCustomizationChoices;
function bulkAddCustomizationChoices(httpClient) {
    return (customizationsChoices, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.bulkAddCustomizationChoices)(customizationsChoices, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAddCustomizationChoices = bulkAddCustomizationChoices;
function bulkUpdateCustomizations(httpClient) {
    return (customizations, options) => (0, stores_catalog_v3_customization_customizations_v_3_universal_js_1.bulkUpdateCustomizations)(customizations, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateCustomizations = bulkUpdateCustomizations;
exports.onCustomizationCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.customization_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.choicesSettings.choices.image' }],
    },
])))();
exports.onCustomizationDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.customization_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.choicesSettings.choices.image' }],
    },
])))();
exports.onCustomizationUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.customization_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.choicesSettings.choices.image' }],
    },
])))();
var stores_catalog_v3_customization_customizations_v_3_universal_js_2 = require("./stores-catalog-v3-customization-customizations-v-3.universal.js");
Object.defineProperty(exports, "ChoiceType", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.ChoiceType; } });
Object.defineProperty(exports, "CustomizationRenderType", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.CustomizationRenderType; } });
Object.defineProperty(exports, "CustomizationType", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.CustomizationType; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_customization_customizations_v_3_universal_js_2.WebhookIdentityType; } });
