// @wix/auto_sdk_stores_customizations-v-3/build/cjs/src/stores-catalog-v3-customization-customizations-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkUpdateCustomizations = exports.bulkAddCustomizationChoices = exports.removeCustomizationChoices = exports.setCustomizationChoices = exports.addCustomizationChoices = exports.bulkCreateCustomizations = exports.queryCustomizations = exports.deleteCustomization = exports.updateCustomization = exports.getCustomization = exports.createCustomization = void 0;
const ambassadorWixStoresCatalogV3Customization = __importStar(require("./stores-catalog-v3-customization-customizations-v-3.http.js"));
function createCustomization() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.createCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCustomization = createCustomization;
function getCustomization() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.getCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/customizations/{customizationId}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCustomization = getCustomization;
function updateCustomization() {
    const payload = { customization: { id: ':customizationId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.updateCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/customizations/{customization.id}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCustomization = updateCustomization;
function deleteCustomization() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.deleteCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/customizations/{customizationId}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCustomization = deleteCustomization;
function queryCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.queryCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryCustomizations = queryCustomizations;
function bulkCreateCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkCreateCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateCustomizations = bulkCreateCustomizations;
function addCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.addCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/add-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addCustomizationChoices = addCustomizationChoices;
function setCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.setCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/set-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setCustomizationChoices = setCustomizationChoices;
function removeCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.removeCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/remove-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeCustomizationChoices = removeCustomizationChoices;
function bulkAddCustomizationChoices() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkAddCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/add-choices',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAddCustomizationChoices = bulkAddCustomizationChoices;
function bulkUpdateCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkUpdateCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateCustomizations = bulkUpdateCustomizations;
