// @wix/auto_sdk_ecom_additional-fees/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ChargeType: () => ChargeType,
  DiscountType: () => DiscountType,
  IdentityType: () => IdentityType,
  PickupMethod: () => PickupMethod,
  VatType: () => VatType,
  provideHandlers: () => provideHandlers2,
  publicProvideHandlers: () => provideHandlers
});
module.exports = __toCommonJS(index_exports);

// src/service-plugins-types.ts
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["HANDLING_FEE"] = "HANDLING_FEE";
  ChargeType2["INSURANCE"] = "INSURANCE";
  return ChargeType2;
})(ChargeType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecommerce-v1-additional-fees-provider.public.ts
var import_sdk_types = require("@wix/sdk-types");
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");
var provideHandlers = (0, import_sdk_types.ServicePluginDefinition)("ECOM_ADDITIONAL_FEES", [
  {
    name: "calculateAdditionalFees",
    primaryHttpMappingPath: "/v1/calculate-additional-fees",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = (0, import_transform_paths.transformPaths)(payload, [
          {
            transformFn: import_address.transformRESTAddressToSDKAddress,
            paths: [
              { path: "request.shippingAddress" },
              {
                path: "request.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address"
              }
            ]
          },
          {
            transformFn: import_float.transformRESTFloatToSDKFloat,
            paths: [{ path: "request.lineItems.physicalProperties.weight" }]
          }
        ]);
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecommerce-v1-additional-fees-provider.context.ts
var import_service_plugin_modules = require("@wix/sdk-runtime/service-plugin-modules");
var provideHandlers2 = (0, import_service_plugin_modules.createServicePluginModule)(
  provideHandlers
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChargeType,
  DiscountType,
  IdentityType,
  PickupMethod,
  VatType,
  provideHandlers,
  publicProvideHandlers
});
