// @wix/auto_sdk_ecom_local-delivery-options/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ConditionTypeOriginal: () => ConditionType,
  DayOfWeekOriginal: () => DayOfWeek,
  LengthUnitOriginal: () => LengthUnit,
  LogicalOperatorOriginal: () => LogicalOperator,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  addDeliveryRegion: () => addDeliveryRegion2,
  bulkCreateLocalDeliveryOption: () => bulkCreateLocalDeliveryOption2,
  bulkDeleteLocalDeliveryOption: () => bulkDeleteLocalDeliveryOption2,
  bulkUpdateLocalDeliveryOption: () => bulkUpdateLocalDeliveryOption2,
  createLocalDeliveryOption: () => createLocalDeliveryOption2,
  deleteLocalDeliveryOption: () => deleteLocalDeliveryOption2,
  getLocalDeliveryOption: () => getLocalDeliveryOption2,
  listLocalDeliveryOptions: () => listLocalDeliveryOptions2,
  queryLocalDeliveryOptions: () => queryLocalDeliveryOptions2,
  removeDeliveryRegion: () => removeDeliveryRegion2,
  updateLocalDeliveryOption: () => updateLocalDeliveryOption2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-local-delivery-option-local-delivery-options.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/wix-ecommerce-shipments-local-delivery-web",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/localDeliveryOptions",
        destPath: "/v1/localDeliveryOptions"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/local-delivery-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/local-delivery-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_local-delivery-options";
function createLocalDeliveryOption(payload) {
  function __createLocalDeliveryOption({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
          { path: "localDeliveryOption.radiusConfig.address.geocode.latitude" },
          {
            path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOption.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOption.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOption.radiusConfig.radius" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOption.createdDate" },
          { path: "localDeliveryOption.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.CreateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createLocalDeliveryOption;
}
function getLocalDeliveryOption(payload) {
  function __getLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "GET",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.GetLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ]),
      fallback: [
        {
          method: "GET",
          url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
            protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
            data: payload,
            host
          }),
          params: (0, import_rest_modules.toURLSearchParams)(payload)
        }
      ]
    };
    return metadata;
  }
  return __getLocalDeliveryOption;
}
function listLocalDeliveryOptions(payload) {
  function __listLocalDeliveryOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "GET",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.ListLocalDeliveryOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/localDeliveryOptions",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listLocalDeliveryOptions;
}
function queryLocalDeliveryOptions(payload) {
  function __queryLocalDeliveryOptions({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.QueryLocalDeliveryOptions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryLocalDeliveryOptions;
}
function updateLocalDeliveryOption(payload) {
  function __updateLocalDeliveryOption({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
          { path: "localDeliveryOption.radiusConfig.address.geocode.latitude" },
          {
            path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOption.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOption.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOption.radiusConfig.radius" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOption.createdDate" },
          { path: "localDeliveryOption.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "PATCH",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.UpdateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOption.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateLocalDeliveryOption;
}
function deleteLocalDeliveryOption(payload) {
  function __deleteLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "DELETE",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.DeleteLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/{localDeliveryOptionId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteLocalDeliveryOption;
}
function bulkCreateLocalDeliveryOption(payload) {
  function __bulkCreateLocalDeliveryOption({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
          },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOptions.radiusConfig.radius" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOptions.createdDate" },
          { path: "localDeliveryOptions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkCreateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/create",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateLocalDeliveryOption;
}
function bulkUpdateLocalDeliveryOption(payload) {
  function __bulkUpdateLocalDeliveryOption({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
          },
          {
            path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
          },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.latitude" },
          { path: "localDeliveryOptions.customAreaConfig.geocodes.longitude" },
          { path: "localDeliveryOptions.radiusConfig.radius" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "localDeliveryOptions.createdDate" },
          { path: "localDeliveryOptions.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "PATCH",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkUpdateLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOptions.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOptions.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOptions.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOptions.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOptions.createdDate" },
            { path: "localDeliveryOptions.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateLocalDeliveryOption;
}
function bulkDeleteLocalDeliveryOption(payload) {
  function __bulkDeleteLocalDeliveryOption({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "DELETE",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkDeleteLocalDeliveryOption",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/bulk/local-delivery-options/delete",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __bulkDeleteLocalDeliveryOption;
}
function addDeliveryRegion(payload) {
  function __addDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.AddDeliveryRegion",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/add-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addDeliveryRegion;
}
function removeDeliveryRegion(payload) {
  function __removeDeliveryRegion({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.local_delivery_option",
      method: "POST",
      methodFqn: "com.wix.ecom.shipments.localdelivery.LocalDelivery.RemoveDeliveryRegion",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
        protoPath: "/v1/local-delivery-options/remove-delivery-region",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "localDeliveryOption.minimumFreeDeliveryOrderAmount" },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.latitude"
            },
            {
              path: "localDeliveryOption.radiusConfig.address.geocode.longitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.latitude"
            },
            {
              path: "localDeliveryOption.customAreaConfig.geocodes.longitude"
            },
            { path: "localDeliveryOption.radiusConfig.radius" }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "localDeliveryOption.createdDate" },
            { path: "localDeliveryOption.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __removeDeliveryRegion;
}

// src/ecom-v1-local-delivery-option-local-delivery-options.types.ts
var LengthUnit = /* @__PURE__ */ ((LengthUnit2) => {
  LengthUnit2["UNSPECIFIED_LENGTH_UNIT"] = "UNSPECIFIED_LENGTH_UNIT";
  LengthUnit2["KM"] = "KM";
  LengthUnit2["MI"] = "MI";
  return LengthUnit2;
})(LengthUnit || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var ConditionType = /* @__PURE__ */ ((ConditionType2) => {
  ConditionType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ConditionType2["BY_TOTAL_WEIGHT"] = "BY_TOTAL_WEIGHT";
  ConditionType2["BY_TOTAL_PRICE"] = "BY_TOTAL_PRICE";
  ConditionType2["BY_TOTAL_QUANTITY"] = "BY_TOTAL_QUANTITY";
  return ConditionType2;
})(ConditionType || {});
var LogicalOperator = /* @__PURE__ */ ((LogicalOperator2) => {
  LogicalOperator2["UNKNOWN_LOGICAL_OPERATOR_TYPE"] = "UNKNOWN_LOGICAL_OPERATOR_TYPE";
  LogicalOperator2["EQ"] = "EQ";
  LogicalOperator2["GT"] = "GT";
  LogicalOperator2["GTE"] = "GTE";
  LogicalOperator2["LT"] = "LT";
  LogicalOperator2["LTE"] = "LTE";
  return LogicalOperator2;
})(LogicalOperator || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/ecom-v1-local-delivery-option-local-delivery-options.meta.ts
function createLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = createLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getLocalDeliveryOption2() {
  const payload = { localDeliveryOptionId: ":localDeliveryOptionId" };
  const getRequestOptions = getLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/local-delivery-options/{localDeliveryOptionId}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listLocalDeliveryOptions2() {
  const payload = {};
  const getRequestOptions = listLocalDeliveryOptions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/localDeliveryOptions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryLocalDeliveryOptions2() {
  const payload = {};
  const getRequestOptions = queryLocalDeliveryOptions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateLocalDeliveryOption2() {
  const payload = {
    localDeliveryOption: { id: ":localDeliveryOptionId" }
  };
  const getRequestOptions = updateLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/local-delivery-options/{localDeliveryOption.id}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteLocalDeliveryOption2() {
  const payload = { localDeliveryOptionId: ":localDeliveryOptionId" };
  const getRequestOptions = deleteLocalDeliveryOption(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/local-delivery-options/{localDeliveryOptionId}",
    pathParams: { localDeliveryOptionId: "localDeliveryOptionId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkCreateLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/local-delivery-options/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkUpdateLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/bulk/local-delivery-options/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteLocalDeliveryOption2() {
  const payload = {};
  const getRequestOptions = bulkDeleteLocalDeliveryOption(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/bulk/local-delivery-options/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addDeliveryRegion2() {
  const payload = {};
  const getRequestOptions = addDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/add-delivery-region",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeDeliveryRegion2() {
  const payload = {};
  const getRequestOptions = removeDeliveryRegion(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/local-delivery-options/remove-delivery-region",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ConditionTypeOriginal,
  DayOfWeekOriginal,
  LengthUnitOriginal,
  LogicalOperatorOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  addDeliveryRegion,
  bulkCreateLocalDeliveryOption,
  bulkDeleteLocalDeliveryOption,
  bulkUpdateLocalDeliveryOption,
  createLocalDeliveryOption,
  deleteLocalDeliveryOption,
  getLocalDeliveryOption,
  listLocalDeliveryOptions,
  queryLocalDeliveryOptions,
  removeDeliveryRegion,
  updateLocalDeliveryOption
});
