// @wix/auto_sdk_stores_wishlist/build/internal/cjs/src/platform-v1-wishlist-wishlist.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getWishlistById = void 0;
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressWishlistWishlistUrl(opts) {
    const domainToMappings = {
        'bo._base_domain_': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/wishlist-server',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'ecom._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '/api',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/stores/v1/wishlists',
                destPath: '/v1/wishlists',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/wishlist-server',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_wishlist';
/** Get wishlist by id */
function getWishlistById(payload) {
    function __getWishlistById({ host }) {
        const metadata = {
            entityFqdn: 'wix.platform.v1.wishlist',
            method: 'POST',
            methodFqn: 'com.wixpress.wishlist.Wishlist.GetWishlistById',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressWishlistWishlistUrl({
                protoPath: '/v1/wishlists/get',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'wishlist.items.dateAdded' }],
                },
            ]),
        };
        return metadata;
    }
    return __getWishlistById;
}
exports.getWishlistById = getWishlistById;
