// @wix/auto_sdk_stores_brands-v-3/build/internal/cjs/src/stores-catalog-v3-brand-brands-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkDeleteBrands = exports.bulkGetOrCreateBrands = exports.getOrCreateBrand = exports.bulkUpdateBrands = exports.bulkCreateBrands = exports.queryBrands = exports.deleteBrand = exports.updateBrand = exports.getBrand = exports.createBrand = void 0;
const ambassadorWixStoresCatalogV3Brand = __importStar(require("./stores-catalog-v3-brand-brands-v-3.http.js"));
function createBrand() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.createBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createBrand = createBrand;
function getBrand() {
    const payload = { brandId: ':brandId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.getBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/brands/{brandId}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getBrand = getBrand;
function updateBrand() {
    const payload = { brand: { id: ':brandId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.updateBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/brands/{brand.id}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateBrand = updateBrand;
function deleteBrand() {
    const payload = { brandId: ':brandId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.deleteBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/brands/{brandId}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteBrand = deleteBrand;
function queryBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.queryBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryBrands = queryBrands;
function bulkCreateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkCreateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateBrands = bulkCreateBrands;
function bulkUpdateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkUpdateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateBrands = bulkUpdateBrands;
function getOrCreateBrand() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.getOrCreateBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrCreateBrand = getOrCreateBrand;
function bulkGetOrCreateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkGetOrCreateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGetOrCreateBrands = bulkGetOrCreateBrands;
function bulkDeleteBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkDeleteBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteBrands = bulkDeleteBrands;
