// @wix/stores/build/internal/cjs/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  abandonedCarts: () => abandonedCarts,
  brandsV3: () => brandsV3,
  catalogImportsV3: () => catalogImportsV3,
  catalogVersioning: () => catalogVersioning,
  collections: () => collections,
  customizationsV3: () => customizationsV3,
  infoSectionsV3: () => infoSectionsV3,
  inventory: () => inventory,
  inventoryItemsV3: () => inventoryItemsV3,
  products: () => products,
  productsV3: () => productsV3,
  readOnlyVariantsV3: () => readOnlyVariantsV3,
  ribbonsV3: () => ribbonsV3,
  storesLocationsV3: () => storesLocationsV3,
  subscriptionOptions: () => subscriptionOptions,
  wishlist: () => wishlist
});
module.exports = __toCommonJS(index_exports);
var wishlist = __toESM(require("@wix/auto_sdk_stores_wishlist"));
var collections = __toESM(require("@wix/auto_sdk_stores_collections"));
var products = __toESM(require("@wix/auto_sdk_stores_products"));
var brandsV3 = __toESM(require("@wix/auto_sdk_stores_brands-v-3"));
var catalogImportsV3 = __toESM(require("@wix/auto_sdk_stores_catalog-imports-v-3"));
var customizationsV3 = __toESM(require("@wix/auto_sdk_stores_customizations-v-3"));
var infoSectionsV3 = __toESM(require("@wix/auto_sdk_stores_info-sections-v-3"));
var inventoryItemsV3 = __toESM(require("@wix/auto_sdk_stores_inventory-items-v-3"));
var productsV3 = __toESM(require("@wix/auto_sdk_stores_products-v-3"));
var catalogVersioning = __toESM(require("@wix/auto_sdk_stores_catalog-versioning"));
var readOnlyVariantsV3 = __toESM(require("@wix/auto_sdk_stores_read-only-variants-v-3"));
var ribbonsV3 = __toESM(require("@wix/auto_sdk_stores_ribbons-v-3"));
var storesLocationsV3 = __toESM(require("@wix/auto_sdk_stores_stores-locations-v-3"));
var abandonedCarts = __toESM(require("@wix/auto_sdk_stores_abandoned-carts"));
var subscriptionOptions = __toESM(require("@wix/auto_sdk_stores_subscription-options"));
var inventory = __toESM(require("@wix/auto_sdk_stores_inventory"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  abandonedCarts,
  brandsV3,
  catalogImportsV3,
  catalogVersioning,
  collections,
  customizationsV3,
  infoSectionsV3,
  inventory,
  inventoryItemsV3,
  products,
  productsV3,
  readOnlyVariantsV3,
  ribbonsV3,
  storesLocationsV3,
  subscriptionOptions,
  wishlist
});
