// @wix/stores/build/internal/cjs/meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.inventory = exports.subscriptionOptions = exports.abandonedCarts = exports.storesLocationsV3 = exports.ribbonsV3 = exports.catalogVersioning = exports.productsV3 = exports.inventoryItemsV3 = exports.infoSectionsV3 = exports.customizationsV3 = exports.brandsV3 = exports.products = exports.collections = exports.wishlist = void 0;
exports.wishlist = __importStar(require("@wix/auto_sdk_stores_wishlist/meta"));
exports.collections = __importStar(require("@wix/auto_sdk_stores_collections/meta"));
exports.products = __importStar(require("@wix/auto_sdk_stores_products/meta"));
exports.brandsV3 = __importStar(require("@wix/auto_sdk_stores_brands-v-3/meta"));
exports.customizationsV3 = __importStar(require("@wix/auto_sdk_stores_customizations-v-3/meta"));
exports.infoSectionsV3 = __importStar(require("@wix/auto_sdk_stores_info-sections-v-3/meta"));
exports.inventoryItemsV3 = __importStar(require("@wix/auto_sdk_stores_inventory-items-v-3/meta"));
exports.productsV3 = __importStar(require("@wix/auto_sdk_stores_products-v-3/meta"));
exports.catalogVersioning = __importStar(require("@wix/auto_sdk_stores_catalog-versioning/meta"));
exports.ribbonsV3 = __importStar(require("@wix/auto_sdk_stores_ribbons-v-3/meta"));
exports.storesLocationsV3 = __importStar(require("@wix/auto_sdk_stores_stores-locations-v-3/meta"));
exports.abandonedCarts = __importStar(require("@wix/auto_sdk_stores_abandoned-carts/meta"));
exports.subscriptionOptions = __importStar(require("@wix/auto_sdk_stores_subscription-options/meta"));
exports.inventory = __importStar(require("@wix/auto_sdk_stores_inventory/meta"));
