// @wix/auto_sdk_stores_products/build/es/src/stores-catalog-v1-product-products.meta.js
import * as ambassadorWixStoresCatalogV1Product from './stores-catalog-v1-product-products.http.js';
export function createProduct() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.createProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateProduct() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteProduct() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/products/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateProductVariants() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{id}/variants',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function resetAllProductVariantData() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.resetAllVariantData(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/variants/resetToDefault',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addProductsToCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductsToCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections/{id}/productIds',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeProductsFromCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductsFromCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections/{id}/productIds/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addProductMedia() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductMedia(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/media',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeProductMedia() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductMedia(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/media/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addProductMediaToChoices() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductMediaToChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{id}/choices/media',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeProductMediaFromChoices() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductMediaFromChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/choices/media/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteProductOptions() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteProductOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/products/{id}/options',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeBrand() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/remove-brand',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.createCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCollection() {
    const payload = { collection: { id: ':collectionId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/collections/{collection.id}',
        pathParams: { collectionId: 'collectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/collections/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeRibbon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/remove-ribbon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateProductsProperty() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.bulkUpdateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/products/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAdjustProductProperty() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.bulkAdjustProductProperties(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/products/adjust-properties',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryProductsPlatformized(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/products/query-platformized',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getProduct() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/products/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCollectionBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getCollectionBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/collections/slug/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getProductOptionsAvailability() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.productOptionsAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/productOptionsAvailability',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryProductVariants() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryProductVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/variants/query',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryStoreVariants() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryStoreVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/variants/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getStoreVariant() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getStoreVariant(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/variants/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
