// @wix/auto_sdk_stores_products/build/es/src/stores-catalog-v1-product-products.types.js
export var ProductType;
(function (ProductType) {
    ProductType["unspecified_product_type"] = "unspecified_product_type";
    ProductType["physical"] = "physical";
    ProductType["digital"] = "digital";
})(ProductType || (ProductType = {}));
export var InventoryStatus;
(function (InventoryStatus) {
    /** In stock */
    InventoryStatus["IN_STOCK"] = "IN_STOCK";
    /** Not in stock */
    InventoryStatus["OUT_OF_STOCK"] = "OUT_OF_STOCK";
    /** Some of the variants are not in stock */
    InventoryStatus["PARTIALLY_OUT_OF_STOCK"] = "PARTIALLY_OUT_OF_STOCK";
})(InventoryStatus || (InventoryStatus = {}));
export var MeasurementUnit;
(function (MeasurementUnit) {
    MeasurementUnit["UNSPECIFIED"] = "UNSPECIFIED";
    MeasurementUnit["ML"] = "ML";
    MeasurementUnit["CL"] = "CL";
    MeasurementUnit["L"] = "L";
    MeasurementUnit["CBM"] = "CBM";
    MeasurementUnit["MG"] = "MG";
    MeasurementUnit["G"] = "G";
    MeasurementUnit["KG"] = "KG";
    MeasurementUnit["MM"] = "MM";
    MeasurementUnit["CM"] = "CM";
    MeasurementUnit["M"] = "M";
    MeasurementUnit["SQM"] = "SQM";
    MeasurementUnit["OZ"] = "OZ";
    MeasurementUnit["LB"] = "LB";
    MeasurementUnit["FLOZ"] = "FLOZ";
    MeasurementUnit["PT"] = "PT";
    MeasurementUnit["QT"] = "QT";
    MeasurementUnit["GAL"] = "GAL";
    MeasurementUnit["IN"] = "IN";
    MeasurementUnit["FT"] = "FT";
    MeasurementUnit["YD"] = "YD";
    MeasurementUnit["SQFT"] = "SQFT";
})(MeasurementUnit || (MeasurementUnit = {}));
export var MediaItemType;
(function (MediaItemType) {
    MediaItemType["unspecified_media_item_type"] = "unspecified_media_item_type";
    /** Image media type. */
    MediaItemType["image"] = "image";
    /** Video media type. */
    MediaItemType["video"] = "video";
    /** Audio media type. */
    MediaItemType["audio"] = "audio";
    /** Document media type. */
    MediaItemType["document"] = "document";
    /** Zip media type. */
    MediaItemType["zip"] = "zip";
})(MediaItemType || (MediaItemType = {}));
export var OptionType;
(function (OptionType) {
    /** Unspecified option type. */
    OptionType["unspecified_option_type"] = "unspecified_option_type";
    /** Drop down. */
    OptionType["drop_down"] = "drop_down";
    /** Color. */
    OptionType["color"] = "color";
})(OptionType || (OptionType = {}));
export var DiscountType;
(function (DiscountType) {
    DiscountType["UNDEFINED"] = "UNDEFINED";
    /** No discount */
    DiscountType["NONE"] = "NONE";
    /** Discount by a fixed amount */
    DiscountType["AMOUNT"] = "AMOUNT";
    /** Discount by a percentage */
    DiscountType["PERCENT"] = "PERCENT";
})(DiscountType || (DiscountType = {}));
export var FileType;
(function (FileType) {
    FileType["UNSPECIFIED"] = "UNSPECIFIED";
    FileType["SECURE_PICTURE"] = "SECURE_PICTURE";
    FileType["SECURE_VIDEO"] = "SECURE_VIDEO";
    FileType["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
    FileType["SECURE_MUSIC"] = "SECURE_MUSIC";
    FileType["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
})(FileType || (FileType = {}));
export var Version;
(function (Version) {
    /** Version 1 of the catalog. */
    Version["V1_CATALOG"] = "V1_CATALOG";
    /** Version 3 of the catalog. */
    Version["V3_CATALOG"] = "V3_CATALOG";
})(Version || (Version = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
