// @wix/auto_sdk_stores_subscription-options/build/cjs/src/stores-v1-subscription-option-subscription-options.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionFrequency = exports.DiscountType = exports.getOneTimePurchasesStatus = exports.getProductIdsForSubscriptionOption = exports.getSubscriptionOptionsForProduct = exports.getSubscriptionOption = exports.allowOneTimePurchases = exports.assignSubscriptionOptionsToProduct = exports.bulkDeleteSubscriptionOptions = exports.bulkUpdateSubscriptionOptions = exports.bulkCreateSubscriptionOptions = exports.deleteSubscriptionOption = exports.updateSubscriptionOption = exports.createSubscriptionOption = exports.__metadata = void 0;
const stores_v1_subscription_option_subscription_options_universal_js_1 = require("./stores-v1-subscription-option-subscription-options.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createSubscriptionOption(httpClient) {
    return (subscriptionOption) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.createSubscriptionOption)(subscriptionOption, 
    // @ts-ignore
    { httpClient });
}
exports.createSubscriptionOption = createSubscriptionOption;
function updateSubscriptionOption(httpClient) {
    return (_id, subscriptionOption) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.updateSubscriptionOption)(_id, subscriptionOption, 
    // @ts-ignore
    { httpClient });
}
exports.updateSubscriptionOption = updateSubscriptionOption;
function deleteSubscriptionOption(httpClient) {
    return (_id) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.deleteSubscriptionOption)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteSubscriptionOption = deleteSubscriptionOption;
function bulkCreateSubscriptionOptions(httpClient) {
    return (subscriptionOptions) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.bulkCreateSubscriptionOptions)(subscriptionOptions, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateSubscriptionOptions = bulkCreateSubscriptionOptions;
function bulkUpdateSubscriptionOptions(httpClient) {
    return (subscriptionOptions) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.bulkUpdateSubscriptionOptions)(subscriptionOptions, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateSubscriptionOptions = bulkUpdateSubscriptionOptions;
function bulkDeleteSubscriptionOptions(httpClient) {
    return (ids) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.bulkDeleteSubscriptionOptions)(ids, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteSubscriptionOptions = bulkDeleteSubscriptionOptions;
function assignSubscriptionOptionsToProduct(httpClient) {
    return (productId, options) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.assignSubscriptionOptionsToProduct)(productId, options, 
    // @ts-ignore
    { httpClient });
}
exports.assignSubscriptionOptionsToProduct = assignSubscriptionOptionsToProduct;
function allowOneTimePurchases(httpClient) {
    return (productId, allowed) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.allowOneTimePurchases)(productId, allowed, 
    // @ts-ignore
    { httpClient });
}
exports.allowOneTimePurchases = allowOneTimePurchases;
function getSubscriptionOption(httpClient) {
    return (_id) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.getSubscriptionOption)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getSubscriptionOption = getSubscriptionOption;
function getSubscriptionOptionsForProduct(httpClient) {
    return (productId, options) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.getSubscriptionOptionsForProduct)(productId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getSubscriptionOptionsForProduct = getSubscriptionOptionsForProduct;
function getProductIdsForSubscriptionOption(httpClient) {
    return (_id, options) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.getProductIdsForSubscriptionOption)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.getProductIdsForSubscriptionOption = getProductIdsForSubscriptionOption;
function getOneTimePurchasesStatus(httpClient) {
    return (productId) => (0, stores_v1_subscription_option_subscription_options_universal_js_1.getOneTimePurchasesStatus)(productId, 
    // @ts-ignore
    { httpClient });
}
exports.getOneTimePurchasesStatus = getOneTimePurchasesStatus;
var stores_v1_subscription_option_subscription_options_universal_js_2 = require("./stores-v1-subscription-option-subscription-options.universal.js");
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return stores_v1_subscription_option_subscription_options_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return stores_v1_subscription_option_subscription_options_universal_js_2.SubscriptionFrequency; } });
