// @wix/auto_sdk_ecom_gift-vouchers/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Status: () => Status,
  getGiftCard: () => getGiftCard2,
  redeemGiftCard: () => redeemGiftCard2,
  voidTransaction: () => voidTransaction2
});
module.exports = __toCommonJS(index_typings_exports);

// src/ecom-v1-gift-card-gift-vouchers.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-v1-gift-card-gift-vouchers.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/gift-cards-spi-host",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_gift-vouchers";
function getGiftCard(payload) {
  function __getGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.GetGiftCard",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/{code}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getGiftCard;
}
function redeemGiftCard(payload) {
  function __redeemGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.RedeemGiftCard",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/redeem",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __redeemGiftCard;
}
function voidTransaction(payload) {
  function __voidTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.VoidTransaction",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/void",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __voidTransaction;
}

// src/ecom-v1-gift-card-gift-vouchers.universal.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["VALID"] = "VALID";
  Status2["EXPIRED"] = "EXPIRED";
  Status2["DISABLED"] = "DISABLED";
  return Status2;
})(Status || {});
async function getGiftCard2(code, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    code,
    context: options?.context,
    appId: options?.appId,
    pin: options?.pin,
    extendedFields: options?.extendedFields
  });
  const reqOpts = getGiftCard(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          code: "$[0]",
          context: "$[1].context",
          appId: "$[1].appId",
          pin: "$[1].pin",
          extendedFields: "$[1].extendedFields"
        },
        singleArgumentUnchanged: false
      },
      ["code", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function redeemGiftCard2(code, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    code,
    amount: options?.amount,
    orderId: options?.orderId,
    appId: options?.appId,
    context: options?.context,
    pin: options?.pin,
    extendedFields: options?.extendedFields
  });
  const reqOpts = redeemGiftCard(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          code: "$[0]",
          amount: "$[1].amount",
          orderId: "$[1].orderId",
          appId: "$[1].appId",
          context: "$[1].context",
          pin: "$[1].pin",
          extendedFields: "$[1].extendedFields"
        },
        singleArgumentUnchanged: false
      },
      ["code", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function voidTransaction2(transactionId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    transactionId,
    appId: options?.appId,
    context: options?.context,
    extendedFields: options?.extendedFields
  });
  const reqOpts = voidTransaction(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          transactionId: "$[0]",
          appId: "$[1].appId",
          context: "$[1].context",
          extendedFields: "$[1].extendedFields"
        },
        singleArgumentUnchanged: false
      },
      ["transactionId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Status,
  getGiftCard,
  redeemGiftCard,
  voidTransaction
});
