// @wix/auto_sdk_ecom_gift-vouchers/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  StatusOriginal: () => Status,
  getGiftCard: () => getGiftCard2,
  redeemGiftCard: () => redeemGiftCard2,
  voidTransaction: () => voidTransaction2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-gift-card-gift-vouchers.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/gift-cards-spi-host",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/gift-cards",
        destPath: "/v1/gift-cards"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_gift-vouchers";
function getGiftCard(payload) {
  function __getGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.GetGiftCard",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/{code}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getGiftCard;
}
function redeemGiftCard(payload) {
  function __redeemGiftCard({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.RedeemGiftCard",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/redeem",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __redeemGiftCard;
}
function voidTransaction(payload) {
  function __voidTransaction({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.gift_card",
      method: "POST",
      methodFqn: "wix.ecom.gift_cards_spi_host.v1.GiftCardsSpiHostService.VoidTransaction",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomGiftCardsSpiHostV1GiftCardsSpiHostServiceUrl({
        protoPath: "/v1/gift-cards/void",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __voidTransaction;
}

// src/ecom-v1-gift-card-gift-vouchers.types.ts
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  Status2["VALID"] = "VALID";
  Status2["EXPIRED"] = "EXPIRED";
  Status2["DISABLED"] = "DISABLED";
  return Status2;
})(Status || {});

// src/ecom-v1-gift-card-gift-vouchers.meta.ts
function getGiftCard2() {
  const payload = { code: ":code" };
  const getRequestOptions = getGiftCard(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/{code}",
    pathParams: { code: "code" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redeemGiftCard2() {
  const payload = {};
  const getRequestOptions = redeemGiftCard(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/redeem",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function voidTransaction2() {
  const payload = {};
  const getRequestOptions = voidTransaction(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/gift-cards/void",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  StatusOriginal,
  getGiftCard,
  redeemGiftCard,
  voidTransaction
});
