// @wix/auto_sdk_stores_products/build/internal/cjs/src/stores-catalog-v1-product-products.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Version = exports.SortOrder = exports.ProductType = exports.OptionType = exports.MediaItemType = exports.MeasurementUnit = exports.InventoryStatus = exports.FileType = exports.DiscountType = exports.onProductVariantsChanged = exports.onProductDeleted = exports.onProductCreated = exports.onProductChanged = exports.onProductCollectionDeleted = exports.onProductCollectionCreated = exports.onProductCollectionChanged = exports.getStoreVariant = exports.queryStoreVariants = exports.queryProductVariants = exports.getProductOptionsAvailability = exports.getCollectionBySlug = exports.getProduct = exports.queryProducts = exports.bulkAdjustProductProperty = exports.bulkUpdateProductsProperty = exports.removeRibbon = exports.deleteCollection = exports.updateCollection = exports.createCollection = exports.removeBrand = exports.deleteProductOptions = exports.removeProductMediaFromChoices = exports.addProductMediaToChoices = exports.removeProductMedia = exports.addProductMedia = exports.removeProductsFromCollection = exports.addProductsToCollection = exports.resetAllProductVariantData = exports.updateProductVariants = exports.deleteProduct = exports.updateProduct = exports.createProduct = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v1_product_products_universal_js_1 = require("./stores-catalog-v1-product-products.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createProduct(httpClient) {
    return (product) => (0, stores_catalog_v1_product_products_universal_js_1.createProduct)(product, 
    // @ts-ignore
    { httpClient });
}
exports.createProduct = createProduct;
function updateProduct(httpClient) {
    return (_id, product) => (0, stores_catalog_v1_product_products_universal_js_1.updateProduct)(_id, product, 
    // @ts-ignore
    { httpClient });
}
exports.updateProduct = updateProduct;
function deleteProduct(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.deleteProduct)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteProduct = deleteProduct;
function updateProductVariants(httpClient) {
    return (_id, variants) => (0, stores_catalog_v1_product_products_universal_js_1.updateProductVariants)(_id, variants, 
    // @ts-ignore
    { httpClient });
}
exports.updateProductVariants = updateProductVariants;
function resetAllProductVariantData(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.resetAllProductVariantData)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.resetAllProductVariantData = resetAllProductVariantData;
function addProductsToCollection(httpClient) {
    return (_id, productIds) => (0, stores_catalog_v1_product_products_universal_js_1.addProductsToCollection)(_id, productIds, 
    // @ts-ignore
    { httpClient });
}
exports.addProductsToCollection = addProductsToCollection;
function removeProductsFromCollection(httpClient) {
    return (_id, productIds) => (0, stores_catalog_v1_product_products_universal_js_1.removeProductsFromCollection)(_id, productIds, 
    // @ts-ignore
    { httpClient });
}
exports.removeProductsFromCollection = removeProductsFromCollection;
function addProductMedia(httpClient) {
    return (_id, media) => (0, stores_catalog_v1_product_products_universal_js_1.addProductMedia)(_id, media, 
    // @ts-ignore
    { httpClient });
}
exports.addProductMedia = addProductMedia;
function removeProductMedia(httpClient) {
    return (_id, mediaIds) => (0, stores_catalog_v1_product_products_universal_js_1.removeProductMedia)(_id, mediaIds, 
    // @ts-ignore
    { httpClient });
}
exports.removeProductMedia = removeProductMedia;
function addProductMediaToChoices(httpClient) {
    return (_id, media) => (0, stores_catalog_v1_product_products_universal_js_1.addProductMediaToChoices)(_id, media, 
    // @ts-ignore
    { httpClient });
}
exports.addProductMediaToChoices = addProductMediaToChoices;
function removeProductMediaFromChoices(httpClient) {
    return (_id, media) => (0, stores_catalog_v1_product_products_universal_js_1.removeProductMediaFromChoices)(_id, media, 
    // @ts-ignore
    { httpClient });
}
exports.removeProductMediaFromChoices = removeProductMediaFromChoices;
function deleteProductOptions(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.deleteProductOptions)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteProductOptions = deleteProductOptions;
function removeBrand(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.removeBrand)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeBrand = removeBrand;
function createCollection(httpClient) {
    return (collection) => (0, stores_catalog_v1_product_products_universal_js_1.createCollection)(collection, 
    // @ts-ignore
    { httpClient });
}
exports.createCollection = createCollection;
function updateCollection(httpClient) {
    return (_id, collection) => (0, stores_catalog_v1_product_products_universal_js_1.updateCollection)(_id, collection, 
    // @ts-ignore
    { httpClient });
}
exports.updateCollection = updateCollection;
function deleteCollection(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.deleteCollection)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCollection = deleteCollection;
function removeRibbon(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.removeRibbon)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeRibbon = removeRibbon;
function bulkUpdateProductsProperty(httpClient) {
    return (ids, set) => (0, stores_catalog_v1_product_products_universal_js_1.bulkUpdateProductsProperty)(ids, set, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateProductsProperty = bulkUpdateProductsProperty;
function bulkAdjustProductProperty(httpClient) {
    return (adjust, ids) => (0, stores_catalog_v1_product_products_universal_js_1.bulkAdjustProductProperty)(adjust, ids, 
    // @ts-ignore
    { httpClient });
}
exports.bulkAdjustProductProperty = bulkAdjustProductProperty;
function queryProducts(httpClient) {
    return () => (0, stores_catalog_v1_product_products_universal_js_1.queryProducts)(
    // @ts-ignore
    { httpClient });
}
exports.queryProducts = queryProducts;
function getProduct(httpClient) {
    return (_id, options) => (0, stores_catalog_v1_product_products_universal_js_1.getProduct)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.getProduct = getProduct;
function getCollectionBySlug(httpClient) {
    return (slug) => (0, stores_catalog_v1_product_products_universal_js_1.getCollectionBySlug)(slug, 
    // @ts-ignore
    { httpClient });
}
exports.getCollectionBySlug = getCollectionBySlug;
function getProductOptionsAvailability(httpClient) {
    return (_id, options) => (0, stores_catalog_v1_product_products_universal_js_1.getProductOptionsAvailability)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.getProductOptionsAvailability = getProductOptionsAvailability;
function queryProductVariants(httpClient) {
    return (_id, options) => (0, stores_catalog_v1_product_products_universal_js_1.queryProductVariants)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.queryProductVariants = queryProductVariants;
function queryStoreVariants(httpClient) {
    return (query) => (0, stores_catalog_v1_product_products_universal_js_1.queryStoreVariants)(query, 
    // @ts-ignore
    { httpClient });
}
exports.queryStoreVariants = queryStoreVariants;
function getStoreVariant(httpClient) {
    return (_id) => (0, stores_catalog_v1_product_products_universal_js_1.getStoreVariant)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getStoreVariant = getStoreVariant;
exports.onProductCollectionChanged = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.CollectionChanged', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
exports.onProductCollectionCreated = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.CollectionCreated', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
exports.onProductCollectionDeleted = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.CollectionDeleted', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
exports.onProductChanged = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.ProductChanged', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
exports.onProductCreated = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.ProductCreated', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'data.price.price' },
            { path: 'data.price.pricePerUnit' },
            { path: 'data.costAndProfitData.itemCost' },
            { path: 'data.price.discountedPrice' },
            { path: 'data.costAndProfitData.profit' },
            { path: 'data.costAndProfitData.profitMargin' },
        ],
    },
])))();
exports.onProductDeleted = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.ProductDeleted', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
exports.onProductVariantsChanged = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.catalog.api.v1.VariantsChanged', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
var stores_catalog_v1_product_products_universal_js_2 = require("./stores-catalog-v1-product-products.universal.js");
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.FileType; } });
Object.defineProperty(exports, "InventoryStatus", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.InventoryStatus; } });
Object.defineProperty(exports, "MeasurementUnit", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.MeasurementUnit; } });
Object.defineProperty(exports, "MediaItemType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.MediaItemType; } });
Object.defineProperty(exports, "OptionType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.OptionType; } });
Object.defineProperty(exports, "ProductType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.ProductType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Version", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.Version; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v1_product_products_universal_js_2.WebhookIdentityType; } });
