// @wix/auto_sdk_stores_products/build/internal/cjs/src/stores-catalog-v1-product-products.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStoreVariant = exports.queryStoreVariants = exports.queryProductVariants = exports.getProductOptionsAvailability = exports.getCollectionBySlug = exports.getProduct = exports.queryProducts = exports.bulkAdjustProductProperty = exports.bulkUpdateProductsProperty = exports.removeRibbon = exports.deleteCollection = exports.updateCollection = exports.createCollection = exports.removeBrand = exports.deleteProductOptions = exports.removeProductMediaFromChoices = exports.addProductMediaToChoices = exports.removeProductMedia = exports.addProductMedia = exports.removeProductsFromCollection = exports.addProductsToCollection = exports.resetAllProductVariantData = exports.updateProductVariants = exports.deleteProduct = exports.updateProduct = exports.createProduct = void 0;
const ambassadorWixStoresCatalogV1Product = __importStar(require("./stores-catalog-v1-product-products.http.js"));
function createProduct() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.createProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createProduct = createProduct;
function updateProduct() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateProduct = updateProduct;
function deleteProduct() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/products/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteProduct = deleteProduct;
function updateProductVariants() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{id}/variants',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateProductVariants = updateProductVariants;
function resetAllProductVariantData() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.resetAllVariantData(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/variants/resetToDefault',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.resetAllProductVariantData = resetAllProductVariantData;
function addProductsToCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductsToCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections/{id}/productIds',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addProductsToCollection = addProductsToCollection;
function removeProductsFromCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductsFromCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections/{id}/productIds/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeProductsFromCollection = removeProductsFromCollection;
function addProductMedia() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductMedia(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/media',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addProductMedia = addProductMedia;
function removeProductMedia() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductMedia(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/media/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeProductMedia = removeProductMedia;
function addProductMediaToChoices() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.addProductMediaToChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/products/{id}/choices/media',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addProductMediaToChoices = addProductMediaToChoices;
function removeProductMediaFromChoices() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeProductMediaFromChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/choices/media/delete',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeProductMediaFromChoices = removeProductMediaFromChoices;
function deleteProductOptions() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteProductOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/products/{id}/options',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteProductOptions = deleteProductOptions;
function removeBrand() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/remove-brand',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeBrand = removeBrand;
function createCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.createCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCollection = createCollection;
function updateCollection() {
    const payload = { collection: { id: ':collectionId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.updateCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/collections/{collection.id}',
        pathParams: { collectionId: 'collectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCollection = updateCollection;
function deleteCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.deleteCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/collections/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCollection = deleteCollection;
function removeRibbon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.removeRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/remove-ribbon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeRibbon = removeRibbon;
function bulkUpdateProductsProperty() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.bulkUpdateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/products/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateProductsProperty = bulkUpdateProductsProperty;
function bulkAdjustProductProperty() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.bulkAdjustProductProperties(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/products/adjust-properties',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkAdjustProductProperty = bulkAdjustProductProperty;
function queryProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryProductsPlatformized(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/products/query-platformized',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryProducts = queryProducts;
function getProduct() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/products/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getProduct = getProduct;
function getCollectionBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getCollectionBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/collections/slug/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCollectionBySlug = getCollectionBySlug;
function getProductOptionsAvailability() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.productOptionsAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/productOptionsAvailability',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getProductOptionsAvailability = getProductOptionsAvailability;
function queryProductVariants() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryProductVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/products/{id}/variants/query',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryProductVariants = queryProductVariants;
function queryStoreVariants() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.queryStoreVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/variants/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryStoreVariants = queryStoreVariants;
function getStoreVariant() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Product.getStoreVariant(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/variants/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getStoreVariant = getStoreVariant;
