// @wix/auto_sdk_stores_inventory/build/cjs/src/stores-v2-inventory-inventory.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ReasonType = exports.onInventoryVariantsChanged = exports.onInventoryItemChanged = exports.incrementInventory = exports.decrementInventory = exports.updateInventoryVariants = exports.queryInventory = exports.getInventoryVariants = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_v2_inventory_inventory_universal_js_1 = require("./stores-v2-inventory-inventory.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function getInventoryVariants(httpClient) {
    return (inventoryId, options) => (0, stores_v2_inventory_inventory_universal_js_1.getInventoryVariants)(inventoryId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getInventoryVariants = getInventoryVariants;
function queryInventory(httpClient) {
    return (options) => (0, stores_v2_inventory_inventory_universal_js_1.queryInventory)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryInventory = queryInventory;
function updateInventoryVariants(httpClient) {
    return (productId, inventoryItem) => (0, stores_v2_inventory_inventory_universal_js_1.updateInventoryVariants)(productId, inventoryItem, 
    // @ts-ignore
    { httpClient });
}
exports.updateInventoryVariants = updateInventoryVariants;
function decrementInventory(httpClient) {
    return (decrementData) => (0, stores_v2_inventory_inventory_universal_js_1.decrementInventory)(decrementData, 
    // @ts-ignore
    { httpClient });
}
exports.decrementInventory = decrementInventory;
function incrementInventory(httpClient) {
    return (incrementData) => (0, stores_v2_inventory_inventory_universal_js_1.incrementInventory)(incrementData, 
    // @ts-ignore
    { httpClient });
}
exports.incrementInventory = incrementInventory;
exports.onInventoryItemChanged = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.inventory.api.v1.InventoryItemChanged', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: field_mask_1.transformRESTFieldMaskToSDKFieldMask,
        paths: [{ path: 'data.fieldMask' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.updatedDate' }],
    },
])))();
exports.onInventoryVariantsChanged = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.inventory.api.v1.InventoryVariantsChanged', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.updatedDate' }],
    },
])))();
var stores_v2_inventory_inventory_universal_js_2 = require("./stores-v2-inventory-inventory.universal.js");
Object.defineProperty(exports, "ReasonType", { enumerable: true, get: function () { return stores_v2_inventory_inventory_universal_js_2.ReasonType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_v2_inventory_inventory_universal_js_2.WebhookIdentityType; } });
