// @wix/auto_sdk_stores_inventory/build/cjs/src/stores-v2-inventory-inventory.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.incrementInventory = exports.decrementInventory = exports.updateInventoryVariants = exports.queryInventory = exports.getInventoryVariants = void 0;
const ambassadorWixStoresV2Inventory = __importStar(require("./stores-v2-inventory-inventory.http.js"));
function getInventoryVariants() {
    const payload = { inventoryId: ':inventoryId' };
    const getRequestOptions = ambassadorWixStoresV2Inventory.getInventoryVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/{inventoryId}/getVariants',
        pathParams: { inventoryId: 'inventoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getInventoryVariants = getInventoryVariants;
function queryInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.queryInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryInventory = queryInventory;
function updateInventoryVariants() {
    const payload = {
        inventoryItem: { productId: ':inventoryItemProductId' },
    };
    const getRequestOptions = ambassadorWixStoresV2Inventory.updateInventoryVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/inventoryItems/product/{inventoryItem.productId}',
        pathParams: { inventoryItemProductId: 'inventoryItemProductId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateInventoryVariants = updateInventoryVariants;
function decrementInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.decrementInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/decrement',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.decrementInventory = decrementInventory;
function incrementInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.incrementInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/increment',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.incrementInventory = incrementInventory;
