// @wix/auto_sdk_stores_abandoned-carts/build/internal/es/src/stores-v1-abandoned-cart-abandoned-carts.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixCartAbandonmentApiV1AbandonedCartsUrl(opts) {
    const domainToMappings = {};
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_abandoned-carts';
/** Returns abandoned carts based on the cart ID */
export function getAbandonedCart(payload) {
    function __getAbandonedCart({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.v1.abandoned_cart',
            method: 'GET',
            methodFqn: 'wix.cart.abandonment.api.v1.AbandonedCarts.GetAbandonedCart',
            packageName: PACKAGE_NAME,
            url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
                protoPath: '/v1/abandonedCarts/{id}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCart.abandonTime' },
                        { path: 'abandonedCart.activities.timestamp' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getAbandonedCart;
}
/** Returns abandoned carts based on the requested query parameters */
export function queryAbandonedCarts(payload) {
    function __queryAbandonedCarts({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.v1.abandoned_cart',
            method: 'POST',
            methodFqn: 'wix.cart.abandonment.api.v1.AbandonedCarts.QueryAbandonedCarts',
            packageName: PACKAGE_NAME,
            url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
                protoPath: '/v1/abandonedCarts/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCarts.abandonTime' },
                        { path: 'abandonedCarts.activities.timestamp' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryAbandonedCarts;
}
