// @wix/auto_sdk_stores_abandoned-carts/build/internal/es/src/stores-v1-abandoned-cart-abandoned-carts.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { getAbandonedCart as universalGetAbandonedCart, queryAbandonedCarts as universalQueryAbandonedCarts, } from './stores-v1-abandoned-cart-abandoned-carts.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/stores' };
/** @internal */
export function getAbandonedCart(httpClient) {
    return (_id) => universalGetAbandonedCart(_id, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function queryAbandonedCarts(httpClient) {
    return (options) => universalQueryAbandonedCarts(options, 
    // @ts-ignore
    { httpClient });
}
export const onAbandonedCartCartAbandonedEvent = EventDefinition('com.wix.ecommerce.cart.abandonment.api.v1.CartAbandonedEvent', false, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.creationTime' }, { path: 'data.abandonTime' }],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            { path: 'data.totals.subtotal' },
            { path: 'data.totals.total' },
        ],
    },
])))();
export const onAbandonedCartCartRecoveredEvent = EventDefinition('com.wix.ecommerce.cart.abandonment.api.v1.CartRecoveredEvent', false, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.recoveredTime' },
            { path: 'data.creationTime' },
            { path: 'data.abandonedTime' },
        ],
    },
])))();
export { ActivityType, Identity, Status, WebhookIdentityType, } from './stores-v1-abandoned-cart-abandoned-carts.universal.js';
