// @wix/auto_sdk_stores_abandoned-carts/build/internal/es/src/stores-v1-abandoned-cart-abandoned-carts.context.js
import { getAbandonedCart as publicGetAbandonedCart, queryAbandonedCarts as publicQueryAbandonedCarts, } from './stores-v1-abandoned-cart-abandoned-carts.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onAbandonedCartCartAbandonedEvent as publicOnAbandonedCartCartAbandonedEvent } from './stores-v1-abandoned-cart-abandoned-carts.public.js';
import { onAbandonedCartCartRecoveredEvent as publicOnAbandonedCartCartRecoveredEvent } from './stores-v1-abandoned-cart-abandoned-carts.public.js';
/** @internal */
export const getAbandonedCart = /*#__PURE__*/ createRESTModule(publicGetAbandonedCart);
/** @internal */
export const queryAbandonedCarts = /*#__PURE__*/ createRESTModule(publicQueryAbandonedCarts);
/**
 * <blockquote class='warning'>
 *
 * __Deprecation Notice:__
 *
 * This webhook has been replaced with the
 * [Abandoned Checkout Created Webhook](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/abandoned-checkout/abandoned-checkout-created)
 * and will be removed on March 31, 2024.
 *
 * </blockquote>
 * @deprecated
 */
export const onAbandonedCartCartAbandonedEvent = createEventModule(publicOnAbandonedCartCartAbandonedEvent);
/**
 * <blockquote class='warning'>
 *
 * __Deprecation Notice:__
 *
 * This webhook has been replaced with the
 * [Abandoned Checkout Recovered Webhook](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/abandoned-checkout/abandoned-checkout-recovered)
 * and will be removed on March 31, 2024.
 *
 * </blockquote>
 * @deprecated
 */
export const onAbandonedCartCartRecoveredEvent = createEventModule(publicOnAbandonedCartCartRecoveredEvent);
export { Status, Identity, ActivityType, WebhookIdentityType, } from './stores-v1-abandoned-cart-abandoned-carts.universal.js';
