// @wix/auto_sdk_ecom_order-invoices/build/internal/cjs/src/ecom-orders-v1-invoice-order-invoices.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addInvoiceToOrder = exports.bulkGenerateInvoices = exports.generateInvoice = exports.listInvoicesForMultipleOrders = void 0;
const ambassadorWixEcomOrdersV1Invoice = __importStar(require("./ecom-orders-v1-invoice-order-invoices.http.js"));
function listInvoicesForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.listInvoicesForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listInvoicesForMultipleOrders = listInvoicesForMultipleOrders;
function generateInvoice() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.generateInvoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.generateInvoice = generateInvoice;
function bulkGenerateInvoices() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.bulkGenerateInvoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/invoices/generate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGenerateInvoices = bulkGenerateInvoices;
function addInvoiceToOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.addInvoiceToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/orders/{orderId}/add-invoice',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addInvoiceToOrder = addInvoiceToOrder;
