// @wix/auto_sdk_stores_brands-v-3/build/internal/es/src/stores-catalog-v3-brand-brands-v-3.meta.js
import * as ambassadorWixStoresCatalogV3Brand from './stores-catalog-v3-brand-brands-v-3.http.js';
export function createBrand() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.createBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getBrand() {
    const payload = { brandId: ':brandId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.getBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/brands/{brandId}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateBrand() {
    const payload = { brand: { id: ':brandId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.updateBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/brands/{brand.id}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteBrand() {
    const payload = { brandId: ':brandId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.deleteBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/brands/{brandId}',
        pathParams: { brandId: 'brandId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.queryBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkCreateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkUpdateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrCreateBrand() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.getOrCreateBrand(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/brands/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGetOrCreateBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkGetOrCreateBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteBrands() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Brand.bulkDeleteBrands(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/brands/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
