// @wix/auto_sdk_stores_brands-v-3/build/internal/es/src/stores-catalog-v3-brand-brands-v-3.context.js
import { createBrand as publicCreateBrand, getBrand as publicGetBrand, updateBrand as publicUpdateBrand, deleteBrand as publicDeleteBrand, queryBrands as publicQueryBrands, bulkCreateBrands as publicBulkCreateBrands, bulkUpdateBrands as publicBulkUpdateBrands, getOrCreateBrand as publicGetOrCreateBrand, bulkGetOrCreateBrands as publicBulkGetOrCreateBrands, bulkDeleteBrands as publicBulkDeleteBrands, } from './stores-catalog-v3-brand-brands-v-3.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onBrandCreated as publicOnBrandCreated } from './stores-catalog-v3-brand-brands-v-3.public.js';
import { onBrandDeleted as publicOnBrandDeleted } from './stores-catalog-v3-brand-brands-v-3.public.js';
import { onBrandUpdated as publicOnBrandUpdated } from './stores-catalog-v3-brand-brands-v-3.public.js';
export const createBrand = /*#__PURE__*/ createRESTModule(publicCreateBrand);
export const getBrand = /*#__PURE__*/ createRESTModule(publicGetBrand);
export const updateBrand = /*#__PURE__*/ createRESTModule(publicUpdateBrand);
export const deleteBrand = /*#__PURE__*/ createRESTModule(publicDeleteBrand);
export const queryBrands = /*#__PURE__*/ createRESTModule(publicQueryBrands);
export const bulkCreateBrands = /*#__PURE__*/ createRESTModule(publicBulkCreateBrands);
export const bulkUpdateBrands = /*#__PURE__*/ createRESTModule(publicBulkUpdateBrands);
export const getOrCreateBrand = /*#__PURE__*/ createRESTModule(publicGetOrCreateBrand);
export const bulkGetOrCreateBrands = /*#__PURE__*/ createRESTModule(publicBulkGetOrCreateBrands);
export const bulkDeleteBrands = /*#__PURE__*/ createRESTModule(publicBulkDeleteBrands);
/**
 * Triggered when a brand is created.
 */
export const onBrandCreated = createEventModule(publicOnBrandCreated);
/**
 * Triggered when a brand is deleted.
 */
export const onBrandDeleted = createEventModule(publicOnBrandDeleted);
/**
 * Triggered when a brand is updated.
 */
export const onBrandUpdated = createEventModule(publicOnBrandUpdated);
export { RequestedFields, SortOrder, WebhookIdentityType, } from './stores-catalog-v3-brand-brands-v-3.universal.js';
