// @wix/ecom/build/internal/cjs/service-plugins.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// service-plugins.ts
var service_plugins_exports = {};
__export(service_plugins_exports, {
  additionalFees: () => additionalFees,
  catalog: () => catalog,
  customTriggers: () => customTriggers,
  discounts: () => discounts,
  giftVouchersProvider: () => giftVouchersProvider,
  inventory: () => inventory,
  memberships: () => memberships,
  paymentSettings: () => paymentSettings,
  recommendationsProvider: () => recommendationsProvider,
  shippingRates: () => shippingRates,
  taxCalculationProvider: () => taxCalculationProvider,
  validations: () => validations
});
module.exports = __toCommonJS(service_plugins_exports);
var taxCalculationProvider = __toESM(require("@wix/auto_sdk_ecom_tax-calculation-provider"));
var giftVouchersProvider = __toESM(require("@wix/auto_sdk_ecom_gift-vouchers-provider"));
var memberships = __toESM(require("@wix/auto_sdk_ecom_memberships"));
var recommendationsProvider = __toESM(require("@wix/auto_sdk_ecom_recommendations-provider"));
var shippingRates = __toESM(require("@wix/auto_sdk_ecom_shipping-rates"));
var additionalFees = __toESM(require("@wix/auto_sdk_ecom_additional-fees"));
var catalog = __toESM(require("@wix/auto_sdk_ecom_catalog"));
var customTriggers = __toESM(require("@wix/auto_sdk_ecom_custom-triggers"));
var discounts = __toESM(require("@wix/auto_sdk_ecom_discounts"));
var inventory = __toESM(require("@wix/auto_sdk_ecom_inventory"));
var paymentSettings = __toESM(require("@wix/auto_sdk_ecom_payment-settings"));
var validations = __toESM(require("@wix/auto_sdk_ecom_validations"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  additionalFees,
  catalog,
  customTriggers,
  discounts,
  giftVouchersProvider,
  inventory,
  memberships,
  paymentSettings,
  recommendationsProvider,
  shippingRates,
  taxCalculationProvider,
  validations
});
