// @wix/ecom_app-extensions/build/cjs/dashboard.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// dashboard.ts
var dashboard_exports = {};
__export(dashboard_exports, {
  pages: () => dashboard_page_builders_exports,
  plugins: () => dashboard_pages_plugin_props_exports
});
module.exports = __toCommonJS(dashboard_exports);

// src/extensions/dashboard-page-builders.ts
var dashboard_page_builders_exports = {};
__export(dashboard_page_builders_exports, {
  deliveryProfile: () => deliveryProfile,
  deliveryProfiles: () => deliveryProfiles,
  editDraftOrder: () => editDraftOrder,
  editOrder: () => editOrder,
  newOrder: () => newOrder,
  orderDetails: () => orderDetails,
  orderList: () => orderList,
  orderRefund: () => orderRefund,
  taxSettings: () => taxSettings
});
function editOrder(options) {
  const { orderId } = options;
  return {
    pageId: "f2526550-194f-4c13-9298-9a6b8abda62f",
    relativeUrl: `${orderId}`
  };
}
function taxSettings() {
  return { pageId: "7f43cf15-e14a-48f7-a7fa-9d0d1b1a6f02" };
}
function orderRefund(options) {
  const { orderId } = options;
  return {
    pageId: "c4898867-69a5-4e05-92b7-c276f9424641",
    relativeUrl: `${orderId}`
  };
}
function deliveryProfiles() {
  return { pageId: "841919b4-9e96-43f4-87ea-fa382bc9d0e8" };
}
function deliveryProfile(options) {
  const { deliveryProfileId } = options;
  return {
    pageId: "841919b4-9e96-43f4-87ea-fa382bc9d0e8",
    relativeUrl: `${deliveryProfileId}`
  };
}
function orderList() {
  return { pageId: "8107f05f-d646-4c81-be90-adf28d321398" };
}
function newOrder() {
  return { pageId: "8454e8a0-5684-4331-9d4a-e4873467553a" };
}
function editDraftOrder(options) {
  const { draftOrderId } = options;
  return {
    pageId: "8454e8a0-5684-4331-9d4a-e4873467553a",
    relativeUrl: `${draftOrderId}`
  };
}
function orderDetails(options) {
  const { id } = options;
  return {
    pageId: "3b96985e-a459-4972-94c3-63f038a73af2",
    relativeUrl: `${id}`
  };
}

// src/extensions/dashboard-pages-plugin-props.ts
var dashboard_pages_plugin_props_exports = {};
__export(dashboard_pages_plugin_props_exports, {
  CollectPayment: () => e5128b65_cb23_4ea4_b63a_434a4e6b6e9e_exports,
  DraftOrderPage: () => e8a0_5684_4331_9d4a_e4873467553a_exports,
  EditOrder: () => f2526550_194f_4c13_9298_9a6b8abda62f_exports,
  OrderDetails: () => b96985e_a459_4972_94c3_63f038a73af2_exports,
  Orders: () => f05f_d646_4c81_be90_adf28d321398_exports
});

// src/extensions/dashboard-page-plugins/f2526550-194f-4c13-9298-9a6b8abda62f.ts
var f2526550_194f_4c13_9298_9a6b8abda62f_exports = {};

// src/extensions/dashboard-page-plugins/8107f05f-d646-4c81-be90-adf28d321398.ts
var f05f_d646_4c81_be90_adf28d321398_exports = {};

// src/extensions/dashboard-page-plugins/8454e8a0-5684-4331-9d4a-e4873467553a.ts
var e8a0_5684_4331_9d4a_e4873467553a_exports = {};

// src/extensions/dashboard-page-plugins/e5128b65-cb23-4ea4-b63a-434a4e6b6e9e.ts
var e5128b65_cb23_4ea4_b63a_434a4e6b6e9e_exports = {};

// src/extensions/dashboard-page-plugins/3b96985e-a459-4972-94c3-63f038a73af2.ts
var b96985e_a459_4972_94c3_63f038a73af2_exports = {};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  pages,
  plugins
});
