// @wix/auto_sdk_stores_ribbons-v-3/build/es/src/stores-catalog-v3-ribbon-ribbons-v-3.meta.js
import * as ambassadorWixStoresCatalogV3Ribbon from './stores-catalog-v3-ribbon-ribbons-v-3.http.js';
export function createRibbon() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.createRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getRibbon() {
    const payload = { ribbonId: ':ribbonId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.getRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/ribbons/{ribbonId}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateRibbon() {
    const payload = { ribbon: { id: ':ribbonId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.updateRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/ribbons/{ribbon.id}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteRibbon() {
    const payload = { ribbonId: ':ribbonId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.deleteRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/ribbons/{ribbonId}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.queryRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkCreateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkUpdateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrCreateRibbon() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.getOrCreateRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGetOrCreateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkGetOrCreateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkDeleteRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
