// @wix/auto_sdk_ecom_tippable-staff/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  IdentityType: () => IdentityType,
  listTippableStaff: () => listTippableStaff2
});
module.exports = __toCommonJS(index_typings_exports);

// src/tips-staff-v1-staff-tippable-staff.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/tips-staff-v1-staff-tippable-staff.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixTipsStaffV1TipStaffServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/tips-staff-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/tips-staff",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/tips/staff/v1/eligible-staff",
        destPath: "/v1/eligible-staff"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tippable-staff";
function listTippableStaff(payload) {
  function __listTippableStaff({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.staff.v1.staff",
      method: "GET",
      methodFqn: "wix.tips.staff.v1.TipStaffService.ListTippableStaff",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsStaffV1TipStaffServiceUrl({
        protoPath: "/v1/eligible-staff",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listTippableStaff;
}

// src/tips-staff-v1-staff-tippable-staff.universal.ts
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
async function listTippableStaff2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    includeWixUsers: options?.includeWixUsers,
    filterByName: options?.filterByName,
    filterByIds: options?.filterByIds
  });
  const reqOpts = listTippableStaff(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          includeWixUsers: "$[0].includeWixUsers",
          filterByName: "$[0].filterByName",
          filterByIds: "$[0].filterByIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  IdentityType,
  listTippableStaff
});
