// @wix/auto_sdk_ecom_tips/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  IdentityTypeOriginal: () => IdentityType,
  QueryByOriginal: () => QueryBy,
  SortOrderOriginal: () => SortOrder,
  SourceOriginal: () => Source,
  TipTypeOriginal: () => TipType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  addTipDistributionsToOrder: () => addTipDistributionsToOrder2,
  getTipByDraftOrderId: () => getTipByDraftOrderId2,
  getTipByOrderId: () => getTipByOrderId2,
  getTipChoice: () => getTipChoice2,
  previewTip: () => previewTip2,
  queryTipDistributions: () => queryTipDistributions2,
  removeTipFromOrder: () => removeTipFromOrder2,
  setTipOnOrder: () => setTipOnOrder2,
  sumTipDistributionAmounts: () => sumTipDistributionAmounts2,
  updateTipChoice: () => updateTipChoice2
});
module.exports = __toCommonJS(meta_exports);

// src/tips-v1-tip-tips.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixTipsTipV1TipsServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/tips-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/tips-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/tips",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/tips-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/tips/v1/tips",
        destPath: "/v1/tips"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_tips";
function previewTip(payload) {
  function __previewTip({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.PreviewTip",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/preview",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tip.distributions.createdDate" },
            { path: "tip.distributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __previewTip;
}
function getTipByOrderId(payload) {
  function __getTipByOrderId({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "GET",
      methodFqn: "wix.tips.tip.v1.TipsService.GetTipByOrderId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/order/{orderId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tip.distributions.createdDate" },
            { path: "tip.distributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTipByOrderId;
}
function getTipByDraftOrderId(payload) {
  function __getTipByDraftOrderId({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "GET",
      methodFqn: "wix.tips.tip.v1.TipsService.GetTipByDraftOrderId",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/draft-order/{draftOrderId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tip.distributions.createdDate" },
            { path: "tip.distributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTipByDraftOrderId;
}
function getTipChoice(payload) {
  function __getTipChoice({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "GET",
      methodFqn: "wix.tips.tip.v1.TipsService.GetTipChoice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/choice/{purchaseFlowId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getTipChoice;
}
function updateTipChoice(payload) {
  function __updateTipChoice({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "PATCH",
      methodFqn: "wix.tips.tip.v1.TipsService.UpdateTipChoice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/choice",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateTipChoice;
}
function setTipOnOrder(payload) {
  function __setTipOnOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "tip.distributions.createdDate" },
          { path: "tip.distributions.transaction.date" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.SetTipOnOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/set",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tip.distributions.createdDate" },
            { path: "tip.distributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __setTipOnOrder;
}
function removeTipFromOrder(payload) {
  function __removeTipFromOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.RemoveTipFromOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/remove",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __removeTipFromOrder;
}
function addTipDistributionsToOrder(payload) {
  function __addTipDistributionsToOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "tipDistributions.createdDate" },
          { path: "tipDistributions.transaction.date" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.AddTipDistributionsToOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/tip-distributions/add",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tipDistributions.createdDate" },
            { path: "tipDistributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addTipDistributionsToOrder;
}
function queryTipDistributions(payload) {
  function __queryTipDistributions({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.QueryTipDistributions",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/tip-distributions/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tipDistributions.createdDate" },
            { path: "tipDistributions.transaction.date" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryTipDistributions;
}
function sumTipDistributionAmounts(payload) {
  function __sumTipDistributionAmounts({ host }) {
    const metadata = {
      entityFqdn: "wix.tips.v1.tip",
      method: "POST",
      methodFqn: "wix.tips.tip.v1.TipsService.SumTipDistributionAmounts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixTipsTipV1TipsServiceUrl({
        protoPath: "/v1/tips/tip-distributions/sum-amounts",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __sumTipDistributionAmounts;
}

// src/tips-v1-tip-tips.types.ts
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["UNKNOWN_SOURCE"] = "UNKNOWN_SOURCE";
  Source2["POS"] = "POS";
  Source2["WEB"] = "WEB";
  Source2["DASHBOARD"] = "DASHBOARD";
  return Source2;
})(Source || {});
var TipType = /* @__PURE__ */ ((TipType2) => {
  TipType2["UNKNOWN_TIP_TYPE"] = "UNKNOWN_TIP_TYPE";
  TipType2["PERCENTAGE"] = "PERCENTAGE";
  TipType2["AMOUNT"] = "AMOUNT";
  TipType2["CUSTOM"] = "CUSTOM";
  return TipType2;
})(TipType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var QueryBy = /* @__PURE__ */ ((QueryBy2) => {
  QueryBy2["UNKNOWN_QUERY_BY"] = "UNKNOWN_QUERY_BY";
  QueryBy2["ORDER"] = "ORDER";
  QueryBy2["TIP"] = "TIP";
  return QueryBy2;
})(QueryBy || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/tips-v1-tip-tips.meta.ts
function previewTip2() {
  const payload = {};
  const getRequestOptions = previewTip(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/preview",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTipByOrderId2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = getTipByOrderId(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tips/order/{orderId}",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTipByDraftOrderId2() {
  const payload = { draftOrderId: ":draftOrderId" };
  const getRequestOptions = getTipByDraftOrderId(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tips/draft-order/{draftOrderId}",
    pathParams: { draftOrderId: "draftOrderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTipChoice2() {
  const payload = { purchaseFlowId: ":purchaseFlowId" };
  const getRequestOptions = getTipChoice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/tips/choice/{purchaseFlowId}",
    pathParams: { purchaseFlowId: "purchaseFlowId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateTipChoice2() {
  const payload = {};
  const getRequestOptions = updateTipChoice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/tips/choice",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setTipOnOrder2() {
  const payload = {};
  const getRequestOptions = setTipOnOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/set",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeTipFromOrder2() {
  const payload = {};
  const getRequestOptions = removeTipFromOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/remove",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addTipDistributionsToOrder2() {
  const payload = {};
  const getRequestOptions = addTipDistributionsToOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/tip-distributions/add",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryTipDistributions2() {
  const payload = {};
  const getRequestOptions = queryTipDistributions(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/tip-distributions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function sumTipDistributionAmounts2() {
  const payload = {};
  const getRequestOptions = sumTipDistributionAmounts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/tips/tip-distributions/sum-amounts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  IdentityTypeOriginal,
  QueryByOriginal,
  SortOrderOriginal,
  SourceOriginal,
  TipTypeOriginal,
  WebhookIdentityTypeOriginal,
  addTipDistributionsToOrder,
  getTipByDraftOrderId,
  getTipByOrderId,
  getTipChoice,
  previewTip,
  queryTipDistributions,
  removeTipFromOrder,
  setTipOnOrder,
  sumTipDistributionAmounts,
  updateTipChoice
});
