// @wix/auto_sdk_ecom_orders-settings/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  InventoryUpdateTrigger: () => InventoryUpdateTrigger,
  WebhookIdentityType: () => WebhookIdentityType,
  getOrdersSettings: () => getOrdersSettings4,
  onOrdersSettingsUpdated: () => onOrdersSettingsUpdated2,
  updateOrdersSettings: () => updateOrdersSettings4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-v1-orders-settings-orders-settings.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/ecom-v1-orders-settings-orders-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-v1-orders-settings-orders-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/orders-settings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_orders-settings";
function getOrdersSettings(payload) {
  function __getOrdersSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.orders_settings",
      method: "GET",
      methodFqn: "wix.ecom.orders_settings.api.v1.OrdersSettingsService.GetOrdersSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
        protoPath: "/v1/orders-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ordersSettings.createdDate" },
            { path: "ordersSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrdersSettings;
}
function updateOrdersSettings(payload) {
  function __updateOrdersSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ordersSettings.createdDate" },
          { path: "ordersSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.orders_settings",
      method: "PATCH",
      methodFqn: "wix.ecom.orders_settings.api.v1.OrdersSettingsService.UpdateOrdersSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
        protoPath: "/v1/orders-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ordersSettings.createdDate" },
            { path: "ordersSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrdersSettings;
}

// src/ecom-v1-orders-settings-orders-settings.universal.ts
var InventoryUpdateTrigger = /* @__PURE__ */ ((InventoryUpdateTrigger2) => {
  InventoryUpdateTrigger2["UNKNOWN_INVENTORY_UPDATE_TRIGGER"] = "UNKNOWN_INVENTORY_UPDATE_TRIGGER";
  InventoryUpdateTrigger2["ON_ORDER_PAID"] = "ON_ORDER_PAID";
  InventoryUpdateTrigger2["ON_ORDER_PLACED"] = "ON_ORDER_PLACED";
  return InventoryUpdateTrigger2;
})(InventoryUpdateTrigger || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getOrdersSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getOrdersSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrdersSettings2(ordersSettings) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ordersSettings
  });
  const reqOpts = updateOrdersSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ordersSettings: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ordersSettings"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-v1-orders-settings-orders-settings.public.ts
function getOrdersSettings3(httpClient) {
  return () => getOrdersSettings2(
    // @ts-ignore
    { httpClient }
  );
}
function updateOrdersSettings3(httpClient) {
  return (ordersSettings) => updateOrdersSettings2(
    ordersSettings,
    // @ts-ignore
    { httpClient }
  );
}
var onOrdersSettingsUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.orders_settings_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "metadata.eventTime" },
          { path: "modifiedFields.createdDate" },
          { path: "modifiedFields.updatedDate" }
        ]
      }
    ])
  )
)();

// src/ecom-v1-orders-settings-orders-settings.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var getOrdersSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOrdersSettings3);
var updateOrdersSettings4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOrdersSettings3);
var onOrdersSettingsUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onOrdersSettingsUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  InventoryUpdateTrigger,
  WebhookIdentityType,
  getOrdersSettings,
  onOrdersSettingsUpdated,
  updateOrdersSettings
});
