// @wix/auto_sdk_ecom_orders-settings/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  InventoryUpdateTriggerOriginal: () => InventoryUpdateTrigger,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  getOrdersSettings: () => getOrdersSettings2,
  updateOrdersSettings: () => updateOrdersSettings2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-orders-settings-orders-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/orders-settings",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/orders-settings",
        destPath: "/v1/orders-settings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_orders-settings";
function getOrdersSettings(payload) {
  function __getOrdersSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.orders_settings",
      method: "GET",
      methodFqn: "wix.ecom.orders_settings.api.v1.OrdersSettingsService.GetOrdersSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
        protoPath: "/v1/orders-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ordersSettings.createdDate" },
            { path: "ordersSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrdersSettings;
}
function updateOrdersSettings(payload) {
  function __updateOrdersSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "ordersSettings.createdDate" },
          { path: "ordersSettings.updatedDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.orders_settings",
      method: "PATCH",
      methodFqn: "wix.ecom.orders_settings.api.v1.OrdersSettingsService.UpdateOrdersSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
        protoPath: "/v1/orders-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "ordersSettings.createdDate" },
            { path: "ordersSettings.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrdersSettings;
}

// src/ecom-v1-orders-settings-orders-settings.types.ts
var InventoryUpdateTrigger = /* @__PURE__ */ ((InventoryUpdateTrigger2) => {
  InventoryUpdateTrigger2["UNKNOWN_INVENTORY_UPDATE_TRIGGER"] = "UNKNOWN_INVENTORY_UPDATE_TRIGGER";
  InventoryUpdateTrigger2["ON_ORDER_PAID"] = "ON_ORDER_PAID";
  InventoryUpdateTrigger2["ON_ORDER_PLACED"] = "ON_ORDER_PLACED";
  return InventoryUpdateTrigger2;
})(InventoryUpdateTrigger || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/ecom-v1-orders-settings-orders-settings.meta.ts
function getOrdersSettings2() {
  const payload = {};
  const getRequestOptions = getOrdersSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/orders-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateOrdersSettings2() {
  const payload = {};
  const getRequestOptions = updateOrdersSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/orders-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  InventoryUpdateTriggerOriginal,
  WebhookIdentityTypeOriginal,
  getOrdersSettings,
  updateOrdersSettings
});
