// @wix/auto_sdk_stores_ribbons-v-3/build/internal/es/src/stores-catalog-v3-ribbon-ribbons-v-3.context.js
import { createRibbon as publicCreateRibbon, getRibbon as publicGetRibbon, updateRibbon as publicUpdateRibbon, deleteRibbon as publicDeleteRibbon, queryRibbons as publicQueryRibbons, bulkCreateRibbons as publicBulkCreateRibbons, bulkUpdateRibbons as publicBulkUpdateRibbons, getOrCreateRibbon as publicGetOrCreateRibbon, bulkGetOrCreateRibbons as publicBulkGetOrCreateRibbons, bulkDeleteRibbons as publicBulkDeleteRibbons, } from './stores-catalog-v3-ribbon-ribbons-v-3.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onRibbonCreated as publicOnRibbonCreated } from './stores-catalog-v3-ribbon-ribbons-v-3.public.js';
import { onRibbonDeleted as publicOnRibbonDeleted } from './stores-catalog-v3-ribbon-ribbons-v-3.public.js';
import { onRibbonUpdated as publicOnRibbonUpdated } from './stores-catalog-v3-ribbon-ribbons-v-3.public.js';
export const createRibbon = /*#__PURE__*/ createRESTModule(publicCreateRibbon);
export const getRibbon = /*#__PURE__*/ createRESTModule(publicGetRibbon);
export const updateRibbon = /*#__PURE__*/ createRESTModule(publicUpdateRibbon);
export const deleteRibbon = /*#__PURE__*/ createRESTModule(publicDeleteRibbon);
export const queryRibbons = /*#__PURE__*/ createRESTModule(publicQueryRibbons);
export const bulkCreateRibbons = /*#__PURE__*/ createRESTModule(publicBulkCreateRibbons);
export const bulkUpdateRibbons = /*#__PURE__*/ createRESTModule(publicBulkUpdateRibbons);
export const getOrCreateRibbon = /*#__PURE__*/ createRESTModule(publicGetOrCreateRibbon);
export const bulkGetOrCreateRibbons = /*#__PURE__*/ createRESTModule(publicBulkGetOrCreateRibbons);
export const bulkDeleteRibbons = /*#__PURE__*/ createRESTModule(publicBulkDeleteRibbons);
/** */
export const onRibbonCreated = createEventModule(publicOnRibbonCreated);
/** */
export const onRibbonDeleted = createEventModule(publicOnRibbonDeleted);
/** */
export const onRibbonUpdated = createEventModule(publicOnRibbonUpdated);
export { RequestedFields, SortOrder, WebhookIdentityType, } from './stores-catalog-v3-ribbon-ribbons-v-3.universal.js';
