// @wix/auto_sdk_stores_stores-locations-v-3/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  LocationTypeOriginal: () => LocationType,
  SortOrderOriginal: () => SortOrder,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  getStoresLocation: () => getStoresLocation2,
  queryStoresLocations: () => queryStoresLocations2
});
module.exports = __toCommonJS(meta_exports);

// src/stores-catalog-v3-stores-location-stores-locations-v-3.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixStoresCatalogLocationV3StoresLocationServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/location",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/location",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/location",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v3/locations",
        destPath: "/v3/locations"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_stores-locations-v-3";
function getStoresLocation(payload) {
  function __getStoresLocation({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.stores_location",
      method: "GET",
      methodFqn: "wix.stores.catalog.location.v3.StoresLocationService.GetStoresLocation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixStoresCatalogLocationV3StoresLocationServiceUrl({
        protoPath: "/v3/locations/{storesLocationId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "storesLocation.createdDate" },
            { path: "storesLocation.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getStoresLocation;
}
function queryStoresLocations(payload) {
  function __queryStoresLocations({ host }) {
    const metadata = {
      entityFqdn: "wix.stores.catalog.v3.stores_location",
      method: "POST",
      methodFqn: "wix.stores.catalog.location.v3.StoresLocationService.QueryStoresLocations",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixStoresCatalogLocationV3StoresLocationServiceUrl({
        protoPath: "/v3/locations/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "storesLocations.createdDate" },
            { path: "storesLocations.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryStoresLocations;
}

// src/stores-catalog-v3-stores-location-stores-locations-v-3.types.ts
var LocationType = /* @__PURE__ */ ((LocationType2) => {
  LocationType2["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
  LocationType2["VIRTUAL"] = "VIRTUAL";
  LocationType2["PHYSICAL"] = "PHYSICAL";
  return LocationType2;
})(LocationType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/stores-catalog-v3-stores-location-stores-locations-v-3.meta.ts
function getStoresLocation2() {
  const payload = { storesLocationId: ":storesLocationId" };
  const getRequestOptions = getStoresLocation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/locations/{storesLocationId}",
    pathParams: { storesLocationId: "storesLocationId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryStoresLocations2() {
  const payload = {};
  const getRequestOptions = queryStoresLocations(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/locations/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LocationTypeOriginal,
  SortOrderOriginal,
  WebhookIdentityTypeOriginal,
  getStoresLocation,
  queryStoresLocations
});
