// @wix/auto_sdk_stores_products-v-3/build/internal/es/src/stores-catalog-v3-product-products-v-3.meta.js
import * as ambassadorWixStoresCatalogV3Product from './stores-catalog-v3-product-products-v-3.http.js';
export function createProduct() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.createProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createProductWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.createProductWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products-with-inventory',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateProduct() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/products/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateProductWithInventory() {
    const payload = { product: { id: ':productId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateProductWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/products-with-inventory/{product.id}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkCreateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateProductsWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkCreateProductsWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products-with-inventory/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateProductsWithInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductsWithInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products-with-inventory/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/{productId}/update-extended-fields',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.deleteProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/products/{productId}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkDeleteProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkDeleteProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/delete-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.getProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/products/{productId}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getProductBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.getProductBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/products/slug/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.searchProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.queryProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.countProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/products/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateProductVariantsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkUpdateProductVariantsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/update-variants-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAdjustProductVariantsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAdjustProductVariantsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/adjust-variants-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAddInfoSectionsToProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddInfoSectionsToProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-info-sections-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAddInfoSectionsToProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddInfoSectionsToProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkRemoveInfoSectionsFromProductsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveInfoSectionsFromProductsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-info-sections-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkRemoveInfoSectionsFromProducts() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveInfoSectionsFromProducts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAddProductsToCategoriesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkAddProductsToCategoriesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/add-to-categories-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkRemoveProductsFromCategoriesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Product.bulkRemoveProductsFromCategoriesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/products/remove-from-categories-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
