// @wix/auto_sdk_ecom_recommendations-provider/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  AlgorithmNotSupportedWixError: () => AlgorithmNotSupportedWixError,
  AlgorithmType: () => AlgorithmType,
  IdentityType: () => IdentityType,
  ItemAppIdNotSupportedWixError: () => ItemAppIdNotSupportedWixError
});
module.exports = __toCommonJS(index_typings_exports);

// src/service-plugins-error-classes.ts
var AlgorithmNotSupportedWixError = class extends Error {
  constructor(data = {}) {
    super("AlgorithmNotSupportedError");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "ALGORITHM_NOT_SUPPORTED";
    this.name = "AlgorithmNotSupportedError";
    this.errorSchemaName = "com.wix.ecom.recommendations.spi.RecommendationAlgorithmNotSupported";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "AlgorithmNotSupportedError",
      applicationCode: "ALGORITHM_NOT_SUPPORTED"
    };
    this.data = data;
  }
};
/** @hidden */
AlgorithmNotSupportedWixError.__type = "wix_spi_error";
var ItemAppIdNotSupportedWixError = class extends Error {
  constructor(data = {}) {
    super("ItemAppIdNotSupportedError");
    this.httpCode = 400;
    this.statusCode = "INVALID_ARGUMENT";
    this.applicationCode = "ITEM_APP_ID_NOT_SUPPORTED";
    this.name = "ItemAppIdNotSupportedError";
    this.errorSchemaName = "com.wix.ecom.recommendations.spi.ItemAppIdNotSupportedError";
    this.errorType = "SPI";
    this.spiErrorData = {
      name: "ItemAppIdNotSupportedError",
      applicationCode: "ITEM_APP_ID_NOT_SUPPORTED"
    };
    this.data = data;
  }
};
/** @hidden */
ItemAppIdNotSupportedWixError.__type = "wix_spi_error";

// src/service-plugins-types.ts
var AlgorithmType = /* @__PURE__ */ ((AlgorithmType2) => {
  AlgorithmType2["UNSPECIFIED"] = "UNSPECIFIED";
  AlgorithmType2["RELATED_ITEMS"] = "RELATED_ITEMS";
  AlgorithmType2["GLOBAL"] = "GLOBAL";
  return AlgorithmType2;
})(AlgorithmType || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlgorithmNotSupportedWixError,
  AlgorithmType,
  IdentityType,
  ItemAppIdNotSupportedWixError
});
