// @wix/auto_sdk_stores_products/build/internal/es/src/stores-catalog-v1-product-products.universal.js
import { transformError as sdkTransformError } from '@wix/sdk-runtime/transform-error';
import { queryBuilder } from '@wix/sdk-runtime/query-builder';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
import * as ambassadorWixStoresCatalogV1Product from './stores-catalog-v1-product-products.http.js';
// @ts-ignore
import { transformRESTImageToSDKImage } from '@wix/sdk-runtime/transformations/image';
import { transformRESTVideoV2ToSDKVideoV2 } from '@wix/sdk-runtime/transformations/video-v2';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
export var ProductType;
(function (ProductType) {
    ProductType["unspecified_product_type"] = "unspecified_product_type";
    ProductType["physical"] = "physical";
    ProductType["digital"] = "digital";
})(ProductType || (ProductType = {}));
export var InventoryStatus;
(function (InventoryStatus) {
    /** In stock */
    InventoryStatus["IN_STOCK"] = "IN_STOCK";
    /** Not in stock */
    InventoryStatus["OUT_OF_STOCK"] = "OUT_OF_STOCK";
    /** Some of the variants are not in stock */
    InventoryStatus["PARTIALLY_OUT_OF_STOCK"] = "PARTIALLY_OUT_OF_STOCK";
})(InventoryStatus || (InventoryStatus = {}));
export var MeasurementUnit;
(function (MeasurementUnit) {
    MeasurementUnit["UNSPECIFIED"] = "UNSPECIFIED";
    MeasurementUnit["ML"] = "ML";
    MeasurementUnit["CL"] = "CL";
    MeasurementUnit["L"] = "L";
    MeasurementUnit["CBM"] = "CBM";
    MeasurementUnit["MG"] = "MG";
    MeasurementUnit["G"] = "G";
    MeasurementUnit["KG"] = "KG";
    MeasurementUnit["MM"] = "MM";
    MeasurementUnit["CM"] = "CM";
    MeasurementUnit["M"] = "M";
    MeasurementUnit["SQM"] = "SQM";
    MeasurementUnit["OZ"] = "OZ";
    MeasurementUnit["LB"] = "LB";
    MeasurementUnit["FLOZ"] = "FLOZ";
    MeasurementUnit["PT"] = "PT";
    MeasurementUnit["QT"] = "QT";
    MeasurementUnit["GAL"] = "GAL";
    MeasurementUnit["IN"] = "IN";
    MeasurementUnit["FT"] = "FT";
    MeasurementUnit["YD"] = "YD";
    MeasurementUnit["SQFT"] = "SQFT";
})(MeasurementUnit || (MeasurementUnit = {}));
export var MediaItemType;
(function (MediaItemType) {
    MediaItemType["unspecified_media_item_type"] = "unspecified_media_item_type";
    /** Image media type. */
    MediaItemType["image"] = "image";
    /** Video media type. */
    MediaItemType["video"] = "video";
    /** Audio media type. */
    MediaItemType["audio"] = "audio";
    /** Document media type. */
    MediaItemType["document"] = "document";
    /** Zip media type. */
    MediaItemType["zip"] = "zip";
})(MediaItemType || (MediaItemType = {}));
export var OptionType;
(function (OptionType) {
    /** Unspecified option type. */
    OptionType["unspecified_option_type"] = "unspecified_option_type";
    /** Drop down. */
    OptionType["drop_down"] = "drop_down";
    /** Color. */
    OptionType["color"] = "color";
})(OptionType || (OptionType = {}));
export var DiscountType;
(function (DiscountType) {
    DiscountType["UNDEFINED"] = "UNDEFINED";
    /** No discount */
    DiscountType["NONE"] = "NONE";
    /** Discount by a fixed amount */
    DiscountType["AMOUNT"] = "AMOUNT";
    /** Discount by a percentage */
    DiscountType["PERCENT"] = "PERCENT";
})(DiscountType || (DiscountType = {}));
export var FileType;
(function (FileType) {
    FileType["UNSPECIFIED"] = "UNSPECIFIED";
    FileType["SECURE_PICTURE"] = "SECURE_PICTURE";
    FileType["SECURE_VIDEO"] = "SECURE_VIDEO";
    FileType["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
    FileType["SECURE_MUSIC"] = "SECURE_MUSIC";
    FileType["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
})(FileType || (FileType = {}));
export var Version;
(function (Version) {
    /** Version 1 of the catalog. */
    Version["V1_CATALOG"] = "V1_CATALOG";
    /** Version 3 of the catalog. */
    Version["V3_CATALOG"] = "V3_CATALOG";
})(Version || (Version = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
/**
 * Creates a new product.
 * @param product - Product information.
 * @public
 * @requiredField product
 * @requiredField product.costAndProfitData.itemCost
 * @requiredField product.name
 * @requiredField product.priceData
 * @requiredField product.priceData.price
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.CreateProduct
 */
export async function createProduct(product) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ product: product });
    const reqOpts = ambassadorWixStoresCatalogV1Product.createProduct(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { product: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['product']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Updates specified fields in a product.
 * @param _id - Product ID (generated automatically by the catalog).
 * @public
 * @requiredField _id
 * @requiredField product
 * @param product - Product info to update.
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.UpdateProduct
 */
export async function updateProduct(_id, product) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        product: { ...product, id: _id },
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.updateProduct(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: { product: '$[1]' },
            explicitPathsToArguments: { 'product.id': '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'product']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Deletes a product.
 * @param _id - ID of the product to delete.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.DeleteProduct
 */
export async function deleteProduct(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.deleteProduct(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Updates variants of a specified product.
 * @param _id - ID of the product with managed variants.
 * @param variants - Variant info to update.
 * @public
 * @requiredField _id
 * @requiredField variants
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.UpdateVariants
 */
export async function updateProductVariants(_id, variants) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        variants: variants,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.updateVariants(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', variants: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'variants']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Resets the data (such as the price and the weight) of all variants for a given product to their default values.
 * @param _id - Product ID.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.ResetAllVariantData
 */
export async function resetAllProductVariantData(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.resetAllVariantData(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Adds products to a specified collection.
 * @param _id - Collection ID.
 * @param productIds - IDs of the products to add to the collection, separated by commas.
 * @public
 * @requiredField _id
 * @requiredField productIds
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.AddProductsToCollection
 */
export async function addProductsToCollection(_id, productIds) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        productIds: productIds,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.addProductsToCollection(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', productIds: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'productIds']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Deletes products from a specified collection.
 * @param _id - ID of the collection from which to remove products.
 * @param productIds - IDs of the products to remove from the collection.
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @requiredField productIds
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.RemoveProductsFromCollection
 */
export async function removeProductsFromCollection(_id, productIds) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        productIds: productIds,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.removeProductsFromCollection(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', productIds: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'productIds']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Adds media items to a specified product, either via URL or existing media ID.
 *
 * > **NOTE:** The URL is not validated and no event is triggered to indicate if the media was added successfully.
 * @param _id - Product ID.
 * @param media - Sources of media items already uploaded to the Wix site.
 * @public
 * @requiredField _id
 * @requiredField media
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.AddProductMedia
 */
export async function addProductMedia(_id, media) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        media: media,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.addProductMedia(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', media: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'media']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Removes specified media items from a product.
 * Pass an empty array to remove all media items.
 * @param _id - Product ID.
 * @param mediaIds - List of media IDs to remove. Pass an empty array to delete all media items for the product.
 * @public
 * @requiredField _id
 * @requiredField mediaIds
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.RemoveProductMedia
 */
export async function removeProductMedia(_id, mediaIds) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        mediaIds: mediaIds,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.removeProductMedia(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', mediaIds: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'mediaIds']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Links media items that are already associated with a specific product to a choice within the same product.
 *
 * Media items can only be set for choices within one option at a time - e.g., if you set media items for some or all of the choices within the Colors option (blue, green, and red), you won't be able to also assign media items to choices within the Size option (S, M, and L).
 *
 * To remove all existing media items, call the [Remove Product Media From Choices](https://dev.wix.com/api/rest/wix-stores/catalog/products/remove-product-media-from-choices) endpoint.
 * @param _id - Product ID.
 * @param media - Product media items and the choices to add the media to.
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @requiredField media
 * @requiredField media.choice
 * @requiredField media.option
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.AddProductMediaToChoices
 */
export async function addProductMediaToChoices(_id, media) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        media: media,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.addProductMediaToChoices(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', media: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'media']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Removes media items from all or some of a product's choices.
 * (Media items can only be set for choices within one option at a time - e.g., if you set media items for some or all of the choices within the Colors option (blue, green, and red), you won't be able to also assign media items to choices within the Size option (S, M, and L).)
 * @param _id - Product ID from whose choices to remove media items.
 * @param media - Media to remove from choices. If an empty array is passed, all media will be removed from all choices for the given product.
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @requiredField media
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.RemoveProductMediaFromChoices
 */
export async function removeProductMediaFromChoices(_id, media) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        media: media,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.removeProductMediaFromChoices(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', media: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'media']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Delete all options from a specific product. Only available when [variant management](https://support.wix.com/en/article/wix-stores-adding-and-customizing-product-options#setting-different-prices-for-variants) is disabled.
 * @param _id - ID of the product with options to delete.
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.DeleteProductOptions
 */
export async function deleteProductOptions(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.deleteProductOptions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Deletes a product's brand.
 * @param _id - Product ID.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.RemoveBrand
 */
export async function removeBrand(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.removeBrand(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Creates a new collection.
 * @param collection - Collection info.
 * @public
 * @requiredField collection
 * @requiredField collection.name
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.CreateCollection
 */
export async function createCollection(collection) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        collection: collection,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.createCollection(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { collection: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['collection']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Updates specified properties of a collection. To add products to a collection, call the [addProductsToCollection](#addproductstocollection) function.
 * @param _id - Collection ID (generated automatically by the catalog).
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @requiredField collection
 * @param collection - Collection info to update.
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.UpdateCollection
 */
export async function updateCollection(_id, collection) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        collection: { ...collection, id: _id },
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.updateCollection(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: { collection: '$[1]' },
            explicitPathsToArguments: { 'collection.id': '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'collection']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Deletes a collection.
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @param _id - ID of the collection to delete.
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.DeleteCollection
 */
export async function deleteCollection(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.deleteCollection(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Deletes a product's ribbon.
 * @param _id - Product ID.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.RemoveRibbon
 */
export async function removeRibbon(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.removeRibbon(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Updates a specified property for up to 100 products at a time.
 * @param ids - Product IDs.
 * @param set - Field to update.
 * @public
 * @documentationMaturity preview
 * @requiredField ids
 * @requiredField set
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.BulkUpdateProducts
 */
export async function bulkUpdateProductsProperty(ids, set) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({ ids: ids, set: set });
    const reqOpts = ambassadorWixStoresCatalogV1Product.bulkUpdateProducts(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { ids: '$[0]', set: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['ids', 'set']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Adjusts a specified numerical property for up to 100 products at a time.
 * The property can be increased or decreased either by percentage or amount.
 * @param adjust - Numerical property to adjust.
 * @param ids - Product IDs.
 * @public
 * @documentationMaturity preview
 * @requiredField adjust
 * @requiredField ids
 * @permissionId WIX_STORES.MODIFY_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @applicableIdentity APP
 * @fqn wix.catalog.api.v1.CatalogWriteApi.BulkAdjustProductProperties
 */
export async function bulkAdjustProductProperty(adjust, ids) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        adjust: adjust,
        ids: ids,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.bulkAdjustProductProperties(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { adjust: '$[0]', ids: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['adjust', 'ids']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Returns a list of up to 100 products, given the provided paging, sorting and filtering.
 * @public
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.QueryProductsPlatformized
 */
export function queryProducts() {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[0];
    return queryBuilder({
        func: async (payload) => {
            const reqOpts = ambassadorWixStoresCatalogV1Product.queryProductsPlatformized(payload);
            sideEffects?.onSiteCall?.();
            try {
                const result = await httpClient.request(reqOpts);
                sideEffects?.onSuccess?.(result);
                return result;
            }
            catch (err) {
                sideEffects?.onError?.(err);
                throw err;
            }
        },
        requestTransformer: (query) => {
            const args = [query, {}];
            return renameKeysFromSDKRequestToRESTRequest({
                ...args?.[1],
                query: args?.[0],
            });
        },
        responseTransformer: ({ data, }) => {
            const transformedData = renameKeysFromRESTResponseToSDKResponse(data);
            return {
                items: transformedData?.products,
                pagingMetadata: transformedData?.metadata,
            };
        },
        errorTransformer: (err) => {
            const transformedError = sdkTransformError(err, {
                spreadPathsToArguments: {},
                explicitPathsToArguments: { query: '$[0]' },
                singleArgumentUnchanged: false,
            });
            throw transformedError;
        },
        pagingMethod: 'OFFSET',
        transformationPaths: {},
    });
}
/**
 * Retrieves a product with the provided ID.
 * @param _id - Requested product ID.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.GetProduct
 */
export async function getProduct(_id, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        includeMerchantSpecificData: options?.includeMerchantSpecificData,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.getProduct(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                id: '$[0]',
                includeMerchantSpecificData: '$[1].includeMerchantSpecificData',
            },
            singleArgumentUnchanged: false,
        }, ['_id', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Retrieves a collection with the provided slug.
 * @param slug - Slug of the collection to retrieve.
 * @public
 * @requiredField slug
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.GetCollectionBySlug
 */
export async function getCollectionBySlug(slug) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ slug: slug });
    const reqOpts = ambassadorWixStoresCatalogV1Product.getCollectionBySlug(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { slug: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['slug']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Gets the availability of relevant product variants based on the product ID and selections provided. See [Use Cases](https://dev.wix.com/api/rest/wix-stores/catalog/use-cases) for an example.
 * @param _id - Requested product ID.
 * @param options - Array containing the selected options. For example, `["color": "Blue", "size": "Large"]`.
 * @public
 * @requiredField _id
 * @requiredField options
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.ProductOptionsAvailability
 */
export async function getProductOptionsAvailability(_id, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        options: options,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.productOptionsAvailability(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]', options: '$[1]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Retrieves product variants, based on either choices (option-choice key-value pairs) or variant IDs.
 * See [Stores Pagination](https://dev.wix.com/api/rest/wix-stores/pagination) for more information.
 * @param _id - Requested product ID.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.QueryProductVariants
 */
export async function queryProductVariants(_id, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        id: _id,
        choices: options?.choices,
        variantIds: options?.variantIds,
        paging: options?.paging,
        includeMerchantSpecificData: options?.includeMerchantSpecificData,
    });
    const reqOpts = ambassadorWixStoresCatalogV1Product.queryProductVariants(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(result.data);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                id: '$[0]',
                choices: '$[1].choices',
                variantIds: '$[1].variantIds',
                paging: '$[1].paging',
                includeMerchantSpecificData: '$[1].includeMerchantSpecificData',
            },
            singleArgumentUnchanged: false,
        }, ['_id', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Retrieves up to 100 store variants, given the provided paging, filtering, and sorting.
 * @param query - Query options.
 * @public
 * @requiredField query
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.QueryStoreVariants
 */
export async function queryStoreVariants(query) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ query: query });
    const reqOpts = ambassadorWixStoresCatalogV1Product.queryStoreVariants(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'variants.media.image' }],
            },
            {
                transformFn: transformRESTVideoV2ToSDKVideoV2,
                paths: [{ path: 'variants.media.video' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { query: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['query']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Retrieves a store variant with the provided ID.
 * @param _id - Store variant ID. Comprised of the `productId` and the `variantId`, separated by a hyphen. For example, `{productId}-{variantId}`.
 * @public
 * @requiredField _id
 * @permissionId WIX_STORES.READ_PRODUCTS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Products
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-PRODUCTS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Read Products
 * @permissionScopeId SCOPE.DC-STORES.READ-PRODUCTS
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn wix.catalog.api.v1.CatalogReadApi.GetStoreVariant
 */
export async function getStoreVariant(_id) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({ id: _id });
    const reqOpts = ambassadorWixStoresCatalogV1Product.getStoreVariant(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'variant.media.image' }],
            },
            {
                transformFn: transformRESTVideoV2ToSDKVideoV2,
                paths: [{ path: 'variant.media.video' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { id: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
