// @wix/auto_sdk_stores_products/build/internal/es/src/stores-catalog-v1-product-products.context.js
import { createProduct as publicCreateProduct, updateProduct as publicUpdateProduct, deleteProduct as publicDeleteProduct, updateProductVariants as publicUpdateProductVariants, resetAllProductVariantData as publicResetAllProductVariantData, addProductsToCollection as publicAddProductsToCollection, removeProductsFromCollection as publicRemoveProductsFromCollection, addProductMedia as publicAddProductMedia, removeProductMedia as publicRemoveProductMedia, addProductMediaToChoices as publicAddProductMediaToChoices, removeProductMediaFromChoices as publicRemoveProductMediaFromChoices, deleteProductOptions as publicDeleteProductOptions, removeBrand as publicRemoveBrand, createCollection as publicCreateCollection, updateCollection as publicUpdateCollection, deleteCollection as publicDeleteCollection, removeRibbon as publicRemoveRibbon, bulkUpdateProductsProperty as publicBulkUpdateProductsProperty, bulkAdjustProductProperty as publicBulkAdjustProductProperty, queryProducts as publicQueryProducts, getProduct as publicGetProduct, getCollectionBySlug as publicGetCollectionBySlug, getProductOptionsAvailability as publicGetProductOptionsAvailability, queryProductVariants as publicQueryProductVariants, queryStoreVariants as publicQueryStoreVariants, getStoreVariant as publicGetStoreVariant, } from './stores-catalog-v1-product-products.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onProductCollectionChanged as publicOnProductCollectionChanged } from './stores-catalog-v1-product-products.public.js';
import { onProductCollectionCreated as publicOnProductCollectionCreated } from './stores-catalog-v1-product-products.public.js';
import { onProductCollectionDeleted as publicOnProductCollectionDeleted } from './stores-catalog-v1-product-products.public.js';
import { onProductChanged as publicOnProductChanged } from './stores-catalog-v1-product-products.public.js';
import { onProductCreated as publicOnProductCreated } from './stores-catalog-v1-product-products.public.js';
import { onProductDeleted as publicOnProductDeleted } from './stores-catalog-v1-product-products.public.js';
import { onProductVariantsChanged as publicOnProductVariantsChanged } from './stores-catalog-v1-product-products.public.js';
export const createProduct = /*#__PURE__*/ createRESTModule(publicCreateProduct);
export const updateProduct = /*#__PURE__*/ createRESTModule(publicUpdateProduct);
export const deleteProduct = /*#__PURE__*/ createRESTModule(publicDeleteProduct);
export const updateProductVariants = /*#__PURE__*/ createRESTModule(publicUpdateProductVariants);
export const resetAllProductVariantData = /*#__PURE__*/ createRESTModule(publicResetAllProductVariantData);
export const addProductsToCollection = /*#__PURE__*/ createRESTModule(publicAddProductsToCollection);
export const removeProductsFromCollection = /*#__PURE__*/ createRESTModule(publicRemoveProductsFromCollection);
export const addProductMedia = /*#__PURE__*/ createRESTModule(publicAddProductMedia);
export const removeProductMedia = /*#__PURE__*/ createRESTModule(publicRemoveProductMedia);
export const addProductMediaToChoices = /*#__PURE__*/ createRESTModule(publicAddProductMediaToChoices);
export const removeProductMediaFromChoices = /*#__PURE__*/ createRESTModule(publicRemoveProductMediaFromChoices);
export const deleteProductOptions = /*#__PURE__*/ createRESTModule(publicDeleteProductOptions);
export const removeBrand = /*#__PURE__*/ createRESTModule(publicRemoveBrand);
export const createCollection = /*#__PURE__*/ createRESTModule(publicCreateCollection);
export const updateCollection = /*#__PURE__*/ createRESTModule(publicUpdateCollection);
export const deleteCollection = /*#__PURE__*/ createRESTModule(publicDeleteCollection);
export const removeRibbon = /*#__PURE__*/ createRESTModule(publicRemoveRibbon);
export const bulkUpdateProductsProperty = /*#__PURE__*/ createRESTModule(publicBulkUpdateProductsProperty);
export const bulkAdjustProductProperty = /*#__PURE__*/ createRESTModule(publicBulkAdjustProductProperty);
export const queryProducts = /*#__PURE__*/ createRESTModule(publicQueryProducts);
export const getProduct = /*#__PURE__*/ createRESTModule(publicGetProduct);
export const getCollectionBySlug = /*#__PURE__*/ createRESTModule(publicGetCollectionBySlug);
export const getProductOptionsAvailability = /*#__PURE__*/ createRESTModule(publicGetProductOptionsAvailability);
export const queryProductVariants = /*#__PURE__*/ createRESTModule(publicQueryProductVariants);
export const queryStoreVariants = /*#__PURE__*/ createRESTModule(publicQueryStoreVariants);
export const getStoreVariant = /*#__PURE__*/ createRESTModule(publicGetStoreVariant);
/**
 * Triggered when a collection is changed.
 */
export const onProductCollectionChanged = createEventModule(publicOnProductCollectionChanged);
/**
 * Triggered when a collection is created.
 */
export const onProductCollectionCreated = createEventModule(publicOnProductCollectionCreated);
/**
 * Triggered when a collection is deleted.
 */
export const onProductCollectionDeleted = createEventModule(publicOnProductCollectionDeleted);
/**
 * Triggered when a product is changed.
 */
export const onProductChanged = createEventModule(publicOnProductChanged);
/**
 * Triggered when a product is created.
 */
export const onProductCreated = createEventModule(publicOnProductCreated);
/**
 * Triggered when a product is deleted.
 */
export const onProductDeleted = createEventModule(publicOnProductDeleted);
/**
 * Triggered when a product variant is changed.
 */
export const onProductVariantsChanged = createEventModule(publicOnProductVariantsChanged);
export { ProductType, InventoryStatus, MeasurementUnit, MediaItemType, OptionType, DiscountType, FileType, Version, SortOrder, WebhookIdentityType, } from './stores-catalog-v1-product-products.universal.js';
