// @wix/auto_sdk_ecom_checkout-content/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  getCheckoutContent: () => getCheckoutContent4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-api-v1-checkout-content-provider-checkout-content.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-api-v1-checkout-content-provider-checkout-content.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/api/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-content";
function getCheckoutContent(payload) {
  function __getCheckoutContent({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.api.v1.checkout_content_provider",
      method: "GET",
      methodFqn: "wix.ecom.checkout_content_spi_host.v1.CheckoutContentSPIHostService.GetCheckoutContent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(
        {
          protoPath: "/v1/checkout-content/{appId}/{componentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCheckoutContent;
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.universal.ts
async function getCheckoutContent2(identifiers, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    appId: identifiers?.appId,
    componentId: identifiers?.componentId,
    languageCode: options?.languageCode
  });
  const reqOpts = getCheckoutContent(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          appId: "$[0].appId",
          componentId: "$[0].componentId",
          languageCode: "$[1].languageCode"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.public.ts
function getCheckoutContent3(httpClient) {
  return (identifiers, options) => getCheckoutContent2(
    identifiers,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getCheckoutContent4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCheckoutContent3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getCheckoutContent
});
