// @wix/auto_sdk_ecom_checkout-content/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getCheckoutContent: () => getCheckoutContent2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-api-v1-checkout-content-provider-checkout-content.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/api/v1/checkout-content",
        destPath: "/v1/checkout-content"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-content";
function getCheckoutContent(payload) {
  function __getCheckoutContent({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.api.v1.checkout_content_provider",
      method: "GET",
      methodFqn: "wix.ecom.checkout_content_spi_host.v1.CheckoutContentSPIHostService.GetCheckoutContent",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomCheckoutContentSpiHostV1CheckoutContentSpiHostServiceUrl(
        {
          protoPath: "/v1/checkout-content/{appId}/{componentId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCheckoutContent;
}

// src/ecom-api-v1-checkout-content-provider-checkout-content.meta.ts
function getCheckoutContent2() {
  const payload = { appId: ":appId", componentId: ":componentId" };
  const getRequestOptions = getCheckoutContent(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout-content/{appId}/{componentId}",
    pathParams: { appId: "appId", componentId: "componentId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getCheckoutContent
});
