// @wix/auto_sdk_ecom_back-in-stock-settings/build/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  getSettings: () => getSettings2,
  startCollectingRequests: () => startCollectingRequests2,
  stopCollectingRequests: () => stopCollectingRequests2
});
module.exports = __toCommonJS(index_typings_exports);

// src/ecom-v1-back-in-stock-settings-back-in-stock-settings.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-v1-back-in-stock-settings-back-in-stock-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/v1/back-in-stock-notification-requests",
        destPath: "/v1/back-in-stock-notification-requests"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/v1/back-in-stock-notification-requests",
        destPath: "/v1/back-in-stock-notification-requests"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_back-in-stock-settings";
function startCollectingRequests(payload) {
  function __startCollectingRequests({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "POST",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.StartCollectingRequests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings/start-collecting",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __startCollectingRequests;
}
function stopCollectingRequests(payload) {
  function __stopCollectingRequests({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "POST",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.StopCollectingRequests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings/stop-collecting",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __stopCollectingRequests;
}
function getSettings(payload) {
  function __getSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "PUT",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.GetSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getSettings;
}

// src/ecom-v1-back-in-stock-settings-back-in-stock-settings.universal.ts
async function startCollectingRequests2(appId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ appId });
  const reqOpts = startCollectingRequests(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { appId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["appId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function stopCollectingRequests2(appId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ appId });
  const reqOpts = stopCollectingRequests(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { appId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["appId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getSettings2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getSettings(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getSettings,
  startCollectingRequests,
  stopCollectingRequests
});
