// @wix/auto_sdk_ecom_back-in-stock-settings/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getSettings: () => getSettings2,
  startCollectingRequests: () => startCollectingRequests2,
  stopCollectingRequests: () => stopCollectingRequests2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-back-in-stock-settings-back-in-stock-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/back-in-stock-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/v1/back-in-stock-notification-requests",
        destPath: "/v1/back-in-stock-notification-requests"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/v1/back-in-stock-notification-requests",
        destPath: "/v1/back-in-stock-notification-requests"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_back-in-stock-settings";
function startCollectingRequests(payload) {
  function __startCollectingRequests({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "POST",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.StartCollectingRequests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings/start-collecting",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __startCollectingRequests;
}
function stopCollectingRequests(payload) {
  function __stopCollectingRequests({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "POST",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.StopCollectingRequests",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings/stop-collecting",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __stopCollectingRequests;
}
function getSettings(payload) {
  function __getSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.back_in_stock_settings",
      method: "PUT",
      methodFqn: "com.wixpress.ecom.back.in.stock.BackInStockSettingsService.GetSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEcomBackInStockBackInStockSettingsServiceUrl({
        protoPath: "/v1/back-in-stock-notification-requests/settings",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getSettings;
}

// src/ecom-v1-back-in-stock-settings-back-in-stock-settings.meta.ts
function startCollectingRequests2() {
  const payload = {};
  const getRequestOptions = startCollectingRequests(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/back-in-stock-notification-requests/settings/start-collecting",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function stopCollectingRequests2() {
  const payload = {};
  const getRequestOptions = stopCollectingRequests(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/back-in-stock-notification-requests/settings/stop-collecting",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getSettings2() {
  const payload = {};
  const getRequestOptions = getSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PUT",
    path: "/v1/back-in-stock-notification-requests/settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getSettings,
  startCollectingRequests,
  stopCollectingRequests
});
