// @wix/auto_sdk_stores_inventory/build/internal/cjs/src/stores-v2-inventory-inventory.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ReasonType = exports.onInventoryVariantsChanged = exports.onInventoryItemChanged = exports.incrementInventory = exports.decrementInventory = exports.updateInventoryVariants = exports.queryInventory = exports.getInventoryVariants = void 0;
const stores_v2_inventory_inventory_public_js_1 = require("./stores-v2-inventory-inventory.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const stores_v2_inventory_inventory_public_js_2 = require("./stores-v2-inventory-inventory.public.js");
const stores_v2_inventory_inventory_public_js_3 = require("./stores-v2-inventory-inventory.public.js");
exports.getInventoryVariants = (0, rest_modules_1.createRESTModule)(stores_v2_inventory_inventory_public_js_1.getInventoryVariants);
exports.queryInventory = (0, rest_modules_1.createRESTModule)(stores_v2_inventory_inventory_public_js_1.queryInventory);
exports.updateInventoryVariants = (0, rest_modules_1.createRESTModule)(stores_v2_inventory_inventory_public_js_1.updateInventoryVariants);
exports.decrementInventory = (0, rest_modules_1.createRESTModule)(stores_v2_inventory_inventory_public_js_1.decrementInventory);
exports.incrementInventory = (0, rest_modules_1.createRESTModule)(stores_v2_inventory_inventory_public_js_1.incrementInventory);
/**
 * Triggered when an inventory item is changed.
 */
exports.onInventoryItemChanged = (0, event_definition_modules_1.createEventModule)(stores_v2_inventory_inventory_public_js_2.onInventoryItemChanged);
/**
 * Triggered when inventory variants are changed.
 */
exports.onInventoryVariantsChanged = (0, event_definition_modules_1.createEventModule)(stores_v2_inventory_inventory_public_js_3.onInventoryVariantsChanged);
var stores_v2_inventory_inventory_universal_js_1 = require("./stores-v2-inventory-inventory.universal.js");
Object.defineProperty(exports, "ReasonType", { enumerable: true, get: function () { return stores_v2_inventory_inventory_universal_js_1.ReasonType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_v2_inventory_inventory_universal_js_1.WebhookIdentityType; } });
