// @wix/auto_sdk_ecom_abandoned-checkouts/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ActivityType: () => ActivityType,
  CommonSortOrder: () => CommonSortOrder,
  DescriptionLineType: () => DescriptionLineType,
  DiscountType: () => DiscountType,
  Identity: () => Identity,
  Mode: () => Mode,
  SortOrder: () => SortOrder,
  Status: () => Status,
  VatType: () => VatType,
  WebhookIdentityType: () => WebhookIdentityType,
  addAbandonedCheckoutActivity: () => addAbandonedCheckoutActivity4,
  bulkDeleteAbandonedCheckouts: () => bulkDeleteAbandonedCheckouts4,
  deleteAbandonedCheckout: () => deleteAbandonedCheckout4,
  getAbandonedCheckout: () => getAbandonedCheckout4,
  onAbandonedCheckoutCreated: () => onAbandonedCheckoutCreated2,
  onAbandonedCheckoutDeleted: () => onAbandonedCheckoutDeleted2,
  onAbandonedCheckoutRecovered: () => onAbandonedCheckoutRecovered2,
  onAbandonedCheckoutUpdated: () => onAbandonedCheckoutUpdated2,
  queryAbandonedCheckouts: () => queryAbandonedCheckouts4,
  redirectToCheckout: () => redirectToCheckout4,
  searchAbandonedCheckouts: () => searchAbandonedCheckouts4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_bytes = require("@wix/sdk-runtime/transformations/bytes");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/abandoned-checkout",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_abandoned-checkouts";
function getAbandonedCheckout(payload) {
  function __getAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.GetAbandonedCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "abandonedCheckout.lineItems.physicalProperties.weight" },
            {
              path: "abandonedCheckout.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAbandonedCheckout;
}
function deleteAbandonedCheckout(payload) {
  function __deleteAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "DELETE",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.DeleteAbandonedCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteAbandonedCheckout;
}
function bulkDeleteAbandonedCheckouts(payload) {
  function __bulkDeleteAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.BulkDeleteAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/bulk/abandoned-checkout/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteAbandonedCheckouts;
}
function queryAbandonedCheckouts(payload) {
  function __queryAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.QueryAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.createdDate" },
            { path: "results.updatedDate" },
            { path: "results.checkoutRecoveredDate" },
            { path: "results.activities.createdDate" },
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.lineItems.physicalProperties.weight" },
            { path: "results.appliedDiscounts.merchantDiscount.percentage" },
            {
              path: "abandonedCheckouts.lineItems.physicalProperties.weight"
            },
            {
              path: "abandonedCheckouts.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAbandonedCheckouts;
}
function searchAbandonedCheckouts(payload) {
  function __searchAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.SearchAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "abandonedCheckouts.lineItems.physicalProperties.weight"
            },
            {
              path: "abandonedCheckouts.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchAbandonedCheckouts;
}
function addAbandonedCheckoutActivity(payload) {
  function __addAbandonedCheckoutActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.AddAbandonedCheckoutActivity",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "abandonedCheckout.lineItems.physicalProperties.weight" },
            {
              path: "abandonedCheckout.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addAbandonedCheckoutActivity;
}
function redirectToCheckout(payload) {
  function __redirectToCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.RedirectToCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "body" }]
        }
      ])
    };
    return metadata;
  }
  return __redirectToCheckout;
}

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var ActivityType = /* @__PURE__ */ ((ActivityType2) => {
  ActivityType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ActivityType2["SCHEDULED"] = "SCHEDULED";
  ActivityType2["EMAIL_SENT"] = "EMAIL_SENT";
  ActivityType2["EMAIL_NOT_SENT"] = "EMAIL_NOT_SENT";
  ActivityType2["NOTIFICATION_SENT"] = "NOTIFICATION_SENT";
  ActivityType2["TASK_CREATED"] = "TASK_CREATED";
  ActivityType2["AUTOMATION_SKIPPED"] = "AUTOMATION_SKIPPED";
  return ActivityType2;
})(ActivityType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ABANDONED"] = "ABANDONED";
  Status2["RECOVERED"] = "RECOVERED";
  return Status2;
})(Status || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var DescriptionLineType = /* @__PURE__ */ ((DescriptionLineType2) => {
  DescriptionLineType2["UNRECOGNISED"] = "UNRECOGNISED";
  DescriptionLineType2["PLAIN_TEXT"] = "PLAIN_TEXT";
  DescriptionLineType2["COLOR"] = "COLOR";
  return DescriptionLineType2;
})(DescriptionLineType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var Identity = /* @__PURE__ */ ((Identity2) => {
  Identity2["ADMIN"] = "ADMIN";
  Identity2["MEMBER"] = "MEMBER";
  Identity2["VISITOR"] = "VISITOR";
  Identity2["CONTACT"] = "CONTACT";
  return Identity2;
})(Identity || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var CommonSortOrder = /* @__PURE__ */ ((CommonSortOrder2) => {
  CommonSortOrder2["ASC"] = "ASC";
  CommonSortOrder2["DESC"] = "DESC";
  return CommonSortOrder2;
})(CommonSortOrder || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getAbandonedCheckout2(abandonedCheckoutId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    abandonedCheckoutId
  });
  const reqOpts = getAbandonedCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [{ path: "abandonedCheckout.lineItems.image" }]
        }
      ])
    )?.abandonedCheckout;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { abandonedCheckoutId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteAbandonedCheckout2(abandonedCheckoutId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    abandonedCheckoutId
  });
  const reqOpts = deleteAbandonedCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { abandonedCheckoutId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteAbandonedCheckouts2(abandonedCheckoutIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    abandonedCheckoutIds
  });
  const reqOpts = bulkDeleteAbandonedCheckouts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { abandonedCheckoutIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryAbandonedCheckouts2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryAbandonedCheckouts(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image.transformRESTImageToSDKImage,
            paths: [
              { path: "results.lineItems.image" },
              { path: "abandonedCheckouts.lineItems.image" }
            ]
          }
        ])
      );
      return {
        items: transformedData?.abandonedCheckouts,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function typedQueryAbandonedCheckouts(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ query });
  const reqOpts = queryAbandonedCheckouts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [
            { path: "results.lineItems.image" },
            { path: "abandonedCheckouts.lineItems.image" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchAbandonedCheckouts2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ search });
  const reqOpts = searchAbandonedCheckouts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [{ path: "abandonedCheckouts.lineItems.image" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addAbandonedCheckoutActivity2(abandonedCheckoutId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    abandonedCheckoutId,
    activityType: options?.activityType
  });
  const reqOpts = addAbandonedCheckoutActivity(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [{ path: "abandonedCheckout.lineItems.image" }]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          abandonedCheckoutId: "$[0]",
          activityType: "$[1].activityType"
        },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function redirectToCheckout2(abandonedCheckoutId, metasiteId) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    abandonedCheckoutId,
    metasiteId
  });
  const reqOpts = redirectToCheckout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          abandonedCheckoutId: "$[0]",
          metasiteId: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["abandonedCheckoutId", "metasiteId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.public.ts
function getAbandonedCheckout3(httpClient) {
  return (abandonedCheckoutId) => getAbandonedCheckout2(
    abandonedCheckoutId,
    // @ts-ignore
    { httpClient }
  );
}
function deleteAbandonedCheckout3(httpClient) {
  return (abandonedCheckoutId) => deleteAbandonedCheckout2(
    abandonedCheckoutId,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteAbandonedCheckouts3(httpClient) {
  return (abandonedCheckoutIds) => bulkDeleteAbandonedCheckouts2(
    abandonedCheckoutIds,
    // @ts-ignore
    { httpClient }
  );
}
function queryAbandonedCheckouts3(httpClient) {
  return () => queryAbandonedCheckouts2(
    // @ts-ignore
    { httpClient }
  );
}
function typedQueryAbandonedCheckouts2(httpClient) {
  return (query) => typedQueryAbandonedCheckouts(
    query,
    // @ts-ignore
    { httpClient }
  );
}
function searchAbandonedCheckouts3(httpClient) {
  return (search) => searchAbandonedCheckouts2(
    search,
    // @ts-ignore
    { httpClient }
  );
}
function addAbandonedCheckoutActivity3(httpClient) {
  return (abandonedCheckoutId, options) => addAbandonedCheckoutActivity2(
    abandonedCheckoutId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function redirectToCheckout3(httpClient) {
  return (abandonedCheckoutId, metasiteId) => redirectToCheckout2(
    abandonedCheckoutId,
    metasiteId,
    // @ts-ignore
    { httpClient }
  );
}
var onAbandonedCheckoutCreated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.abandoned_checkout_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.checkoutRecoveredDate" },
          { path: "entity.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "entity.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.lineItems.physicalProperties.weight" },
          { path: "entity.appliedDiscounts.merchantDiscount.percentage" }
        ]
      }
    ])
  )
)();
var onAbandonedCheckoutDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.abandoned_checkout_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.checkoutRecoveredDate" },
          { path: "undefined.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "undefined.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [
          { path: "undefined.lineItems.physicalProperties.weight" },
          { path: "undefined.appliedDiscounts.merchantDiscount.percentage" }
        ]
      }
    ])
  )
)();
var onAbandonedCheckoutRecovered = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.abandoned_checkout_recovered",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.abandonedCheckout.createdDate" },
          { path: "data.abandonedCheckout.updatedDate" },
          { path: "data.abandonedCheckout.checkoutRecoveredDate" },
          { path: "data.abandonedCheckout.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "data.abandonedCheckout.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "data.abandonedCheckout.lineItems.physicalProperties.weight"
          },
          {
            path: "data.abandonedCheckout.appliedDiscounts.merchantDiscount.percentage"
          }
        ]
      }
    ])
  )
)();
var onAbandonedCheckoutUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.abandoned_checkout_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.checkoutRecoveredDate" },
          { path: "entity.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "entity.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.lineItems.physicalProperties.weight" },
          { path: "entity.appliedDiscounts.merchantDiscount.percentage" }
        ]
      }
    ])
  )
)();

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var import_query_method_router = require("@wix/sdk-runtime/query-method-router");
function customQueryAbandonedCheckouts(httpClient) {
  const router = (0, import_query_method_router.createQueryOverloadRouter)({
    builderQueryFunction: () => queryAbandonedCheckouts3(httpClient)(),
    typedQueryFunction: (query) => typedQueryAbandonedCheckouts2(httpClient)(query),
    hasOptionsParameter: false
  });
  function overloadedQuery(query) {
    return router(...arguments);
  }
  return overloadedQuery;
}
var getAbandonedCheckout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getAbandonedCheckout3);
var deleteAbandonedCheckout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteAbandonedCheckout3);
var bulkDeleteAbandonedCheckouts4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteAbandonedCheckouts3);
var searchAbandonedCheckouts4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(searchAbandonedCheckouts3);
var addAbandonedCheckoutActivity4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(addAbandonedCheckoutActivity3);
var redirectToCheckout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(redirectToCheckout3);
var queryAbandonedCheckouts4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(customQueryAbandonedCheckouts);
var onAbandonedCheckoutCreated2 = (0, import_event_definition_modules.createEventModule)(
  onAbandonedCheckoutCreated
);
var onAbandonedCheckoutDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onAbandonedCheckoutDeleted
);
var onAbandonedCheckoutRecovered2 = (0, import_event_definition_modules.createEventModule)(
  onAbandonedCheckoutRecovered
);
var onAbandonedCheckoutUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onAbandonedCheckoutUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActivityType,
  CommonSortOrder,
  DescriptionLineType,
  DiscountType,
  Identity,
  Mode,
  SortOrder,
  Status,
  VatType,
  WebhookIdentityType,
  addAbandonedCheckoutActivity,
  bulkDeleteAbandonedCheckouts,
  deleteAbandonedCheckout,
  getAbandonedCheckout,
  onAbandonedCheckoutCreated,
  onAbandonedCheckoutDeleted,
  onAbandonedCheckoutRecovered,
  onAbandonedCheckoutUpdated,
  queryAbandonedCheckouts,
  redirectToCheckout,
  searchAbandonedCheckouts
});
