// @wix/auto_sdk_ecom_abandoned-checkouts/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  ActivityTypeOriginal: () => ActivityType,
  CommonSortOrderOriginal: () => CommonSortOrder,
  DescriptionLineTypeOriginal: () => DescriptionLineType,
  DiscountTypeOriginal: () => DiscountType,
  IdentityOriginal: () => Identity,
  ModeOriginal: () => Mode,
  SortOrderOriginal: () => SortOrder,
  StatusOriginal: () => Status,
  VatTypeOriginal: () => VatType,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  addAbandonedCheckoutActivity: () => addAbandonedCheckoutActivity2,
  bulkDeleteAbandonedCheckouts: () => bulkDeleteAbandonedCheckouts2,
  deleteAbandonedCheckout: () => deleteAbandonedCheckout2,
  getAbandonedCheckout: () => getAbandonedCheckout2,
  queryAbandonedCheckouts: () => queryAbandonedCheckouts2,
  redirectToCheckout: () => redirectToCheckout2,
  searchAbandonedCheckouts: () => searchAbandonedCheckouts2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_bytes = require("@wix/sdk-runtime/transformations/bytes");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/abandoned-checkout",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/abandoned-checkout",
        destPath: "/v1/abandoned-checkout"
      },
      {
        srcPath: "/ecom/v1/bulk/abandoned-checkout",
        destPath: "/v1/bulk/abandoned-checkout"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_abandoned-checkouts";
function getAbandonedCheckout(payload) {
  function __getAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.GetAbandonedCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "abandonedCheckout.lineItems.physicalProperties.weight" },
            {
              path: "abandonedCheckout.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getAbandonedCheckout;
}
function deleteAbandonedCheckout(payload) {
  function __deleteAbandonedCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "DELETE",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.DeleteAbandonedCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteAbandonedCheckout;
}
function bulkDeleteAbandonedCheckouts(payload) {
  function __bulkDeleteAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.BulkDeleteAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/bulk/abandoned-checkout/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteAbandonedCheckouts;
}
function queryAbandonedCheckouts(payload) {
  function __queryAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.QueryAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.createdDate" },
            { path: "results.updatedDate" },
            { path: "results.checkoutRecoveredDate" },
            { path: "results.activities.createdDate" },
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.lineItems.physicalProperties.weight" },
            { path: "results.appliedDiscounts.merchantDiscount.percentage" },
            {
              path: "abandonedCheckouts.lineItems.physicalProperties.weight"
            },
            {
              path: "abandonedCheckouts.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryAbandonedCheckouts;
}
function searchAbandonedCheckouts(payload) {
  function __searchAbandonedCheckouts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.SearchAbandonedCheckouts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckouts.createdDate" },
            { path: "abandonedCheckouts.updatedDate" },
            { path: "abandonedCheckouts.checkoutRecoveredDate" },
            { path: "abandonedCheckouts.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "abandonedCheckouts.lineItems.physicalProperties.weight"
            },
            {
              path: "abandonedCheckouts.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchAbandonedCheckouts;
}
function addAbandonedCheckoutActivity(payload) {
  function __addAbandonedCheckoutActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "POST",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.AddAbandonedCheckoutActivity",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "abandonedCheckout.createdDate" },
            { path: "abandonedCheckout.updatedDate" },
            { path: "abandonedCheckout.checkoutRecoveredDate" },
            { path: "abandonedCheckout.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "abandonedCheckout.lineItems.physicalProperties.weight" },
            {
              path: "abandonedCheckout.appliedDiscounts.merchantDiscount.percentage"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addAbandonedCheckoutActivity;
}
function redirectToCheckout(payload) {
  function __redirectToCheckout({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.abandoned_checkout",
      method: "GET",
      methodFqn: "wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.RedirectToCheckout",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
        protoPath: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "body" }]
        }
      ])
    };
    return metadata;
  }
  return __redirectToCheckout;
}

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.types.ts
var ActivityType = /* @__PURE__ */ ((ActivityType2) => {
  ActivityType2["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
  ActivityType2["SCHEDULED"] = "SCHEDULED";
  ActivityType2["EMAIL_SENT"] = "EMAIL_SENT";
  ActivityType2["EMAIL_NOT_SENT"] = "EMAIL_NOT_SENT";
  ActivityType2["NOTIFICATION_SENT"] = "NOTIFICATION_SENT";
  ActivityType2["TASK_CREATED"] = "TASK_CREATED";
  ActivityType2["AUTOMATION_SKIPPED"] = "AUTOMATION_SKIPPED";
  return ActivityType2;
})(ActivityType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["ABANDONED"] = "ABANDONED";
  Status2["RECOVERED"] = "RECOVERED";
  return Status2;
})(Status || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var DescriptionLineType = /* @__PURE__ */ ((DescriptionLineType2) => {
  DescriptionLineType2["UNRECOGNISED"] = "UNRECOGNISED";
  DescriptionLineType2["PLAIN_TEXT"] = "PLAIN_TEXT";
  DescriptionLineType2["COLOR"] = "COLOR";
  return DescriptionLineType2;
})(DescriptionLineType || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var Identity = /* @__PURE__ */ ((Identity2) => {
  Identity2["ADMIN"] = "ADMIN";
  Identity2["MEMBER"] = "MEMBER";
  Identity2["VISITOR"] = "VISITOR";
  Identity2["CONTACT"] = "CONTACT";
  return Identity2;
})(Identity || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var CommonSortOrder = /* @__PURE__ */ ((CommonSortOrder2) => {
  CommonSortOrder2["ASC"] = "ASC";
  CommonSortOrder2["DESC"] = "DESC";
  return CommonSortOrder2;
})(CommonSortOrder || {});
var Mode = /* @__PURE__ */ ((Mode2) => {
  Mode2["OR"] = "OR";
  Mode2["AND"] = "AND";
  return Mode2;
})(Mode || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/ecom-v1-abandoned-checkout-abandoned-checkouts.meta.ts
function getAbandonedCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = getAbandonedCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteAbandonedCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = deleteAbandonedCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteAbandonedCheckouts2() {
  const payload = {};
  const getRequestOptions = bulkDeleteAbandonedCheckouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/abandoned-checkout/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryAbandonedCheckouts2() {
  const payload = {};
  const getRequestOptions = queryAbandonedCheckouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkout/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function searchAbandonedCheckouts2() {
  const payload = {};
  const getRequestOptions = searchAbandonedCheckouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkout/search",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addAbandonedCheckoutActivity2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = addAbandonedCheckoutActivity(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/abandoned-checkouts/{abandonedCheckoutId}/add-activity",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function redirectToCheckout2() {
  const payload = { abandonedCheckoutId: ":abandonedCheckoutId" };
  const getRequestOptions = redirectToCheckout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout",
    pathParams: { abandonedCheckoutId: "abandonedCheckoutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActivityTypeOriginal,
  CommonSortOrderOriginal,
  DescriptionLineTypeOriginal,
  DiscountTypeOriginal,
  IdentityOriginal,
  ModeOriginal,
  SortOrderOriginal,
  StatusOriginal,
  VatTypeOriginal,
  WebhookIdentityTypeOriginal,
  addAbandonedCheckoutActivity,
  bulkDeleteAbandonedCheckouts,
  deleteAbandonedCheckout,
  getAbandonedCheckout,
  queryAbandonedCheckouts,
  redirectToCheckout,
  searchAbandonedCheckouts
});
