// @wix/ecom_app-extensions/build/cjs/src/extensions/dashboard-page-builders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderDetails = exports.editDraftOrder = exports.newOrder = exports.orderList = exports.deliveryProfile = exports.deliveryProfiles = exports.orderRefund = exports.editOrder = void 0;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Edit Order page in the dashboard.
 * @dashobardPageBuilder
 */
function editOrder(options) {
    const { orderId } = options;
    return {
        pageId: 'f2526550-194f-4c13-9298-9a6b8abda62f',
        relativeUrl: `${orderId}`,
    };
}
exports.editOrder = editOrder;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Order Refund page in the dashboard.
 * @dashobardPageBuilder
 */
function orderRefund(options) {
    const { orderId } = options;
    return {
        pageId: 'c4898867-69a5-4e05-92b7-c276f9424641',
        relativeUrl: `${orderId}`,
    };
}
exports.orderRefund = orderRefund;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Delivery Profiles page in the dashboard.
 * @dashobardPageBuilder
 */
function deliveryProfiles() {
    return { pageId: '841919b4-9e96-43f4-87ea-fa382bc9d0e8' };
}
exports.deliveryProfiles = deliveryProfiles;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Delivery Profile page in the dashboard.
 * @dashobardPageBuilder
 */
function deliveryProfile(options) {
    const { deliveryProfileId } = options;
    return {
        pageId: '841919b4-9e96-43f4-87ea-fa382bc9d0e8',
        relativeUrl: `${deliveryProfileId}`,
    };
}
exports.deliveryProfile = deliveryProfile;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Order List page in the dashboard.
 * @dashobardPageBuilder
 */
function orderList() {
    return { pageId: '8107f05f-d646-4c81-be90-adf28d321398' };
}
exports.orderList = orderList;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the New Order page in the dashboard.
 * @dashobardPageBuilder
 */
function newOrder() {
    return { pageId: '8454e8a0-5684-4331-9d4a-e4873467553a' };
}
exports.newOrder = newOrder;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Edit Draft Order page in the dashboard.
 * @dashobardPageBuilder
 */
function editDraftOrder(options) {
    const { draftOrderId } = options;
    return {
        pageId: '8454e8a0-5684-4331-9d4a-e4873467553a',
        relativeUrl: `${draftOrderId}`,
    };
}
exports.editDraftOrder = editDraftOrder;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Order Details page in the dashboard.
 * @dashobardPageBuilder
 */
function orderDetails(options) {
    const { id } = options;
    return {
        pageId: '3b96985e-a459-4972-94c3-63f038a73af2',
        relativeUrl: `${id}`,
    };
}
exports.orderDetails = orderDetails;
