// @wix/auto_sdk_ecom_discounts-custom-trigger/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  listTriggers: () => listTriggers4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/custom-triggers-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      },
      {
        srcPath: "/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/custom-triggers-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/custom-triggers-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_discounts-custom-trigger";
function listTriggers(payload) {
  function __listTriggers({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.custom_trigger.v1.custom_trigger",
      method: "POST",
      methodFqn: "com.wix.ecom.custom_trigger.v1.CustomTriggersService.ListTriggers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl({
        protoPath: "/v1/discount-custom-triggers/list",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listTriggers;
}

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.universal.ts
async function listTriggers2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listTriggers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.public.ts
function listTriggers3(httpClient) {
  return () => listTriggers2(
    // @ts-ignore
    { httpClient }
  );
}

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.context.ts
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
var listTriggers4 = /* @__PURE__ */ (0, import_rest_modules2.createRESTModule)(listTriggers3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listTriggers
});
