// @wix/auto_sdk_ecom_discounts-custom-trigger/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  listTriggers: () => listTriggers2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/custom-triggers-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      },
      {
        srcPath: "/v1/discount-custom-triggers",
        destPath: "/v1/discount-custom-triggers"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/custom-triggers-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/custom-triggers-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_discounts-custom-trigger";
function listTriggers(payload) {
  function __listTriggers({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.custom_trigger.v1.custom_trigger",
      method: "POST",
      methodFqn: "com.wix.ecom.custom_trigger.v1.CustomTriggersService.ListTriggers",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl({
        protoPath: "/v1/discount-custom-triggers/list",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listTriggers;
}

// src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.meta.ts
function listTriggers2() {
  const payload = {};
  const getRequestOptions = listTriggers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/discount-custom-triggers/list",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listTriggers
});
