// @wix/auto_sdk_stores_inventory-items-v-3/build/es/src/stores-catalog-v3-inventory-item-inventory-items-v-3.meta.js
import * as ambassadorWixStoresCatalogV3InventoryItem from './stores-catalog-v3-inventory-item-inventory-items-v-3.http.js';
export function createInventoryItem() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.createInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkCreateInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getInventoryItem() {
    const payload = { inventoryItemId: ':inventoryItemId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.getInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/inventory-items/{inventoryItemId}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateInventoryItem() {
    const payload = { inventoryItem: { id: ':inventoryItemId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.updateInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/inventory-items/{inventoryItem.id}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkUpdateInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateInventoryItemsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkUpdateInventoryItemsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteInventoryItem() {
    const payload = { inventoryItemId: ':inventoryItemId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.deleteInventoryItem(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/inventory-items/{inventoryItemId}',
        pathParams: { inventoryItemId: 'inventoryItemId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDeleteInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.queryInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.searchInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/inventory-items/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDecrementInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDecrementInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/decrement',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkIncrementInventoryItems() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkIncrementInventoryItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/increment',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDecrementInventoryItemsByVariantAndLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkDecrementInventoryItemsByVariantAndLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/decrement-by-variant-and-location',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkIncrementInventoryItemsByVariantAndLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkIncrementInventoryItemsByVariantAndLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/increment-by-variant-and-location',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkSetOrAdjustInventoryItemsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InventoryItem.bulkSetOrAdjustInventoryItemsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/inventory-items/set-or-adjust-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
