// @wix/auto_sdk_stores_catalog-versioning/build/internal/cjs/src/stores-catalog-v3-provision-catalog-versioning.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCatalogVersion = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl(opts) {
    const domainToMappings = {
        'www.wixapis.com': [
            {
                srcPath: '/stores/v3/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/stores/v3/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/provision/version',
                destPath: '/v3/provision/version',
            },
            {
                srcPath: '/v3/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
        _: [
            {
                srcPath: '/_api/stores/v3/provision/version',
                destPath: '/v3/provision/version',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_catalog-versioning';
/** Retrieves the version of Stores Catalog installed on a site. */
function getCatalogVersion(payload) {
    function __getCatalogVersion({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.provision',
            method: 'GET',
            methodFqn: 'wix.stores.catalog.provision.v1.StoresProvisionService.GetCatalogVersion',
            packageName: PACKAGE_NAME,
            url: resolveWixStoresCatalogProvisionV1StoresProvisionServiceUrl({
                protoPath: '/v3/provision/version',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getCatalogVersion;
}
exports.getCatalogVersion = getCatalogVersion;
