// @wix/auto_sdk_stores_collections/build/internal/es/src/stores-catalog-v1-collection-collections.meta.js
import * as ambassadorWixStoresCatalogV1Collection from './stores-catalog-v1-collection-collections.http.js';
export function queryCollections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV1Collection.queryCollections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCollection() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Collection.getCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCollectionBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixStoresCatalogV1Collection.getCollectionBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections/slug/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
