// @wix/auto_sdk_stores_subscription-options/build/es/src/stores-v1-subscription-option-subscription-options.meta.js
import * as ambassadorWixStoresV1SubscriptionOption from './stores-v1-subscription-option-subscription-options.http.js';
export function createSubscriptionOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.createSubscriptionOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/subscription-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateSubscriptionOption() {
    const payload = {
        subscriptionOption: { id: ':subscriptionOptionId' },
    };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.updateSubscriptionOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/subscription-options/{subscriptionOption.id}',
        pathParams: { subscriptionOptionId: 'subscriptionOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteSubscriptionOption() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.deleteSubscriptionOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/subscription-options/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateSubscriptionOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.bulkCreateSubscriptionOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/subscription-options/createBulk',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateSubscriptionOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.bulkUpdateSubscriptionOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/subscription-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteSubscriptionOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.bulkDeleteSubscriptionOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/subscription-options/deleteBulk',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function assignSubscriptionOptionsToProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.assignSubscriptionOptionsToProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/subscription-options/product/{productId}/assign',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function allowOneTimePurchases() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.allowOneTimePurchases(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/subscription-options/product/{productId}/allowOneTimePurchase',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getSubscriptionOption() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.getSubscriptionOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/subscription-options/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getSubscriptionOptionsForProduct() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.getSubscriptionOptionsForProduct(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/subscription-options/byProduct/{productId}',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getProductIdsForSubscriptionOption() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.getProductIdsForSubscriptionOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/subscription-options/{id}/productIds',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOneTimePurchasesStatus() {
    const payload = { productId: ':productId' };
    const getRequestOptions = ambassadorWixStoresV1SubscriptionOption.getOneTimePurchasesStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/subscription-options/product/{productId}/oneTimePurchasesStatus',
        pathParams: { productId: 'productId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
