// @wix/auto_sdk_ecom_payment-settings/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ActivityType: () => ActivityType,
  AdditionalFeeAdditionalFeeSource: () => AdditionalFeeAdditionalFeeSource,
  AdditionalFeeSource: () => AdditionalFeeSource,
  AdjustmentType: () => AdjustmentType,
  AppliedDiscountDiscountType: () => AppliedDiscountDiscountType,
  AttributionSource: () => AttributionSource,
  ChannelType: () => ChannelType,
  ChargeType: () => ChargeType,
  DescriptionLineType: () => DescriptionLineType,
  DiscountReason: () => DiscountReason,
  DiscountType: () => DiscountType,
  FallbackReason: () => FallbackReason,
  FileType: () => FileType,
  FulfillmentStatus: () => FulfillmentStatus,
  IdentityType: () => IdentityType,
  ItemAvailabilityStatus: () => ItemAvailabilityStatus,
  ItemTypePreset: () => ItemTypePreset,
  JurisdictionType: () => JurisdictionType,
  LineItemQuantityChangeType: () => LineItemQuantityChangeType,
  ManualCalculationReason: () => ManualCalculationReason,
  NameInLineItem: () => NameInLineItem,
  NameInOther: () => NameInOther,
  OrderActionType: () => OrderActionType,
  OrderActivityTypeEnumActivityType: () => OrderActivityTypeEnumActivityType,
  OrderStatus: () => OrderStatus,
  PaymentOptionType: () => PaymentOptionType,
  PaymentStatus: () => PaymentStatus,
  PickupDetailsPickupMethod: () => PickupDetailsPickupMethod,
  PickupMethod: () => PickupMethod,
  RateType: () => RateType,
  RuleType: () => RuleType,
  Severity: () => Severity,
  SubscriptionFrequency: () => SubscriptionFrequency,
  SuggestedFix: () => SuggestedFix,
  TaxableAddressType: () => TaxableAddressType,
  V1ChargeType: () => V1ChargeType,
  V1JurisdictionType: () => V1JurisdictionType,
  V1PaymentOption: () => V1PaymentOption,
  VatType: () => VatType,
  WeightUnit: () => WeightUnit,
  provideHandlers: () => provideHandlers2,
  publicProvideHandlers: () => provideHandlers
});
module.exports = __toCommonJS(index_exports);

// src/service-plugins-types.ts
var DescriptionLineType = /* @__PURE__ */ ((DescriptionLineType2) => {
  DescriptionLineType2["UNRECOGNISED"] = "UNRECOGNISED";
  DescriptionLineType2["PLAIN_TEXT"] = "PLAIN_TEXT";
  DescriptionLineType2["COLOR"] = "COLOR";
  return DescriptionLineType2;
})(DescriptionLineType || {});
var ItemTypePreset = /* @__PURE__ */ ((ItemTypePreset2) => {
  ItemTypePreset2["UNRECOGNISED"] = "UNRECOGNISED";
  ItemTypePreset2["PHYSICAL"] = "PHYSICAL";
  ItemTypePreset2["DIGITAL"] = "DIGITAL";
  ItemTypePreset2["GIFT_CARD"] = "GIFT_CARD";
  ItemTypePreset2["SERVICE"] = "SERVICE";
  return ItemTypePreset2;
})(ItemTypePreset || {});
var PaymentOptionType = /* @__PURE__ */ ((PaymentOptionType2) => {
  PaymentOptionType2["FULL_PAYMENT_ONLINE"] = "FULL_PAYMENT_ONLINE";
  PaymentOptionType2["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
  PaymentOptionType2["MEMBERSHIP"] = "MEMBERSHIP";
  PaymentOptionType2["DEPOSIT_ONLINE"] = "DEPOSIT_ONLINE";
  PaymentOptionType2["MEMBERSHIP_OFFLINE"] = "MEMBERSHIP_OFFLINE";
  PaymentOptionType2["MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER"] = "MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER";
  return PaymentOptionType2;
})(PaymentOptionType || {});
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});
var SubscriptionFrequency = /* @__PURE__ */ ((SubscriptionFrequency2) => {
  SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  SubscriptionFrequency2["DAY"] = "DAY";
  SubscriptionFrequency2["WEEK"] = "WEEK";
  SubscriptionFrequency2["MONTH"] = "MONTH";
  SubscriptionFrequency2["YEAR"] = "YEAR";
  return SubscriptionFrequency2;
})(SubscriptionFrequency || {});
var AdjustmentType = /* @__PURE__ */ ((AdjustmentType2) => {
  AdjustmentType2["UNKNOWN_ADJUSTMENT_TYPE"] = "UNKNOWN_ADJUSTMENT_TYPE";
  AdjustmentType2["EXTRA_CHARGE"] = "EXTRA_CHARGE";
  AdjustmentType2["CREDIT"] = "CREDIT";
  return AdjustmentType2;
})(AdjustmentType || {});
var TaxableAddressType = /* @__PURE__ */ ((TaxableAddressType2) => {
  TaxableAddressType2["UNKNOWN_TAXABLE_ADDRESS"] = "UNKNOWN_TAXABLE_ADDRESS";
  TaxableAddressType2["BUSINESS"] = "BUSINESS";
  TaxableAddressType2["BILLING"] = "BILLING";
  TaxableAddressType2["SHIPPING"] = "SHIPPING";
  return TaxableAddressType2;
})(TaxableAddressType || {});
var PaymentStatus = /* @__PURE__ */ ((PaymentStatus2) => {
  PaymentStatus2["UNSPECIFIED"] = "UNSPECIFIED";
  PaymentStatus2["NOT_PAID"] = "NOT_PAID";
  PaymentStatus2["PAID"] = "PAID";
  PaymentStatus2["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
  PaymentStatus2["FULLY_REFUNDED"] = "FULLY_REFUNDED";
  PaymentStatus2["PENDING"] = "PENDING";
  PaymentStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  PaymentStatus2["PENDING_MERCHANT"] = "PENDING_MERCHANT";
  PaymentStatus2["CANCELED"] = "CANCELED";
  PaymentStatus2["DECLINED"] = "DECLINED";
  return PaymentStatus2;
})(PaymentStatus || {});
var FulfillmentStatus = /* @__PURE__ */ ((FulfillmentStatus2) => {
  FulfillmentStatus2["NOT_FULFILLED"] = "NOT_FULFILLED";
  FulfillmentStatus2["FULFILLED"] = "FULFILLED";
  FulfillmentStatus2["PARTIALLY_FULFILLED"] = "PARTIALLY_FULFILLED";
  return FulfillmentStatus2;
})(FulfillmentStatus || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["INITIALIZED"] = "INITIALIZED";
  OrderStatus2["APPROVED"] = "APPROVED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["REJECTED"] = "REJECTED";
  return OrderStatus2;
})(OrderStatus || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var DiscountReason = /* @__PURE__ */ ((DiscountReason2) => {
  DiscountReason2["UNSPECIFIED"] = "UNSPECIFIED";
  DiscountReason2["EXCHANGED_ITEMS"] = "EXCHANGED_ITEMS";
  DiscountReason2["BILLING_ADJUSTMENT"] = "BILLING_ADJUSTMENT";
  return DiscountReason2;
})(DiscountReason || {});
var LineItemQuantityChangeType = /* @__PURE__ */ ((LineItemQuantityChangeType2) => {
  LineItemQuantityChangeType2["QUANTITY_INCREASED"] = "QUANTITY_INCREASED";
  LineItemQuantityChangeType2["QUANTITY_DECREASED"] = "QUANTITY_DECREASED";
  return LineItemQuantityChangeType2;
})(LineItemQuantityChangeType || {});
var ActivityType = /* @__PURE__ */ ((ActivityType2) => {
  ActivityType2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  ActivityType2["ORDER_PLACED"] = "ORDER_PLACED";
  ActivityType2["ORDER_PAID"] = "ORDER_PAID";
  ActivityType2["ORDER_FULFILLED"] = "ORDER_FULFILLED";
  ActivityType2["ORDER_NOT_FULFILLED"] = "ORDER_NOT_FULFILLED";
  ActivityType2["ORDER_CANCELED"] = "ORDER_CANCELED";
  ActivityType2["DOWNLOAD_LINK_SENT"] = "DOWNLOAD_LINK_SENT";
  ActivityType2["TRACKING_NUMBER_ADDED"] = "TRACKING_NUMBER_ADDED";
  ActivityType2["TRACKING_NUMBER_EDITED"] = "TRACKING_NUMBER_EDITED";
  ActivityType2["TRACKING_LINK_ADDED"] = "TRACKING_LINK_ADDED";
  ActivityType2["SHIPPING_CONFIRMATION_EMAIL_SENT"] = "SHIPPING_CONFIRMATION_EMAIL_SENT";
  ActivityType2["INVOICE_ADDED"] = "INVOICE_ADDED";
  ActivityType2["INVOICE_REMOVED"] = "INVOICE_REMOVED";
  ActivityType2["INVOICE_SENT"] = "INVOICE_SENT";
  ActivityType2["FULFILLER_EMAIL_SENT"] = "FULFILLER_EMAIL_SENT";
  ActivityType2["SHIPPING_ADDRESS_EDITED"] = "SHIPPING_ADDRESS_EDITED";
  ActivityType2["EMAIL_EDITED"] = "EMAIL_EDITED";
  ActivityType2["PICKUP_READY_EMAIL_SENT"] = "PICKUP_READY_EMAIL_SENT";
  ActivityType2["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
  ActivityType2["MERCHANT_COMMENT"] = "MERCHANT_COMMENT";
  ActivityType2["ORDER_CREATED_FROM_EXCHANGE"] = "ORDER_CREATED_FROM_EXCHANGE";
  ActivityType2["NEW_EXCHANGE_ORDER_CREATED"] = "NEW_EXCHANGE_ORDER_CREATED";
  ActivityType2["ORDER_PARTIALLY_PAID"] = "ORDER_PARTIALLY_PAID";
  ActivityType2["DRAFT_ORDER_CHANGES_APPLIED"] = "DRAFT_ORDER_CHANGES_APPLIED";
  ActivityType2["SAVED_PAYMENT_METHOD"] = "SAVED_PAYMENT_METHOD";
  ActivityType2["AUTHORIZED_PAYMENT_CREATED"] = "AUTHORIZED_PAYMENT_CREATED";
  ActivityType2["AUTHORIZED_PAYMENT_CAPTURED"] = "AUTHORIZED_PAYMENT_CAPTURED";
  ActivityType2["AUTHORIZED_PAYMENT_VOIDED"] = "AUTHORIZED_PAYMENT_VOIDED";
  ActivityType2["REFUND_INITIATED"] = "REFUND_INITIATED";
  ActivityType2["PAYMENT_REFUNDED"] = "PAYMENT_REFUNDED";
  ActivityType2["PAYMENT_REFUND_FAILED"] = "PAYMENT_REFUND_FAILED";
  ActivityType2["REFUNDED_AS_STORE_CREDIT"] = "REFUNDED_AS_STORE_CREDIT";
  ActivityType2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  ActivityType2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  ActivityType2["PAYMENT_DECLINED"] = "PAYMENT_DECLINED";
  ActivityType2["ORDER_PENDING"] = "ORDER_PENDING";
  ActivityType2["ORDER_REJECTED"] = "ORDER_REJECTED";
  ActivityType2["RECEIPT_CREATED"] = "RECEIPT_CREATED";
  ActivityType2["RECEIPT_SENT"] = "RECEIPT_SENT";
  ActivityType2["CHARGEBACK_CREATED"] = "CHARGEBACK_CREATED";
  ActivityType2["CHARGEBACK_REVERSED"] = "CHARGEBACK_REVERSED";
  return ActivityType2;
})(ActivityType || {});
var OrderActivityTypeEnumActivityType = /* @__PURE__ */ ((OrderActivityTypeEnumActivityType2) => {
  OrderActivityTypeEnumActivityType2["UNKNOWN_ACTIVITY_TYPE"] = "UNKNOWN_ACTIVITY_TYPE";
  OrderActivityTypeEnumActivityType2["ORDER_PLACED"] = "ORDER_PLACED";
  OrderActivityTypeEnumActivityType2["ORDER_PAID"] = "ORDER_PAID";
  OrderActivityTypeEnumActivityType2["ORDER_FULFILLED"] = "ORDER_FULFILLED";
  OrderActivityTypeEnumActivityType2["ORDER_NOT_FULFILLED"] = "ORDER_NOT_FULFILLED";
  OrderActivityTypeEnumActivityType2["ORDER_CANCELED"] = "ORDER_CANCELED";
  OrderActivityTypeEnumActivityType2["DOWNLOAD_LINK_SENT"] = "DOWNLOAD_LINK_SENT";
  OrderActivityTypeEnumActivityType2["TRACKING_NUMBER_ADDED"] = "TRACKING_NUMBER_ADDED";
  OrderActivityTypeEnumActivityType2["TRACKING_NUMBER_EDITED"] = "TRACKING_NUMBER_EDITED";
  OrderActivityTypeEnumActivityType2["TRACKING_LINK_ADDED"] = "TRACKING_LINK_ADDED";
  OrderActivityTypeEnumActivityType2["SHIPPING_CONFIRMATION_EMAIL_SENT"] = "SHIPPING_CONFIRMATION_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["INVOICE_ADDED"] = "INVOICE_ADDED";
  OrderActivityTypeEnumActivityType2["INVOICE_REMOVED"] = "INVOICE_REMOVED";
  OrderActivityTypeEnumActivityType2["INVOICE_SENT"] = "INVOICE_SENT";
  OrderActivityTypeEnumActivityType2["FULFILLER_EMAIL_SENT"] = "FULFILLER_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["SHIPPING_ADDRESS_EDITED"] = "SHIPPING_ADDRESS_EDITED";
  OrderActivityTypeEnumActivityType2["EMAIL_EDITED"] = "EMAIL_EDITED";
  OrderActivityTypeEnumActivityType2["PICKUP_READY_EMAIL_SENT"] = "PICKUP_READY_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
  OrderActivityTypeEnumActivityType2["MERCHANT_COMMENT"] = "MERCHANT_COMMENT";
  OrderActivityTypeEnumActivityType2["ORDER_CREATED_FROM_EXCHANGE"] = "ORDER_CREATED_FROM_EXCHANGE";
  OrderActivityTypeEnumActivityType2["NEW_EXCHANGE_ORDER_CREATED"] = "NEW_EXCHANGE_ORDER_CREATED";
  OrderActivityTypeEnumActivityType2["ORDER_PARTIALLY_PAID"] = "ORDER_PARTIALLY_PAID";
  OrderActivityTypeEnumActivityType2["DRAFT_ORDER_CHANGES_APPLIED"] = "DRAFT_ORDER_CHANGES_APPLIED";
  OrderActivityTypeEnumActivityType2["SAVED_PAYMENT_METHOD"] = "SAVED_PAYMENT_METHOD";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_CREATED"] = "AUTHORIZED_PAYMENT_CREATED";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_CAPTURED"] = "AUTHORIZED_PAYMENT_CAPTURED";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_VOIDED"] = "AUTHORIZED_PAYMENT_VOIDED";
  OrderActivityTypeEnumActivityType2["REFUND_INITIATED"] = "REFUND_INITIATED";
  OrderActivityTypeEnumActivityType2["PAYMENT_REFUNDED"] = "PAYMENT_REFUNDED";
  OrderActivityTypeEnumActivityType2["PAYMENT_REFUND_FAILED"] = "PAYMENT_REFUND_FAILED";
  OrderActivityTypeEnumActivityType2["REFUNDED_AS_STORE_CREDIT"] = "REFUNDED_AS_STORE_CREDIT";
  OrderActivityTypeEnumActivityType2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  OrderActivityTypeEnumActivityType2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  OrderActivityTypeEnumActivityType2["PAYMENT_DECLINED"] = "PAYMENT_DECLINED";
  OrderActivityTypeEnumActivityType2["ORDER_PENDING"] = "ORDER_PENDING";
  OrderActivityTypeEnumActivityType2["ORDER_REJECTED"] = "ORDER_REJECTED";
  OrderActivityTypeEnumActivityType2["RECEIPT_CREATED"] = "RECEIPT_CREATED";
  OrderActivityTypeEnumActivityType2["RECEIPT_SENT"] = "RECEIPT_SENT";
  OrderActivityTypeEnumActivityType2["CHARGEBACK_CREATED"] = "CHARGEBACK_CREATED";
  OrderActivityTypeEnumActivityType2["CHARGEBACK_REVERSED"] = "CHARGEBACK_REVERSED";
  OrderActivityTypeEnumActivityType2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  return OrderActivityTypeEnumActivityType2;
})(OrderActivityTypeEnumActivityType || {});
var AttributionSource = /* @__PURE__ */ ((AttributionSource2) => {
  AttributionSource2["UNSPECIFIED"] = "UNSPECIFIED";
  AttributionSource2["FACEBOOK_ADS"] = "FACEBOOK_ADS";
  return AttributionSource2;
})(AttributionSource || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["UNSPECIFIED"] = "UNSPECIFIED";
  ChannelType2["WEB"] = "WEB";
  ChannelType2["POS"] = "POS";
  ChannelType2["EBAY"] = "EBAY";
  ChannelType2["AMAZON"] = "AMAZON";
  ChannelType2["OTHER_PLATFORM"] = "OTHER_PLATFORM";
  ChannelType2["WIX_APP_STORE"] = "WIX_APP_STORE";
  ChannelType2["WIX_INVOICES"] = "WIX_INVOICES";
  ChannelType2["BACKOFFICE_MERCHANT"] = "BACKOFFICE_MERCHANT";
  ChannelType2["WISH"] = "WISH";
  ChannelType2["CLASS_PASS"] = "CLASS_PASS";
  ChannelType2["GLOBAL_E"] = "GLOBAL_E";
  ChannelType2["FACEBOOK"] = "FACEBOOK";
  ChannelType2["ETSY"] = "ETSY";
  ChannelType2["TIKTOK"] = "TIKTOK";
  ChannelType2["FAIRE_COM"] = "FAIRE_COM";
  ChannelType2["PAYPAL_AGENTIC_CHECKOUT"] = "PAYPAL_AGENTIC_CHECKOUT";
  return ChannelType2;
})(ChannelType || {});
var AdditionalFeeSource = /* @__PURE__ */ ((AdditionalFeeSource2) => {
  AdditionalFeeSource2["UNKNOWN_ADDITIONAL_FEE_SOURCE"] = "UNKNOWN_ADDITIONAL_FEE_SOURCE";
  AdditionalFeeSource2["SERVICE_PLUGIN"] = "SERVICE_PLUGIN";
  AdditionalFeeSource2["ITEM"] = "ITEM";
  AdditionalFeeSource2["MANUAL"] = "MANUAL";
  AdditionalFeeSource2["SHIPPING"] = "SHIPPING";
  AdditionalFeeSource2["PLATFORM"] = "PLATFORM";
  return AdditionalFeeSource2;
})(AdditionalFeeSource || {});
var OrderActionType = /* @__PURE__ */ ((OrderActionType2) => {
  OrderActionType2["EDIT"] = "EDIT";
  OrderActionType2["UPDATE"] = "UPDATE";
  OrderActionType2["REFUND"] = "REFUND";
  OrderActionType2["APPROVE"] = "APPROVE";
  OrderActionType2["CANCEL"] = "CANCEL";
  OrderActionType2["FULFILL"] = "FULFILL";
  OrderActionType2["COLLECT_PAYMENTS"] = "COLLECT_PAYMENTS";
  OrderActionType2["SEND_NOTIFICATIONS"] = "SEND_NOTIFICATIONS";
  OrderActionType2["ADD_MERCHANT_ACTIVITY"] = "ADD_MERCHANT_ACTIVITY";
  OrderActionType2["ADD_ACTIVITY"] = "ADD_ACTIVITY";
  OrderActionType2["CREATE_INVOICE"] = "CREATE_INVOICE";
  OrderActionType2["CREATE_RECEIPTS"] = "CREATE_RECEIPTS";
  OrderActionType2["ASSIGN_TAG"] = "ASSIGN_TAG";
  OrderActionType2["PRINT_PACKAGING_SLIP"] = "PRINT_PACKAGING_SLIP";
  OrderActionType2["MODIFY_INVENTORY"] = "MODIFY_INVENTORY";
  return OrderActionType2;
})(OrderActionType || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["UNKNOWN_CHARGE_TYPE"] = "UNKNOWN_CHARGE_TYPE";
  ChargeType2["PASS_ON"] = "PASS_ON";
  ChargeType2["ABSORBED"] = "ABSORBED";
  return ChargeType2;
})(ChargeType || {});
var V1JurisdictionType = /* @__PURE__ */ ((V1JurisdictionType2) => {
  V1JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  V1JurisdictionType2["COUNTRY"] = "COUNTRY";
  V1JurisdictionType2["STATE"] = "STATE";
  V1JurisdictionType2["COUNTY"] = "COUNTY";
  V1JurisdictionType2["CITY"] = "CITY";
  V1JurisdictionType2["SPECIAL"] = "SPECIAL";
  return V1JurisdictionType2;
})(V1JurisdictionType || {});
var ItemAvailabilityStatus = /* @__PURE__ */ ((ItemAvailabilityStatus2) => {
  ItemAvailabilityStatus2["AVAILABLE"] = "AVAILABLE";
  ItemAvailabilityStatus2["NOT_FOUND"] = "NOT_FOUND";
  ItemAvailabilityStatus2["NOT_AVAILABLE"] = "NOT_AVAILABLE";
  ItemAvailabilityStatus2["PARTIALLY_AVAILABLE"] = "PARTIALLY_AVAILABLE";
  return ItemAvailabilityStatus2;
})(ItemAvailabilityStatus || {});
var FileType = /* @__PURE__ */ ((FileType2) => {
  FileType2["UNSPECIFIED"] = "UNSPECIFIED";
  FileType2["SECURE_PICTURE"] = "SECURE_PICTURE";
  FileType2["SECURE_VIDEO"] = "SECURE_VIDEO";
  FileType2["SECURE_DOCUMENT"] = "SECURE_DOCUMENT";
  FileType2["SECURE_MUSIC"] = "SECURE_MUSIC";
  FileType2["SECURE_ARCHIVE"] = "SECURE_ARCHIVE";
  return FileType2;
})(FileType || {});
var PickupDetailsPickupMethod = /* @__PURE__ */ ((PickupDetailsPickupMethod2) => {
  PickupDetailsPickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupDetailsPickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupDetailsPickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupDetailsPickupMethod2;
})(PickupDetailsPickupMethod || {});
var V1ChargeType = /* @__PURE__ */ ((V1ChargeType2) => {
  V1ChargeType2["HANDLING_FEE"] = "HANDLING_FEE";
  V1ChargeType2["INSURANCE"] = "INSURANCE";
  return V1ChargeType2;
})(V1ChargeType || {});
var RuleType = /* @__PURE__ */ ((RuleType2) => {
  RuleType2["VALIDATION"] = "VALIDATION";
  RuleType2["OTHER"] = "OTHER";
  RuleType2["MAX"] = "MAX";
  RuleType2["MIN"] = "MIN";
  RuleType2["MAX_LENGTH"] = "MAX_LENGTH";
  RuleType2["MIN_LENGTH"] = "MIN_LENGTH";
  RuleType2["MAX_SIZE"] = "MAX_SIZE";
  RuleType2["MIN_SIZE"] = "MIN_SIZE";
  RuleType2["FORMAT"] = "FORMAT";
  RuleType2["DECIMAL_LTE"] = "DECIMAL_LTE";
  RuleType2["DECIMAL_GTE"] = "DECIMAL_GTE";
  RuleType2["DECIMAL_LT"] = "DECIMAL_LT";
  RuleType2["DECIMAL_GT"] = "DECIMAL_GT";
  RuleType2["DECIMAL_MAX_SCALE"] = "DECIMAL_MAX_SCALE";
  RuleType2["INVALID_ENUM_VALUE"] = "INVALID_ENUM_VALUE";
  RuleType2["REQUIRED_FIELD"] = "REQUIRED_FIELD";
  RuleType2["FIELD_NOT_ALLOWED"] = "FIELD_NOT_ALLOWED";
  RuleType2["ONE_OF_ALIGNMENT"] = "ONE_OF_ALIGNMENT";
  RuleType2["EXACT_LENGTH"] = "EXACT_LENGTH";
  RuleType2["EXACT_SIZE"] = "EXACT_SIZE";
  RuleType2["REQUIRED_ONE_OF_FIELD"] = "REQUIRED_ONE_OF_FIELD";
  return RuleType2;
})(RuleType || {});
var AppliedDiscountDiscountType = /* @__PURE__ */ ((AppliedDiscountDiscountType2) => {
  AppliedDiscountDiscountType2["GLOBAL"] = "GLOBAL";
  AppliedDiscountDiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  AppliedDiscountDiscountType2["SHIPPING"] = "SHIPPING";
  return AppliedDiscountDiscountType2;
})(AppliedDiscountDiscountType || {});
var RateType = /* @__PURE__ */ ((RateType2) => {
  RateType2["NO_TAX_COLLECTED"] = "NO_TAX_COLLECTED";
  RateType2["MANUAL_RATE"] = "MANUAL_RATE";
  RateType2["AUTO_RATE"] = "AUTO_RATE";
  RateType2["FALLBACK_RATE"] = "FALLBACK_RATE";
  return RateType2;
})(RateType || {});
var ManualCalculationReason = /* @__PURE__ */ ((ManualCalculationReason2) => {
  ManualCalculationReason2["GLOBAL_SETTING_TO_MANUAL"] = "GLOBAL_SETTING_TO_MANUAL";
  ManualCalculationReason2["REGION_SETTING_TO_MANUAL"] = "REGION_SETTING_TO_MANUAL";
  return ManualCalculationReason2;
})(ManualCalculationReason || {});
var FallbackReason = /* @__PURE__ */ ((FallbackReason2) => {
  FallbackReason2["AUTO_TAX_FAILED"] = "AUTO_TAX_FAILED";
  FallbackReason2["AUTO_TAX_DEACTIVATED"] = "AUTO_TAX_DEACTIVATED";
  return FallbackReason2;
})(FallbackReason || {});
var AdditionalFeeAdditionalFeeSource = /* @__PURE__ */ ((AdditionalFeeAdditionalFeeSource2) => {
  AdditionalFeeAdditionalFeeSource2["UNKNOWN_ADDITIONAL_FEE_SOURCE"] = "UNKNOWN_ADDITIONAL_FEE_SOURCE";
  AdditionalFeeAdditionalFeeSource2["SERVICE_PLUGIN"] = "SERVICE_PLUGIN";
  AdditionalFeeAdditionalFeeSource2["ITEM"] = "ITEM";
  AdditionalFeeAdditionalFeeSource2["MANUAL"] = "MANUAL";
  AdditionalFeeAdditionalFeeSource2["SHIPPING"] = "SHIPPING";
  AdditionalFeeAdditionalFeeSource2["PLATFORM"] = "PLATFORM";
  return AdditionalFeeAdditionalFeeSource2;
})(AdditionalFeeAdditionalFeeSource || {});
var Severity = /* @__PURE__ */ ((Severity2) => {
  Severity2["WARNING"] = "WARNING";
  Severity2["ERROR"] = "ERROR";
  return Severity2;
})(Severity || {});
var NameInOther = /* @__PURE__ */ ((NameInOther2) => {
  NameInOther2["OTHER_DEFAULT"] = "OTHER_DEFAULT";
  return NameInOther2;
})(NameInOther || {});
var NameInLineItem = /* @__PURE__ */ ((NameInLineItem2) => {
  NameInLineItem2["LINE_ITEM_DEFAULT"] = "LINE_ITEM_DEFAULT";
  return NameInLineItem2;
})(NameInLineItem || {});
var SuggestedFix = /* @__PURE__ */ ((SuggestedFix2) => {
  SuggestedFix2["UNKNOWN_SUGGESTED_FIX"] = "UNKNOWN_SUGGESTED_FIX";
  SuggestedFix2["REMOVE_LINE_ITEM"] = "REMOVE_LINE_ITEM";
  return SuggestedFix2;
})(SuggestedFix || {});
var V1PaymentOption = /* @__PURE__ */ ((V1PaymentOption2) => {
  V1PaymentOption2["UNKNOWN_PAYMENT_OPTION"] = "UNKNOWN_PAYMENT_OPTION";
  V1PaymentOption2["MANUAL"] = "MANUAL";
  return V1PaymentOption2;
})(V1PaymentOption || {});
var IdentityType = /* @__PURE__ */ ((IdentityType2) => {
  IdentityType2["UNKNOWN"] = "UNKNOWN";
  IdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  IdentityType2["MEMBER"] = "MEMBER";
  IdentityType2["WIX_USER"] = "WIX_USER";
  IdentityType2["APP"] = "APP";
  return IdentityType2;
})(IdentityType || {});

// src/interfaces-ecommerce-v1-payment-settings-provider.public.ts
var import_sdk_types = require("@wix/sdk-types");
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_page_url_v2 = require("@wix/sdk-runtime/transformations/page-url-v2");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");
var provideHandlers = (0, import_sdk_types.ServicePluginDefinition)("ECOM_PAYMENT_SETTINGS", [
  {
    name: "getPaymentSettings",
    primaryHttpMappingPath: "/v1/payment-settings",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = (0, import_transform_paths.transformPaths)(payload, [
          {
            transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
            paths: [
              { path: "request.order.createdDate" },
              { path: "request.order.updatedDate" },
              { path: "request.order.purchasedDate" },
              { path: "request.order.lineItems.digitalFile.expirationDate" },
              {
                path: "request.order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
              },
              { path: "request.order.shippingInfo.logistics.deliverByDate" },
              {
                path: "request.order.shippingInfo.logistics.deliveryTimeSlot.from"
              },
              {
                path: "request.order.shippingInfo.logistics.deliveryTimeSlot.to"
              },
              { path: "request.order.activities.createdDate" }
            ]
          },
          {
            transformFn: import_image.transformRESTImageToSDKImage,
            paths: [{ path: "request.order.lineItems.image" }]
          },
          {
            transformFn: import_float.transformRESTFloatToSDKFloat,
            paths: [
              { path: "request.order.lineItems.physicalProperties.weight" }
            ]
          },
          {
            transformFn: import_address.transformRESTAddressToSDKAddress,
            paths: [
              { path: "request.order.billingInfo.address" },
              {
                path: "request.order.shippingInfo.logistics.shippingDestination.address"
              },
              {
                path: "request.order.shippingInfo.logistics.pickupDetails.address"
              },
              { path: "request.order.recipientInfo.address" }
            ]
          }
        ]);
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  },
  {
    name: "getPaymentSettingsForCheckout",
    primaryHttpMappingPath: "/v1/payment-settings-for-checkout",
    transformations: {
      toREST: (payload) => {
        const toRestResponse = payload;
        return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
      },
      fromREST: (payload) => {
        const fromRestRequest = (0, import_transform_paths.transformPaths)(payload, [
          {
            transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
            paths: [
              { path: "request.checkout.createdDate" },
              { path: "request.checkout.updatedDate" },
              {
                path: "request.checkout.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate"
              },
              {
                path: "request.checkout.lineItems.serviceProperties.scheduledDate"
              },
              {
                path: "request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from"
              },
              {
                path: "request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to"
              },
              {
                path: "request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from"
              },
              {
                path: "request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to"
              },
              {
                path: "request.checkout.membershipOptions.eligibleMemberships.expirationDate"
              },
              {
                path: "request.checkout.membershipOptions.invalidMemberships.membership.expirationDate"
              },
              {
                path: "request.checkout.subscriptionCharges.charges.cycleBillingDate"
              }
            ]
          },
          {
            transformFn: import_page_url_v2.transformRESTPageURLV2ToSDKPageURLV2,
            paths: [{ path: "request.checkout.lineItems.url" }]
          },
          {
            transformFn: import_image.transformRESTImageToSDKImage,
            paths: [
              { path: "request.checkout.lineItems.media" },
              {
                path: "request.checkout.lineItems.catalogOverrideFields.image"
              }
            ]
          },
          {
            transformFn: import_float.transformRESTFloatToSDKFloat,
            paths: [
              { path: "request.checkout.lineItems.physicalProperties.weight" },
              {
                path: "request.checkout.lineItems.catalogOverrideFields.physicalProperties.weight"
              },
              {
                path: "request.checkout.appliedDiscounts.merchantDiscount.percentage"
              }
            ]
          },
          {
            transformFn: import_address.transformRESTAddressToSDKAddress,
            paths: [
              { path: "request.checkout.billingInfo.address" },
              {
                path: "request.checkout.shippingInfo.shippingDestination.address"
              },
              {
                path: "request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address"
              },
              {
                path: "request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address"
              }
            ]
          }
        ]);
        return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
      }
    }
  }
]);

// src/interfaces-ecommerce-v1-payment-settings-provider.context.ts
var import_service_plugin_modules = require("@wix/sdk-runtime/service-plugin-modules");
var provideHandlers2 = (0, import_service_plugin_modules.createServicePluginModule)(
  provideHandlers
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActivityType,
  AdditionalFeeAdditionalFeeSource,
  AdditionalFeeSource,
  AdjustmentType,
  AppliedDiscountDiscountType,
  AttributionSource,
  ChannelType,
  ChargeType,
  DescriptionLineType,
  DiscountReason,
  DiscountType,
  FallbackReason,
  FileType,
  FulfillmentStatus,
  IdentityType,
  ItemAvailabilityStatus,
  ItemTypePreset,
  JurisdictionType,
  LineItemQuantityChangeType,
  ManualCalculationReason,
  NameInLineItem,
  NameInOther,
  OrderActionType,
  OrderActivityTypeEnumActivityType,
  OrderStatus,
  PaymentOptionType,
  PaymentStatus,
  PickupDetailsPickupMethod,
  PickupMethod,
  RateType,
  RuleType,
  Severity,
  SubscriptionFrequency,
  SuggestedFix,
  TaxableAddressType,
  V1ChargeType,
  V1JurisdictionType,
  V1PaymentOption,
  VatType,
  WeightUnit,
  provideHandlers,
  publicProvideHandlers
});
