// @wix/auto_sdk_ecom_order-invoices/build/es/src/ecom-orders-v1-invoice-order-invoices.meta.js
import * as ambassadorWixEcomOrdersV1Invoice from './ecom-orders-v1-invoice-order-invoices.http.js';
export function listInvoicesForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.listInvoicesForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function generateInvoice() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.generateInvoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGenerateInvoices() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.bulkGenerateInvoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/invoices/generate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addInvoiceToOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomOrdersV1Invoice.addInvoiceToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/invoices/orders/{orderId}/add-invoice',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
