// @wix/auto_sdk_stores_abandoned-carts/build/cjs/src/stores-v1-abandoned-cart-abandoned-carts.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ActivityType = exports.Identity = exports.Status = exports.onAbandonedCartCartRecoveredEvent = exports.onAbandonedCartCartAbandonedEvent = exports.queryAbandonedCarts = exports.getAbandonedCart = void 0;
const stores_v1_abandoned_cart_abandoned_carts_public_js_1 = require("./stores-v1-abandoned-cart-abandoned-carts.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const stores_v1_abandoned_cart_abandoned_carts_public_js_2 = require("./stores-v1-abandoned-cart-abandoned-carts.public.js");
const stores_v1_abandoned_cart_abandoned_carts_public_js_3 = require("./stores-v1-abandoned-cart-abandoned-carts.public.js");
/** @internal */
exports.getAbandonedCart = (0, rest_modules_1.createRESTModule)(stores_v1_abandoned_cart_abandoned_carts_public_js_1.getAbandonedCart);
/** @internal */
exports.queryAbandonedCarts = (0, rest_modules_1.createRESTModule)(stores_v1_abandoned_cart_abandoned_carts_public_js_1.queryAbandonedCarts);
/**
 * <blockquote class='warning'>
 *
 * __Deprecation Notice:__
 *
 * This webhook has been replaced with the
 * [Abandoned Checkout Created Webhook](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/abandoned-checkout/abandoned-checkout-created)
 * and will be removed on March 31, 2024.
 *
 * </blockquote>
 * @deprecated
 */
exports.onAbandonedCartCartAbandonedEvent = (0, event_definition_modules_1.createEventModule)(stores_v1_abandoned_cart_abandoned_carts_public_js_2.onAbandonedCartCartAbandonedEvent);
/**
 * <blockquote class='warning'>
 *
 * __Deprecation Notice:__
 *
 * This webhook has been replaced with the
 * [Abandoned Checkout Recovered Webhook](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/abandoned-checkout/abandoned-checkout-recovered)
 * and will be removed on March 31, 2024.
 *
 * </blockquote>
 * @deprecated
 */
exports.onAbandonedCartCartRecoveredEvent = (0, event_definition_modules_1.createEventModule)(stores_v1_abandoned_cart_abandoned_carts_public_js_3.onAbandonedCartCartRecoveredEvent);
var stores_v1_abandoned_cart_abandoned_carts_universal_js_1 = require("./stores-v1-abandoned-cart-abandoned-carts.universal.js");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_1.Status; } });
Object.defineProperty(exports, "Identity", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_1.Identity; } });
Object.defineProperty(exports, "ActivityType", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_1.ActivityType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_1.WebhookIdentityType; } });
