// @wix/auto_sdk_stores_abandoned-carts/build/cjs/src/stores-v1-abandoned-cart-abandoned-carts.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.Identity = exports.ActivityType = exports.onAbandonedCartCartRecoveredEvent = exports.onAbandonedCartCartAbandonedEvent = exports.queryAbandonedCarts = exports.getAbandonedCart = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_v1_abandoned_cart_abandoned_carts_universal_js_1 = require("./stores-v1-abandoned-cart-abandoned-carts.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
/** @internal */
function getAbandonedCart(httpClient) {
    return (_id) => (0, stores_v1_abandoned_cart_abandoned_carts_universal_js_1.getAbandonedCart)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getAbandonedCart = getAbandonedCart;
/** @internal */
function queryAbandonedCarts(httpClient) {
    return (options) => (0, stores_v1_abandoned_cart_abandoned_carts_universal_js_1.queryAbandonedCarts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryAbandonedCarts = queryAbandonedCarts;
exports.onAbandonedCartCartAbandonedEvent = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.cart.abandonment.api.v1.CartAbandonedEvent', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.creationTime' }, { path: 'data.abandonTime' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'data.totals.subtotal' },
            { path: 'data.totals.total' },
        ],
    },
])))();
exports.onAbandonedCartCartRecoveredEvent = (0, sdk_types_1.EventDefinition)('com.wix.ecommerce.cart.abandonment.api.v1.CartRecoveredEvent', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.recoveredTime' },
            { path: 'data.creationTime' },
            { path: 'data.abandonedTime' },
        ],
    },
])))();
var stores_v1_abandoned_cart_abandoned_carts_universal_js_2 = require("./stores-v1-abandoned-cart-abandoned-carts.universal.js");
Object.defineProperty(exports, "ActivityType", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_2.ActivityType; } });
Object.defineProperty(exports, "Identity", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_2.Identity; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_v1_abandoned_cart_abandoned_carts_universal_js_2.WebhookIdentityType; } });
