// @wix/auto_sdk_stores_abandoned-carts/build/cjs/src/stores-v1-abandoned-cart-abandoned-carts.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryAbandonedCarts = exports.getAbandonedCart = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCartAbandonmentApiV1AbandonedCartsUrl(opts) {
    const domainToMappings = {};
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_abandoned-carts';
/** Returns abandoned carts based on the cart ID */
function getAbandonedCart(payload) {
    function __getAbandonedCart({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.v1.abandoned_cart',
            method: 'GET',
            methodFqn: 'wix.cart.abandonment.api.v1.AbandonedCarts.GetAbandonedCart',
            packageName: PACKAGE_NAME,
            url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
                protoPath: '/v1/abandonedCarts/{id}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCart.abandonTime' },
                        { path: 'abandonedCart.activities.timestamp' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getAbandonedCart;
}
exports.getAbandonedCart = getAbandonedCart;
/** Returns abandoned carts based on the requested query parameters */
function queryAbandonedCarts(payload) {
    function __queryAbandonedCarts({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.v1.abandoned_cart',
            method: 'POST',
            methodFqn: 'wix.cart.abandonment.api.v1.AbandonedCarts.QueryAbandonedCarts',
            packageName: PACKAGE_NAME,
            url: resolveWixCartAbandonmentApiV1AbandonedCartsUrl({
                protoPath: '/v1/abandonedCarts/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCarts.abandonTime' },
                        { path: 'abandonedCarts.activities.timestamp' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryAbandonedCarts;
}
exports.queryAbandonedCarts = queryAbandonedCarts;
