// @wix/auto_sdk_stores_customizations-v-3/build/es/src/stores-catalog-v3-customization-customizations-v-3.context.js
import { createCustomization as publicCreateCustomization, getCustomization as publicGetCustomization, updateCustomization as publicUpdateCustomization, deleteCustomization as publicDeleteCustomization, queryCustomizations as publicQueryCustomizations, bulkCreateCustomizations as publicBulkCreateCustomizations, addCustomizationChoices as publicAddCustomizationChoices, setCustomizationChoices as publicSetCustomizationChoices, removeCustomizationChoices as publicRemoveCustomizationChoices, bulkAddCustomizationChoices as publicBulkAddCustomizationChoices, bulkUpdateCustomizations as publicBulkUpdateCustomizations, } from './stores-catalog-v3-customization-customizations-v-3.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCustomizationCreated as publicOnCustomizationCreated } from './stores-catalog-v3-customization-customizations-v-3.public.js';
import { onCustomizationDeleted as publicOnCustomizationDeleted } from './stores-catalog-v3-customization-customizations-v-3.public.js';
import { onCustomizationUpdated as publicOnCustomizationUpdated } from './stores-catalog-v3-customization-customizations-v-3.public.js';
export const createCustomization = /*#__PURE__*/ createRESTModule(publicCreateCustomization);
export const getCustomization = /*#__PURE__*/ createRESTModule(publicGetCustomization);
export const updateCustomization = /*#__PURE__*/ createRESTModule(publicUpdateCustomization);
export const deleteCustomization = /*#__PURE__*/ createRESTModule(publicDeleteCustomization);
export const queryCustomizations = /*#__PURE__*/ createRESTModule(publicQueryCustomizations);
export const bulkCreateCustomizations = /*#__PURE__*/ createRESTModule(publicBulkCreateCustomizations);
export const addCustomizationChoices = /*#__PURE__*/ createRESTModule(publicAddCustomizationChoices);
export const setCustomizationChoices = /*#__PURE__*/ createRESTModule(publicSetCustomizationChoices);
export const removeCustomizationChoices = /*#__PURE__*/ createRESTModule(publicRemoveCustomizationChoices);
export const bulkAddCustomizationChoices = /*#__PURE__*/ createRESTModule(publicBulkAddCustomizationChoices);
export const bulkUpdateCustomizations = /*#__PURE__*/ createRESTModule(publicBulkUpdateCustomizations);
/**
 * Triggered when a customization is created.
 */
export const onCustomizationCreated = createEventModule(publicOnCustomizationCreated);
/**
 * Triggered when a customization is deleted.
 */
export const onCustomizationDeleted = createEventModule(publicOnCustomizationDeleted);
/**
 * Triggered when a customization is updated.
 */
export const onCustomizationUpdated = createEventModule(publicOnCustomizationUpdated);
export { CustomizationType, CustomizationRenderType, ChoiceType, RequestedFields, SortOrder, WebhookIdentityType, } from './stores-catalog-v3-customization-customizations-v-3.universal.js';
