// @wix/auto_sdk_stores_customizations-v-3/build/es/src/stores-catalog-v3-customization-customizations-v-3.meta.js
import * as ambassadorWixStoresCatalogV3Customization from './stores-catalog-v3-customization-customizations-v-3.http.js';
export function createCustomization() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.createCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCustomization() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.getCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/customizations/{customizationId}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCustomization() {
    const payload = { customization: { id: ':customizationId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.updateCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/customizations/{customization.id}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCustomization() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.deleteCustomization(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/customizations/{customizationId}',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.queryCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkCreateCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.addCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/add-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.setCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/set-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeCustomizationChoices() {
    const payload = { customizationId: ':customizationId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.removeCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/customizations/{customizationId}/remove-choices',
        pathParams: { customizationId: 'customizationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkAddCustomizationChoices() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkAddCustomizationChoices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/add-choices',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateCustomizations() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Customization.bulkUpdateCustomizations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/customizations/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
