// @wix/ecom/build/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  abandonedCheckouts: () => abandonedCheckouts,
  backInStockNotifications: () => backInStockNotifications,
  backInStockSettings: () => backInStockSettings,
  cart: () => cart,
  cartV2: () => cartV2,
  checkout: () => checkout,
  checkoutContent: () => checkoutContent,
  checkoutSettings: () => checkoutSettings,
  checkoutTemplates: () => checkoutTemplates,
  currencies: () => currencies,
  currentCart: () => currentCart,
  currentCartV2: () => currentCartV2,
  deliveryProfile: () => deliveryProfile,
  deliverySolutions: () => deliverySolutions,
  discountRules: () => discountRules,
  discountsCustomTrigger: () => discountsCustomTrigger,
  draftOrders: () => draftOrders,
  giftVouchers: () => giftVouchers,
  localDeliveryOptions: () => localDeliveryOptions,
  manualTaxMappings: () => manualTaxMappings,
  orderBilling: () => orderBilling,
  orderFulfillments: () => orderFulfillments,
  orderInvoices: () => orderInvoices,
  orderPaymentRequests: () => orderPaymentRequests,
  orderTransactions: () => orderTransactions,
  orders: () => orders,
  ordersSettings: () => ordersSettings,
  pickupLocations: () => pickupLocations,
  recommendations: () => recommendations,
  shippingOptions: () => shippingOptions,
  shippoConfigurations: () => shippoConfigurations,
  subscriptionContracts: () => subscriptionContracts,
  taxCalculation: () => taxCalculation,
  taxGroups: () => taxGroups,
  taxRegions: () => taxRegions,
  tipSettings: () => tipSettings,
  tippableStaff: () => tippableStaff,
  tips: () => tips,
  totalsCalculator: () => totalsCalculator
});
module.exports = __toCommonJS(meta_exports);
var manualTaxMappings = __toESM(require("@wix/auto_sdk_ecom_manual-tax-mappings/meta"));
var taxGroups = __toESM(require("@wix/auto_sdk_ecom_tax-groups/meta"));
var taxRegions = __toESM(require("@wix/auto_sdk_ecom_tax-regions/meta"));
var checkoutContent = __toESM(require("@wix/auto_sdk_ecom_checkout-content/meta"));
var discountsCustomTrigger = __toESM(require("@wix/auto_sdk_ecom_discounts-custom-trigger/meta"));
var discountRules = __toESM(require("@wix/auto_sdk_ecom_discount-rules/meta"));
var orderInvoices = __toESM(require("@wix/auto_sdk_ecom_order-invoices/meta"));
var recommendations = __toESM(require("@wix/auto_sdk_ecom_recommendations/meta"));
var subscriptionContracts = __toESM(require("@wix/auto_sdk_ecom_subscription-contracts/meta"));
var abandonedCheckouts = __toESM(require("@wix/auto_sdk_ecom_abandoned-checkouts/meta"));
var backInStockNotifications = __toESM(require("@wix/auto_sdk_ecom_back-in-stock-notifications/meta"));
var backInStockSettings = __toESM(require("@wix/auto_sdk_ecom_back-in-stock-settings/meta"));
var cart = __toESM(require("@wix/auto_sdk_ecom_cart/meta"));
var currentCart = __toESM(require("@wix/auto_sdk_ecom_current-cart/meta"));
var checkout = __toESM(require("@wix/auto_sdk_ecom_checkout/meta"));
var checkoutSettings = __toESM(require("@wix/auto_sdk_ecom_checkout-settings/meta"));
var checkoutTemplates = __toESM(require("@wix/auto_sdk_ecom_checkout-templates/meta"));
var deliveryProfile = __toESM(require("@wix/auto_sdk_ecom_delivery-profile/meta"));
var draftOrders = __toESM(require("@wix/auto_sdk_ecom_draft-orders/meta"));
var orderFulfillments = __toESM(require("@wix/auto_sdk_ecom_order-fulfillments/meta"));
var giftVouchers = __toESM(require("@wix/auto_sdk_ecom_gift-vouchers/meta"));
var localDeliveryOptions = __toESM(require("@wix/auto_sdk_ecom_local-delivery-options/meta"));
var orders = __toESM(require("@wix/auto_sdk_ecom_orders/meta"));
var orderBilling = __toESM(require("@wix/auto_sdk_ecom_order-billing/meta"));
var orderPaymentRequests = __toESM(require("@wix/auto_sdk_ecom_order-payment-requests/meta"));
var orderTransactions = __toESM(require("@wix/auto_sdk_ecom_order-transactions/meta"));
var ordersSettings = __toESM(require("@wix/auto_sdk_ecom_orders-settings/meta"));
var pickupLocations = __toESM(require("@wix/auto_sdk_ecom_pickup-locations/meta"));
var shippingOptions = __toESM(require("@wix/auto_sdk_ecom_shipping-options/meta"));
var shippoConfigurations = __toESM(require("@wix/auto_sdk_ecom_shippo-configurations/meta"));
var totalsCalculator = __toESM(require("@wix/auto_sdk_ecom_totals-calculator/meta"));
var cartV2 = __toESM(require("@wix/auto_sdk_ecom_cart-v-2/meta"));
var currentCartV2 = __toESM(require("@wix/auto_sdk_ecom_current-cart-v-2/meta"));
var currencies = __toESM(require("@wix/auto_sdk_ecom_currencies/meta"));
var taxCalculation = __toESM(require("@wix/auto_sdk_ecom_tax-calculation/meta"));
var deliverySolutions = __toESM(require("@wix/auto_sdk_ecom_delivery-solutions/meta"));
var tipSettings = __toESM(require("@wix/auto_sdk_ecom_tip-settings/meta"));
var tippableStaff = __toESM(require("@wix/auto_sdk_ecom_tippable-staff/meta"));
var tips = __toESM(require("@wix/auto_sdk_ecom_tips/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  abandonedCheckouts,
  backInStockNotifications,
  backInStockSettings,
  cart,
  cartV2,
  checkout,
  checkoutContent,
  checkoutSettings,
  checkoutTemplates,
  currencies,
  currentCart,
  currentCartV2,
  deliveryProfile,
  deliverySolutions,
  discountRules,
  discountsCustomTrigger,
  draftOrders,
  giftVouchers,
  localDeliveryOptions,
  manualTaxMappings,
  orderBilling,
  orderFulfillments,
  orderInvoices,
  orderPaymentRequests,
  orderTransactions,
  orders,
  ordersSettings,
  pickupLocations,
  recommendations,
  shippingOptions,
  shippoConfigurations,
  subscriptionContracts,
  taxCalculation,
  taxGroups,
  taxRegions,
  tipSettings,
  tippableStaff,
  tips,
  totalsCalculator
});
