// @wix/auto_sdk_ecom_checkout-settings/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AlignmentOriginal: () => Alignment,
  CheckoutBillingFieldsOriginal: () => CheckoutBillingFields,
  DisplayTypeOriginal: () => DisplayType,
  LogoSizeOriginal: () => LogoSize,
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  getCheckoutSettings: () => getCheckoutSettings2,
  updateCheckoutSettings: () => updateCheckoutSettings2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-v1-checkout-settings-checkout-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/checkout-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/checkout-settings",
        destPath: "/v1/checkout-settings"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/checkout-settings",
        destPath: "/v1/checkout-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/checkout-settings",
        destPath: "/v1/checkout-settings"
      }
    ],
    _: [
      {
        srcPath: "/ecom/v1/checkout-settings",
        destPath: "/v1/checkout-settings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_checkout-settings";
function getCheckoutSettings(payload) {
  function __getCheckoutSettings({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_settings",
      method: "GET",
      methodFqn: "wix.ecom.checkout_settings.api.v1.CheckoutSettingsService.GetCheckoutSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl({
        protoPath: "/v1/checkout-settings",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "checkoutSettings.createdDate" },
            { path: "checkoutSettings.updatedDate" },
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.x"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCheckoutSettings;
}
function updateCheckoutSettings(payload) {
  function __updateCheckoutSettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "checkoutSettings.createdDate" },
          { path: "checkoutSettings.updatedDate" },
          {
            path: "checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate"
          },
          {
            path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x"
          },
          {
            path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y"
          },
          {
            path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.x"
          },
          {
            path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.y"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.checkout_settings",
      method: "PATCH",
      methodFqn: "wix.ecom.checkout_settings.api.v1.CheckoutSettingsService.UpdateCheckoutSettings",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl({
        protoPath: "/v1/checkout-settings",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "checkoutSettings.createdDate" },
            { path: "checkoutSettings.updatedDate" },
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.x"
            },
            {
              path: "checkoutSettings.checkoutBrand.header.backgroundFill.image.focalPoint.y"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCheckoutSettings;
}

// src/ecom-v1-checkout-settings-checkout-settings.types.ts
var LogoSize = /* @__PURE__ */ ((LogoSize2) => {
  LogoSize2["UNKNOWN_LOGO_SIZE"] = "UNKNOWN_LOGO_SIZE";
  LogoSize2["SMALL"] = "SMALL";
  LogoSize2["MEDIUM"] = "MEDIUM";
  LogoSize2["LARGE"] = "LARGE";
  return LogoSize2;
})(LogoSize || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["UNKNOWN_ALIGNMENT"] = "UNKNOWN_ALIGNMENT";
  Alignment2["SIDE"] = "SIDE";
  Alignment2["CENTER"] = "CENTER";
  return Alignment2;
})(Alignment || {});
var DisplayType = /* @__PURE__ */ ((DisplayType2) => {
  DisplayType2["UNKNOWN_DISPLAY_TYPE"] = "UNKNOWN_DISPLAY_TYPE";
  DisplayType2["TEXT"] = "TEXT";
  DisplayType2["ICON"] = "ICON";
  return DisplayType2;
})(DisplayType || {});
var CheckoutBillingFields = /* @__PURE__ */ ((CheckoutBillingFields2) => {
  CheckoutBillingFields2["UNKNOWN_CHECKOUT_BILLING_FIELDS"] = "UNKNOWN_CHECKOUT_BILLING_FIELDS";
  CheckoutBillingFields2["All_FIELDS"] = "All_FIELDS";
  CheckoutBillingFields2["MINIMUM_REQUIRED_FIELDS"] = "MINIMUM_REQUIRED_FIELDS";
  return CheckoutBillingFields2;
})(CheckoutBillingFields || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/ecom-v1-checkout-settings-checkout-settings.meta.ts
function getCheckoutSettings2() {
  const payload = {};
  const getRequestOptions = getCheckoutSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/checkout-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCheckoutSettings2() {
  const payload = {};
  const getRequestOptions = updateCheckoutSettings(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/checkout-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlignmentOriginal,
  CheckoutBillingFieldsOriginal,
  DisplayTypeOriginal,
  LogoSizeOriginal,
  WebhookIdentityTypeOriginal,
  getCheckoutSettings,
  updateCheckoutSettings
});
