// @wix/auto_sdk_stores_stores-locations-v-3/build/es/src/stores-catalog-v3-stores-location-stores-locations-v-3.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixStoresCatalogLocationV3StoresLocationServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/location',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/location',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/location',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/stores/v3/locations',
                destPath: '/v3/locations',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_stores-locations-v-3';
/** Retrieves a Stores location. */
export function getStoresLocation(payload) {
    function __getStoresLocation({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.stores_location',
            method: 'GET',
            methodFqn: 'wix.stores.catalog.location.v3.StoresLocationService.GetStoresLocation',
            packageName: PACKAGE_NAME,
            url: resolveWixStoresCatalogLocationV3StoresLocationServiceUrl({
                protoPath: '/v3/locations/{storesLocationId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'storesLocation.createdDate' },
                        { path: 'storesLocation.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getStoresLocation;
}
/**
 * Retrieves a list of up to 100 Stores locations, given the provided filtering, sorting, and cursor paging.
 * Pass supported values to the `fields` array in the request to include those fields in the response.
 *
 *
 * Query Stores Locations runs with these defaults, which you can override:
 *
 * - `createdDate` is sorted in `DESC` order
 * - `cursorPaging.limit` is `100`
 *
 * For field support for filters and sorting,
 * see [Stores Locations: Supported Filters and Sorting](https://dev.wix.com/docs/rest/business-solutions/stores/catalog-v3/stores-locations-v3/supported-filters-and-sorting).
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/docs/rest/articles/getting-started/api-query-language),
 * and [Sorting and Paging](https://dev.wix.com/docs/rest/articles/getting-started/sorting-and-paging).
 */
export function queryStoresLocations(payload) {
    function __queryStoresLocations({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.stores_location',
            method: 'POST',
            methodFqn: 'wix.stores.catalog.location.v3.StoresLocationService.QueryStoresLocations',
            packageName: PACKAGE_NAME,
            url: resolveWixStoresCatalogLocationV3StoresLocationServiceUrl({
                protoPath: '/v3/locations/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'storesLocations.createdDate' },
                        { path: 'storesLocations.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryStoresLocations;
}
