// @wix/auto_sdk_stores_info-sections-v-3/build/cjs/src/stores-catalog-v3-info-section-info-sections-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkDeleteInfoSections = exports.bulkUpdateInfoSections = exports.bulkCreateInfoSections = exports.queryInfoSections = exports.deleteInfoSection = exports.updateInfoSection = exports.bulkGetOrCreateInfoSections = exports.getOrCreateInfoSection = exports.getInfoSection = exports.createInfoSection = void 0;
const ambassadorWixStoresCatalogV3InfoSection = __importStar(require("./stores-catalog-v3-info-section-info-sections-v-3.http.js"));
function createInfoSection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.createInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createInfoSection = createInfoSection;
function getInfoSection() {
    const payload = { infoSectionId: ':infoSectionId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.getInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/info-sections/{infoSectionId}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getInfoSection = getInfoSection;
function getOrCreateInfoSection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.getOrCreateInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrCreateInfoSection = getOrCreateInfoSection;
function bulkGetOrCreateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkGetOrCreateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGetOrCreateInfoSections = bulkGetOrCreateInfoSections;
function updateInfoSection() {
    const payload = { infoSection: { id: ':infoSectionId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.updateInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/info-sections/{infoSection.id}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateInfoSection = updateInfoSection;
function deleteInfoSection() {
    const payload = { infoSectionId: ':infoSectionId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.deleteInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/info-sections/{infoSectionId}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteInfoSection = deleteInfoSection;
function queryInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.queryInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryInfoSections = queryInfoSections;
function bulkCreateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkCreateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateInfoSections = bulkCreateInfoSections;
function bulkUpdateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkUpdateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateInfoSections = bulkUpdateInfoSections;
function bulkDeleteInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkDeleteInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteInfoSections = bulkDeleteInfoSections;
