// @wix/auto_sdk_stores_info-sections-v-3/build/cjs/src/stores-catalog-v3-info-section-info-sections-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidthType = exports.Width = exports.WebhookIdentityType = exports.VoteRole = exports.ViewRole = exports.ViewMode = exports.VerticalAlignment = exports.Type = exports.ThumbnailsAlignment = exports.TextAlignment = exports.Target = exports.Source = exports.SortOrder = exports.RequestedFields = exports.PollLayoutType = exports.PollLayoutDirection = exports.PluginContainerDataAlignment = exports.Orientation = exports.NullValue = exports.NodeType = exports.MapType = exports.LineStyle = exports.LayoutType = exports.InitialExpandedItems = exports.GIFType = exports.FontType = exports.Direction = exports.DecorationType = exports.Crop = exports.BackgroundType = exports.AppType = exports.Alignment = exports.onInfoSectionUpdated = exports.onInfoSectionDeleted = exports.onInfoSectionCreated = exports.bulkDeleteInfoSections = exports.bulkUpdateInfoSections = exports.bulkCreateInfoSections = exports.queryInfoSections = exports.deleteInfoSection = exports.updateInfoSection = exports.bulkGetOrCreateInfoSections = exports.getOrCreateInfoSection = exports.getInfoSection = exports.createInfoSection = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_info_section_info_sections_v_3_universal_js_1 = require("./stores-catalog-v3-info-section-info-sections-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createInfoSection(httpClient) {
    return (infoSection) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.createInfoSection)(infoSection, 
    // @ts-ignore
    { httpClient });
}
exports.createInfoSection = createInfoSection;
function getInfoSection(httpClient) {
    return (infoSectionId, options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.getInfoSection)(infoSectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getInfoSection = getInfoSection;
function getOrCreateInfoSection(httpClient) {
    return (options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.getOrCreateInfoSection)(options, 
    // @ts-ignore
    { httpClient });
}
exports.getOrCreateInfoSection = getOrCreateInfoSection;
function bulkGetOrCreateInfoSections(httpClient) {
    return (options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.bulkGetOrCreateInfoSections)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkGetOrCreateInfoSections = bulkGetOrCreateInfoSections;
function updateInfoSection(httpClient) {
    return (_id, infoSection, options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.updateInfoSection)(_id, infoSection, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateInfoSection = updateInfoSection;
function deleteInfoSection(httpClient) {
    return (infoSectionId) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.deleteInfoSection)(infoSectionId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteInfoSection = deleteInfoSection;
function queryInfoSections(httpClient) {
    return (options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.queryInfoSections)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryInfoSections = queryInfoSections;
function bulkCreateInfoSections(httpClient) {
    return (infoSections, options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.bulkCreateInfoSections)(infoSections, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateInfoSections = bulkCreateInfoSections;
function bulkUpdateInfoSections(httpClient) {
    return (infoSections, options) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.bulkUpdateInfoSections)(infoSections, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateInfoSections = bulkUpdateInfoSections;
function bulkDeleteInfoSections(httpClient) {
    return (infoSectionIds) => (0, stores_catalog_v3_info_section_info_sections_v_3_universal_js_1.bulkDeleteInfoSections)(infoSectionIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteInfoSections = bulkDeleteInfoSections;
exports.onInfoSectionCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.info_section_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
        ],
    },
])))();
exports.onInfoSectionDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.info_section_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'undefined.description.metadata.createdTimestamp' },
            { path: 'undefined.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'undefined.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.options.item.ratio',
            },
            { path: 'undefined.description.nodes.imageData.image.duration' },
            { path: 'undefined.description.nodes.mapData.mapSettings.lat' },
            { path: 'undefined.description.nodes.mapData.mapSettings.lng' },
            {
                path: 'undefined.description.nodes.pollData.poll.image.duration',
            },
            {
                path: 'undefined.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'undefined.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'undefined.description.nodes.appEmbedData.image.duration' },
            { path: 'undefined.description.nodes.videoData.video.duration' },
            {
                path: 'undefined.description.nodes.videoData.thumbnail.duration',
            },
            { path: 'undefined.description.nodes.audioData.audio.duration' },
            {
                path: 'undefined.description.nodes.audioData.coverImage.duration',
            },
        ],
    },
])))();
exports.onInfoSectionUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.info_section_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
        ],
    },
])))();
var stores_catalog_v3_info_section_info_sections_v_3_universal_js_2 = require("./stores-catalog-v3-info-section-info-sections-v-3.universal.js");
Object.defineProperty(exports, "Alignment", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Alignment; } });
Object.defineProperty(exports, "AppType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.AppType; } });
Object.defineProperty(exports, "BackgroundType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.BackgroundType; } });
Object.defineProperty(exports, "Crop", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Crop; } });
Object.defineProperty(exports, "DecorationType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.DecorationType; } });
Object.defineProperty(exports, "Direction", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Direction; } });
Object.defineProperty(exports, "FontType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.FontType; } });
Object.defineProperty(exports, "GIFType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.GIFType; } });
Object.defineProperty(exports, "InitialExpandedItems", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.InitialExpandedItems; } });
Object.defineProperty(exports, "LayoutType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.LayoutType; } });
Object.defineProperty(exports, "LineStyle", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.LineStyle; } });
Object.defineProperty(exports, "MapType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.MapType; } });
Object.defineProperty(exports, "NodeType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.NodeType; } });
Object.defineProperty(exports, "NullValue", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.NullValue; } });
Object.defineProperty(exports, "Orientation", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Orientation; } });
Object.defineProperty(exports, "PluginContainerDataAlignment", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.PluginContainerDataAlignment; } });
Object.defineProperty(exports, "PollLayoutDirection", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.PollLayoutDirection; } });
Object.defineProperty(exports, "PollLayoutType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.PollLayoutType; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Source; } });
Object.defineProperty(exports, "Target", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Target; } });
Object.defineProperty(exports, "TextAlignment", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.TextAlignment; } });
Object.defineProperty(exports, "ThumbnailsAlignment", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.ThumbnailsAlignment; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Type; } });
Object.defineProperty(exports, "VerticalAlignment", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.VerticalAlignment; } });
Object.defineProperty(exports, "ViewMode", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.ViewMode; } });
Object.defineProperty(exports, "ViewRole", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.ViewRole; } });
Object.defineProperty(exports, "VoteRole", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.VoteRole; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "Width", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.Width; } });
Object.defineProperty(exports, "WidthType", { enumerable: true, get: function () { return stores_catalog_v3_info_section_info_sections_v_3_universal_js_2.WidthType; } });
