// @wix/auto_sdk_ecom_currencies/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  convertCurrency: () => convertCurrency2,
  getConversionRate: () => getConversionRate,
  listCurrencies: () => listCurrencies2
});
module.exports = __toCommonJS(meta_exports);

// src/ecommerce-currency-converter-v1-currency-rate-currencies.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/currency-converter",
        destPath: ""
      }
    ],
    "cronulla-jobs._base_domain_": [
      {
        srcPath: "/currency-converter-bo",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "currency-converter.wixapps.net": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/currency_converter",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_currencies";
function listCurrencies(payload) {
  function __listCurrencies({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ListCurrencies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listCurrencies;
}
function convertCurrency(payload) {
  function __convertCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "POST",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConvertCurrency",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/amounts/{from}/convert/{to}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __convertCurrency;
}
function conversionRate(payload) {
  function __conversionRate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConversionRate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/rate/{from}/convert/{to}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __conversionRate;
}

// src/ecommerce-currency-converter-v1-currency-rate-currencies.meta.ts
function listCurrencies2() {
  const payload = {};
  const getRequestOptions = listCurrencies(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/currencies",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function convertCurrency2() {
  const payload = { from: ":from", to: ":to" };
  const getRequestOptions = convertCurrency(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/currencies/amounts/{from}/convert/{to}",
    pathParams: { from: "from", to: "to" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getConversionRate() {
  const payload = { from: ":from", to: ":to" };
  const getRequestOptions = conversionRate(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/currencies/rate/{from}/convert/{to}",
    pathParams: { from: "from", to: "to" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  convertCurrency,
  getConversionRate,
  listCurrencies
});
