// @wix/auto_sdk_ecom_currencies/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  convertCurrency: () => convertCurrency4,
  getConversionRate: () => getConversionRate3,
  listCurrencies: () => listCurrencies4
});
module.exports = __toCommonJS(index_exports);

// src/ecommerce-currency-converter-v1-currency-rate-currencies.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecommerce-currency-converter-v1-currency-rate-currencies.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/currency-converter",
        destPath: ""
      }
    ],
    "cronulla-jobs._base_domain_": [
      {
        srcPath: "/currency-converter-bo",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "currency-converter.wixapps.net": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/currency_converter",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/currency-converter",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_currencies";
function listCurrencies(payload) {
  function __listCurrencies({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ListCurrencies",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listCurrencies;
}
function convertCurrency(payload) {
  function __convertCurrency({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "POST",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConvertCurrency",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/amounts/{from}/convert/{to}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __convertCurrency;
}
function conversionRate(payload) {
  function __conversionRate({ host }) {
    const metadata = {
      entityFqdn: "wix.ecommerce.currency_converter.v1.currency_rate",
      method: "GET",
      methodFqn: "com.wixpress.currency.converter.api.v1.CurrencyConverter.ConversionRate",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
        protoPath: "/v1/currencies/rate/{from}/convert/{to}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "rateTimestamp" }]
        }
      ])
    };
    return metadata;
  }
  return __conversionRate;
}

// src/ecommerce-currency-converter-v1-currency-rate-currencies.universal.ts
async function listCurrencies2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listCurrencies(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function convertCurrency2(identifiers, amounts) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    from: identifiers?.from,
    to: identifiers?.to,
    amounts
  });
  const reqOpts = convertCurrency(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          from: "$[0].from",
          to: "$[0].to",
          amounts: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "amounts"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getConversionRate(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    from: identifiers?.from,
    to: identifiers?.to
  });
  const reqOpts = conversionRate(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { from: "$[0].from", to: "$[0].to" },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecommerce-currency-converter-v1-currency-rate-currencies.public.ts
function listCurrencies3(httpClient) {
  return () => listCurrencies2(
    // @ts-ignore
    { httpClient }
  );
}
function convertCurrency3(httpClient) {
  return (identifiers, amounts) => convertCurrency2(
    identifiers,
    amounts,
    // @ts-ignore
    { httpClient }
  );
}
function getConversionRate2(httpClient) {
  return (identifiers) => getConversionRate(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}

// src/ecommerce-currency-converter-v1-currency-rate-currencies.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var listCurrencies4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listCurrencies3);
var convertCurrency4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(convertCurrency3);
var getConversionRate3 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getConversionRate2);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  convertCurrency,
  getConversionRate,
  listCurrencies
});
