// @wix/auto_sdk_ecom_orders/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  ActionType: () => ActionType,
  ActivityType: () => ActivityType,
  AdditionalFeeSource: () => AdditionalFeeSource,
  AdjustmentType: () => AdjustmentType,
  AttributionSource: () => AttributionSource,
  AuthorizationCaptureStatus: () => AuthorizationCaptureStatus,
  AuthorizationVoidStatus: () => AuthorizationVoidStatus,
  ChannelType: () => ChannelType,
  ChargeType: () => ChargeType,
  ChargebackStatus: () => ChargebackStatus,
  CustomFieldGroup: () => CustomFieldGroup,
  DeltaPaymentOptionType: () => DeltaPaymentOptionType,
  DepositType: () => DepositType,
  DescriptionLineType: () => DescriptionLineType,
  DiscountReason: () => DiscountReason,
  DiscountType: () => DiscountType,
  DurationUnit: () => DurationUnit,
  FulfillmentStatus: () => FulfillmentStatus,
  InventoryAction: () => InventoryAction,
  InvoiceStatus: () => InvoiceStatus,
  ItemTypePreset: () => ItemTypePreset,
  JurisdictionType: () => JurisdictionType,
  LineItemQuantityChangeType: () => LineItemQuantityChangeType,
  ManuallyRefundableReason: () => ManuallyRefundableReason,
  MembershipPaymentStatus: () => MembershipPaymentStatus,
  NonRefundableReason: () => NonRefundableReason,
  OrderActionType: () => OrderActionType,
  OrderActivityTypeEnumActivityType: () => OrderActivityTypeEnumActivityType,
  OrderApprovalStrategy: () => OrderApprovalStrategy,
  OrderStatus: () => OrderStatus,
  PaymentCollectabilityStatus: () => PaymentCollectabilityStatus,
  PaymentOptionType: () => PaymentOptionType,
  PaymentStatus: () => PaymentStatus,
  PickupMethod: () => PickupMethod,
  Placement: () => Placement,
  PreviewEmailType: () => PreviewEmailType,
  Reason: () => Reason,
  RefundStatus: () => RefundStatus,
  RefundableStatus: () => RefundableStatus,
  RestockType: () => RestockType,
  ScheduledAction: () => ScheduledAction,
  SortOrder: () => SortOrder,
  SourceType: () => SourceType,
  SubdivisionType: () => SubdivisionType,
  SubscriptionFrequency: () => SubscriptionFrequency,
  TaxableAddressType: () => TaxableAddressType,
  TransactionStatus: () => TransactionStatus,
  ValueType: () => ValueType,
  VatType: () => VatType,
  VersioningMode: () => VersioningMode,
  WebhookIdentityType: () => WebhookIdentityType,
  WeightUnit: () => WeightUnit,
  addActivities: () => addActivities4,
  addActivity: () => addActivity4,
  aggregateOrders: () => aggregateOrders4,
  bulkDeleteImportedOrders: () => bulkDeleteImportedOrders4,
  bulkUpdateOrderTags: () => bulkUpdateOrderTags4,
  bulkUpdateOrders: () => bulkUpdateOrders4,
  cancelOrder: () => cancelOrder4,
  captureAuthorizedPayments: () => captureAuthorizedPayments4,
  chargeMemberships: () => chargeMemberships4,
  createOrder: () => createOrder4,
  deleteActivity: () => deleteActivity4,
  getOrder: () => getOrder4,
  getPaymentCollectabilityStatus: () => getPaymentCollectabilityStatus4,
  getRefundabilityStatus: () => getRefundabilityStatus4,
  importOrder: () => importOrder4,
  onOrderApproved: () => onOrderApproved2,
  onOrderCanceled: () => onOrderCanceled2,
  onOrderCreated: () => onOrderCreated2,
  onOrderPaymentStatusUpdated: () => onOrderPaymentStatusUpdated2,
  onOrderUpdated: () => onOrderUpdated2,
  paymentCollectionBulkMarkOrdersAsPaid: () => paymentCollectionBulkMarkOrdersAsPaid4,
  paymentCollectionCreatePaymentGatewayOrder: () => paymentCollectionCreatePaymentGatewayOrder4,
  paymentCollectionMarkOrderAsPaid: () => paymentCollectionMarkOrderAsPaid4,
  preparePaymentCollection: () => preparePaymentCollection4,
  recordManuallyCollectedPayment: () => recordManuallyCollectedPayment4,
  searchOrders: () => searchOrders4,
  setOrderNumberCounter: () => setOrderNumberCounter4,
  triggerRefund: () => triggerRefund4,
  updateActivity: () => updateActivity4,
  updateOrder: () => updateOrder4,
  updateOrderLineItem: () => updateOrderLineItem4,
  updateOrderStatus: () => updateOrderStatus4,
  voidAuthorizedPayments: () => voidAuthorizedPayments4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-v1-order-orders.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_address3 = require("@wix/sdk-runtime/transformations/address");
var import_float3 = require("@wix/sdk-runtime/transformations/float");
var import_image3 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/ecom-v1-order-orders.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-v1-order-orders.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomOrdersV1OrdersUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/ecom-orders",
        destPath: ""
      },
      {
        srcPath: "/_api/orders",
        destPath: ""
      },
      {
        srcPath: "/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/ecom/v1/bulk/orders",
        destPath: "/v1/bulk/orders"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/ecom-orders",
        destPath: ""
      },
      {
        srcPath: "/_api/ep-orders",
        destPath: "/api"
      },
      {
        srcPath: "/ecom/v1/ep-orders",
        destPath: "/v1/orders"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/bulk/orders",
        destPath: "/v1/bulk/orders"
      },
      {
        srcPath: "/ecom/v1/orders",
        destPath: "/v1/orders"
      },
      {
        srcPath: "/ecom/v1/ep-orders",
        destPath: "/v1/orders"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/bulk/orders",
        destPath: "/v1/bulk/orders"
      },
      {
        srcPath: "/ecom/v1/orders",
        destPath: "/v1/orders"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/payments-collector",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/payments-collector",
        destPath: "/v1/payments-collector"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/payments-collector",
        destPath: "/v1/payments-collector"
      },
      {
        srcPath: "/_api/v1/payments-collector",
        destPath: "/v1/payments-collector"
      },
      {
        srcPath: "/v1/payments-collector",
        destPath: "/v1/payments-collector"
      },
      {
        srcPath: "/_api/payments-collector",
        destPath: "/v1/payments-collector"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixEcomOrdersImporterV1OrderImportsUrl(opts) {
  const domainToMappings = {};
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_orders";
function importOrder(payload) {
  function __importOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "order.createdDate" },
          { path: "order.updatedDate" },
          { path: "order.purchasedDate" },
          { path: "order.lineItems.image.urlExpirationDate" },
          { path: "order.lineItems.digitalFile.expirationDate" },
          {
            path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "order.shippingInfo.logistics.deliverByDate" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "order.activities.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "order.lineItems.image.focalPoint.x" },
          { path: "order.lineItems.image.focalPoint.y" },
          { path: "order.lineItems.physicalProperties.weight" },
          { path: "order.billingInfo.address.geocode.latitude" },
          { path: "order.billingInfo.address.geocode.longitude" },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
          },
          { path: "order.recipientInfo.address.geocode.latitude" },
          { path: "order.recipientInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.importer.v1.OrderImports.ImportOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersImporterV1OrderImportsUrl({
        protoPath: "/v1/orders/import",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __importOrder;
}
function setOrderNumberCounter(payload) {
  function __setOrderNumberCounter({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.importer.v1.OrderImports.SetOrderNumberCounter",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersImporterV1OrderImportsUrl({
        protoPath: "/v1/orders/import/set-order-number-counter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setOrderNumberCounter;
}
function bulkDeleteImportedOrders(payload) {
  function __bulkDeleteImportedOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.importer.v1.OrderImports.BulkDeleteImportedOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersImporterV1OrderImportsUrl({
        protoPath: "/v1/orders/import/bulk-delete-imported-orders",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteImportedOrders;
}
function preparePaymentCollection(payload) {
  function __preparePaymentCollection({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.PreparePaymentCollection",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/prepare-payment-collection",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __preparePaymentCollection;
}
function getPaymentCollectabilityStatus(payload) {
  function __getPaymentCollectabilityStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "GET",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.GetPaymentCollectabilityStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/payment-collectability-status",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getPaymentCollectabilityStatus;
}
function recordManuallyCollectedPayment(payload) {
  function __recordManuallyCollectedPayment({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.RecordManuallyCollectedPayment",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{orderId}/manual-payment",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __recordManuallyCollectedPayment;
}
function paymentCollectionMarkOrderAsPaid(payload) {
  function __paymentCollectionMarkOrderAsPaid({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.MarkOrderAsPaid",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/mark-as-paid",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __paymentCollectionMarkOrderAsPaid;
}
function paymentCollectionBulkMarkOrdersAsPaid(payload) {
  function __paymentCollectionBulkMarkOrdersAsPaid({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.BulkMarkOrdersAsPaid",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/bulk/payments-collector/mark-as-paid",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.purchasedDate" },
            { path: "results.item.lineItems.image.urlExpirationDate" },
            { path: "results.item.lineItems.digitalFile.expirationDate" },
            {
              path: "results.item.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "results.item.shippingInfo.logistics.deliverByDate" },
            {
              path: "results.item.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "results.item.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "results.item.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.lineItems.image.focalPoint.x" },
            { path: "results.item.lineItems.image.focalPoint.y" },
            { path: "results.item.lineItems.physicalProperties.weight" },
            { path: "results.item.billingInfo.address.geocode.latitude" },
            { path: "results.item.billingInfo.address.geocode.longitude" },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "results.item.recipientInfo.address.geocode.latitude" },
            { path: "results.item.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __paymentCollectionBulkMarkOrdersAsPaid;
}
function getRefundabilityStatus(payload) {
  function __getRefundabilityStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "GET",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.GetRefundabilityStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/refundability",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getRefundabilityStatus;
}
function paymentCollectionCreatePaymentGatewayOrder(payload) {
  function __paymentCollectionCreatePaymentGatewayOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.CreatePaymentGatewayOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/create-payment-gateway-order",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __paymentCollectionCreatePaymentGatewayOrder;
}
function chargeMemberships(payload) {
  function __chargeMemberships({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "membershipCharges.serviceProperties.scheduledDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.ChargeMemberships",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/charge-membership",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __chargeMemberships;
}
function triggerRefund(payload) {
  function __triggerRefund({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.TriggerRefund",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/trigger-refund",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __triggerRefund;
}
function voidAuthorizedPayments(payload) {
  function __voidAuthorizedPayments({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.VoidAuthorizedPayments",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/void-authorized-payments",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __voidAuthorizedPayments;
}
function captureAuthorizedPayments(payload) {
  function __captureAuthorizedPayments({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.payments_collector.v1.PaymentsCollector.CaptureAuthorizedPayments",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersPaymentsCollectorV1PaymentsCollectorUrl({
        protoPath: "/v1/payments-collector/orders/{ecomOrderId}/capture-authorized-payments",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orderTransactions.payments.createdDate" },
            { path: "orderTransactions.payments.updatedDate" },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.createdDate"
            },
            {
              path: "orderTransactions.payments.regularPaymentDetails.chargebacks.updatedDate"
            },
            { path: "orderTransactions.refunds.createdDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __captureAuthorizedPayments;
}
function getOrder(payload) {
  function __getOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "GET",
      methodFqn: "com.wix.ecom.orders.v1.Orders.GetOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getOrder;
}
function searchOrders(payload) {
  function __searchOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.SearchOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/search",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "orders.createdDate" },
            { path: "orders.updatedDate" },
            { path: "orders.purchasedDate" },
            { path: "orders.lineItems.image.urlExpirationDate" },
            { path: "orders.lineItems.digitalFile.expirationDate" },
            {
              path: "orders.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "orders.shippingInfo.logistics.deliverByDate" },
            { path: "orders.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "orders.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "orders.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "orders.lineItems.image.focalPoint.x" },
            { path: "orders.lineItems.image.focalPoint.y" },
            { path: "orders.lineItems.physicalProperties.weight" },
            { path: "orders.billingInfo.address.geocode.latitude" },
            { path: "orders.billingInfo.address.geocode.longitude" },
            {
              path: "orders.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "orders.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "orders.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "orders.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "orders.recipientInfo.address.geocode.latitude" },
            { path: "orders.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __searchOrders;
}
function createOrder(payload) {
  function __createOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "order.createdDate" },
          { path: "order.updatedDate" },
          { path: "order.purchasedDate" },
          { path: "order.lineItems.image.urlExpirationDate" },
          { path: "order.lineItems.digitalFile.expirationDate" },
          {
            path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "order.shippingInfo.logistics.deliverByDate" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "order.activities.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "order.lineItems.image.focalPoint.x" },
          { path: "order.lineItems.image.focalPoint.y" },
          { path: "order.lineItems.physicalProperties.weight" },
          { path: "order.billingInfo.address.geocode.latitude" },
          { path: "order.billingInfo.address.geocode.longitude" },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
          },
          { path: "order.recipientInfo.address.geocode.latitude" },
          { path: "order.recipientInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.CreateOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createOrder;
}
function updateOrder(payload) {
  function __updateOrder({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "order.createdDate" },
          { path: "order.updatedDate" },
          { path: "order.purchasedDate" },
          { path: "order.lineItems.image.urlExpirationDate" },
          { path: "order.lineItems.digitalFile.expirationDate" },
          {
            path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "order.shippingInfo.logistics.deliverByDate" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "order.activities.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "order.lineItems.image.focalPoint.x" },
          { path: "order.lineItems.image.focalPoint.y" },
          { path: "order.lineItems.physicalProperties.weight" },
          { path: "order.billingInfo.address.geocode.latitude" },
          { path: "order.billingInfo.address.geocode.longitude" },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
          },
          { path: "order.recipientInfo.address.geocode.latitude" },
          { path: "order.recipientInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "PATCH",
      methodFqn: "com.wix.ecom.orders.v1.Orders.UpdateOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{order.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrder;
}
function bulkUpdateOrders(payload) {
  function __bulkUpdateOrders({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "orders.fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "orders.order.createdDate" },
          { path: "orders.order.updatedDate" },
          { path: "orders.order.purchasedDate" },
          { path: "orders.order.lineItems.image.urlExpirationDate" },
          { path: "orders.order.lineItems.digitalFile.expirationDate" },
          {
            path: "orders.order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "orders.order.shippingInfo.logistics.deliverByDate" },
          { path: "orders.order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "orders.order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "orders.order.activities.createdDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "orders.order.lineItems.image.focalPoint.x" },
          { path: "orders.order.lineItems.image.focalPoint.y" },
          { path: "orders.order.lineItems.physicalProperties.weight" },
          { path: "orders.order.billingInfo.address.geocode.latitude" },
          { path: "orders.order.billingInfo.address.geocode.longitude" },
          {
            path: "orders.order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
          },
          {
            path: "orders.order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
          },
          {
            path: "orders.order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
          },
          {
            path: "orders.order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
          },
          { path: "orders.order.recipientInfo.address.geocode.latitude" },
          { path: "orders.order.recipientInfo.address.geocode.longitude" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.BulkUpdateOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/bulk/orders/update",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.createdDate" },
            { path: "results.item.updatedDate" },
            { path: "results.item.purchasedDate" },
            { path: "results.item.lineItems.image.urlExpirationDate" },
            { path: "results.item.lineItems.digitalFile.expirationDate" },
            {
              path: "results.item.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "results.item.shippingInfo.logistics.deliverByDate" },
            {
              path: "results.item.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "results.item.shippingInfo.logistics.deliveryTimeSlot.to"
            },
            { path: "results.item.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.lineItems.image.focalPoint.x" },
            { path: "results.item.lineItems.image.focalPoint.y" },
            { path: "results.item.lineItems.physicalProperties.weight" },
            { path: "results.item.billingInfo.address.geocode.latitude" },
            { path: "results.item.billingInfo.address.geocode.longitude" },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "results.item.recipientInfo.address.geocode.latitude" },
            { path: "results.item.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateOrders;
}
function updateOrderLineItem(payload) {
  function __updateOrderLineItem({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "lineItem.image.urlExpirationDate" },
          { path: "lineItem.digitalFile.expirationDate" },
          { path: "lineItem.subscriptionInfo.subscriptionSettings.startDate" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "lineItem.image.focalPoint.x" },
          { path: "lineItem.image.focalPoint.y" },
          { path: "lineItem.physicalProperties.weight" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "PATCH",
      methodFqn: "com.wix.ecom.orders.v1.Orders.UpdateOrderLineItem",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}/line-item/{lineItem.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrderLineItem;
}
function addActivity(payload) {
  function __addActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.AddActivity",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}/activities",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addActivity;
}
function addActivities(payload) {
  function __addActivities({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "orderActivities.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.AddActivities",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{orderId}/activities/add",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __addActivities;
}
function updateActivity(payload) {
  function __updateActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "PATCH",
      methodFqn: "com.wix.ecom.orders.v1.Orders.UpdateActivity",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}/activities/{activityId}",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateActivity;
}
function deleteActivity(payload) {
  function __deleteActivity({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "DELETE",
      methodFqn: "com.wix.ecom.orders.v1.Orders.DeleteActivity",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}/activities/{activityId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteActivity;
}
function cancelOrder(payload) {
  function __cancelOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.CancelOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{id}/cancel",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelOrder;
}
function updateOrderStatus(payload) {
  function __updateOrderStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.UpdateOrderStatus",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/{orderId}/update-status",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "order.createdDate" },
            { path: "order.updatedDate" },
            { path: "order.purchasedDate" },
            { path: "order.lineItems.image.urlExpirationDate" },
            { path: "order.lineItems.digitalFile.expirationDate" },
            {
              path: "order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            { path: "order.shippingInfo.logistics.deliverByDate" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.from" },
            { path: "order.shippingInfo.logistics.deliveryTimeSlot.to" },
            { path: "order.activities.createdDate" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "order.lineItems.image.focalPoint.x" },
            { path: "order.lineItems.image.focalPoint.y" },
            { path: "order.lineItems.physicalProperties.weight" },
            { path: "order.billingInfo.address.geocode.latitude" },
            { path: "order.billingInfo.address.geocode.longitude" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "order.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            { path: "order.recipientInfo.address.geocode.latitude" },
            { path: "order.recipientInfo.address.geocode.longitude" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateOrderStatus;
}
function aggregateOrders(payload) {
  function __aggregateOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.AggregateOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/orders/aggregate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __aggregateOrders;
}
function bulkUpdateOrderTags(payload) {
  function __bulkUpdateOrderTags({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.v1.order",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.v1.Orders.BulkUpdateOrderTags",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersV1OrdersUrl({
        protoPath: "/v1/bulk/orders/update-tags",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkUpdateOrderTags;
}

// src/ecom-v1-order-orders.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var DescriptionLineType = /* @__PURE__ */ ((DescriptionLineType2) => {
  DescriptionLineType2["UNRECOGNISED"] = "UNRECOGNISED";
  DescriptionLineType2["PLAIN_TEXT"] = "PLAIN_TEXT";
  DescriptionLineType2["COLOR"] = "COLOR";
  return DescriptionLineType2;
})(DescriptionLineType || {});
var ItemTypePreset = /* @__PURE__ */ ((ItemTypePreset2) => {
  ItemTypePreset2["UNRECOGNISED"] = "UNRECOGNISED";
  ItemTypePreset2["PHYSICAL"] = "PHYSICAL";
  ItemTypePreset2["DIGITAL"] = "DIGITAL";
  ItemTypePreset2["GIFT_CARD"] = "GIFT_CARD";
  ItemTypePreset2["SERVICE"] = "SERVICE";
  return ItemTypePreset2;
})(ItemTypePreset || {});
var PaymentOptionType = /* @__PURE__ */ ((PaymentOptionType2) => {
  PaymentOptionType2["FULL_PAYMENT_ONLINE"] = "FULL_PAYMENT_ONLINE";
  PaymentOptionType2["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
  PaymentOptionType2["MEMBERSHIP"] = "MEMBERSHIP";
  PaymentOptionType2["DEPOSIT_ONLINE"] = "DEPOSIT_ONLINE";
  PaymentOptionType2["MEMBERSHIP_OFFLINE"] = "MEMBERSHIP_OFFLINE";
  PaymentOptionType2["MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER"] = "MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER";
  return PaymentOptionType2;
})(PaymentOptionType || {});
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});
var SubscriptionFrequency = /* @__PURE__ */ ((SubscriptionFrequency2) => {
  SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  SubscriptionFrequency2["DAY"] = "DAY";
  SubscriptionFrequency2["WEEK"] = "WEEK";
  SubscriptionFrequency2["MONTH"] = "MONTH";
  SubscriptionFrequency2["YEAR"] = "YEAR";
  return SubscriptionFrequency2;
})(SubscriptionFrequency || {});
var AdjustmentType = /* @__PURE__ */ ((AdjustmentType2) => {
  AdjustmentType2["UNKNOWN_ADJUSTMENT_TYPE"] = "UNKNOWN_ADJUSTMENT_TYPE";
  AdjustmentType2["EXTRA_CHARGE"] = "EXTRA_CHARGE";
  AdjustmentType2["CREDIT"] = "CREDIT";
  return AdjustmentType2;
})(AdjustmentType || {});
var TaxableAddressType = /* @__PURE__ */ ((TaxableAddressType2) => {
  TaxableAddressType2["UNKNOWN_TAXABLE_ADDRESS"] = "UNKNOWN_TAXABLE_ADDRESS";
  TaxableAddressType2["BUSINESS"] = "BUSINESS";
  TaxableAddressType2["BILLING"] = "BILLING";
  TaxableAddressType2["SHIPPING"] = "SHIPPING";
  return TaxableAddressType2;
})(TaxableAddressType || {});
var PaymentStatus = /* @__PURE__ */ ((PaymentStatus2) => {
  PaymentStatus2["UNSPECIFIED"] = "UNSPECIFIED";
  PaymentStatus2["NOT_PAID"] = "NOT_PAID";
  PaymentStatus2["PAID"] = "PAID";
  PaymentStatus2["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
  PaymentStatus2["FULLY_REFUNDED"] = "FULLY_REFUNDED";
  PaymentStatus2["PENDING"] = "PENDING";
  PaymentStatus2["PARTIALLY_PAID"] = "PARTIALLY_PAID";
  PaymentStatus2["PENDING_MERCHANT"] = "PENDING_MERCHANT";
  PaymentStatus2["CANCELED"] = "CANCELED";
  PaymentStatus2["DECLINED"] = "DECLINED";
  return PaymentStatus2;
})(PaymentStatus || {});
var FulfillmentStatus = /* @__PURE__ */ ((FulfillmentStatus2) => {
  FulfillmentStatus2["NOT_FULFILLED"] = "NOT_FULFILLED";
  FulfillmentStatus2["FULFILLED"] = "FULFILLED";
  FulfillmentStatus2["PARTIALLY_FULFILLED"] = "PARTIALLY_FULFILLED";
  return FulfillmentStatus2;
})(FulfillmentStatus || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var OrderStatus = /* @__PURE__ */ ((OrderStatus2) => {
  OrderStatus2["INITIALIZED"] = "INITIALIZED";
  OrderStatus2["APPROVED"] = "APPROVED";
  OrderStatus2["CANCELED"] = "CANCELED";
  OrderStatus2["PENDING"] = "PENDING";
  OrderStatus2["REJECTED"] = "REJECTED";
  return OrderStatus2;
})(OrderStatus || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var DiscountReason = /* @__PURE__ */ ((DiscountReason2) => {
  DiscountReason2["UNSPECIFIED"] = "UNSPECIFIED";
  DiscountReason2["EXCHANGED_ITEMS"] = "EXCHANGED_ITEMS";
  DiscountReason2["BILLING_ADJUSTMENT"] = "BILLING_ADJUSTMENT";
  return DiscountReason2;
})(DiscountReason || {});
var LineItemQuantityChangeType = /* @__PURE__ */ ((LineItemQuantityChangeType2) => {
  LineItemQuantityChangeType2["QUANTITY_INCREASED"] = "QUANTITY_INCREASED";
  LineItemQuantityChangeType2["QUANTITY_DECREASED"] = "QUANTITY_DECREASED";
  return LineItemQuantityChangeType2;
})(LineItemQuantityChangeType || {});
var ActivityType = /* @__PURE__ */ ((ActivityType2) => {
  ActivityType2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  ActivityType2["ORDER_PLACED"] = "ORDER_PLACED";
  ActivityType2["ORDER_PAID"] = "ORDER_PAID";
  ActivityType2["ORDER_FULFILLED"] = "ORDER_FULFILLED";
  ActivityType2["ORDER_NOT_FULFILLED"] = "ORDER_NOT_FULFILLED";
  ActivityType2["ORDER_CANCELED"] = "ORDER_CANCELED";
  ActivityType2["DOWNLOAD_LINK_SENT"] = "DOWNLOAD_LINK_SENT";
  ActivityType2["TRACKING_NUMBER_ADDED"] = "TRACKING_NUMBER_ADDED";
  ActivityType2["TRACKING_NUMBER_EDITED"] = "TRACKING_NUMBER_EDITED";
  ActivityType2["TRACKING_LINK_ADDED"] = "TRACKING_LINK_ADDED";
  ActivityType2["SHIPPING_CONFIRMATION_EMAIL_SENT"] = "SHIPPING_CONFIRMATION_EMAIL_SENT";
  ActivityType2["INVOICE_ADDED"] = "INVOICE_ADDED";
  ActivityType2["INVOICE_REMOVED"] = "INVOICE_REMOVED";
  ActivityType2["INVOICE_SENT"] = "INVOICE_SENT";
  ActivityType2["FULFILLER_EMAIL_SENT"] = "FULFILLER_EMAIL_SENT";
  ActivityType2["SHIPPING_ADDRESS_EDITED"] = "SHIPPING_ADDRESS_EDITED";
  ActivityType2["EMAIL_EDITED"] = "EMAIL_EDITED";
  ActivityType2["PICKUP_READY_EMAIL_SENT"] = "PICKUP_READY_EMAIL_SENT";
  ActivityType2["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
  ActivityType2["MERCHANT_COMMENT"] = "MERCHANT_COMMENT";
  ActivityType2["ORDER_CREATED_FROM_EXCHANGE"] = "ORDER_CREATED_FROM_EXCHANGE";
  ActivityType2["NEW_EXCHANGE_ORDER_CREATED"] = "NEW_EXCHANGE_ORDER_CREATED";
  ActivityType2["ORDER_PARTIALLY_PAID"] = "ORDER_PARTIALLY_PAID";
  ActivityType2["DRAFT_ORDER_CHANGES_APPLIED"] = "DRAFT_ORDER_CHANGES_APPLIED";
  ActivityType2["SAVED_PAYMENT_METHOD"] = "SAVED_PAYMENT_METHOD";
  ActivityType2["AUTHORIZED_PAYMENT_CREATED"] = "AUTHORIZED_PAYMENT_CREATED";
  ActivityType2["AUTHORIZED_PAYMENT_CAPTURED"] = "AUTHORIZED_PAYMENT_CAPTURED";
  ActivityType2["AUTHORIZED_PAYMENT_VOIDED"] = "AUTHORIZED_PAYMENT_VOIDED";
  ActivityType2["REFUND_INITIATED"] = "REFUND_INITIATED";
  ActivityType2["PAYMENT_REFUNDED"] = "PAYMENT_REFUNDED";
  ActivityType2["PAYMENT_REFUND_FAILED"] = "PAYMENT_REFUND_FAILED";
  ActivityType2["REFUNDED_AS_STORE_CREDIT"] = "REFUNDED_AS_STORE_CREDIT";
  ActivityType2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  ActivityType2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  ActivityType2["PAYMENT_DECLINED"] = "PAYMENT_DECLINED";
  ActivityType2["ORDER_PENDING"] = "ORDER_PENDING";
  ActivityType2["ORDER_REJECTED"] = "ORDER_REJECTED";
  ActivityType2["RECEIPT_CREATED"] = "RECEIPT_CREATED";
  ActivityType2["RECEIPT_SENT"] = "RECEIPT_SENT";
  ActivityType2["CHARGEBACK_CREATED"] = "CHARGEBACK_CREATED";
  ActivityType2["CHARGEBACK_REVERSED"] = "CHARGEBACK_REVERSED";
  return ActivityType2;
})(ActivityType || {});
var OrderActivityTypeEnumActivityType = /* @__PURE__ */ ((OrderActivityTypeEnumActivityType2) => {
  OrderActivityTypeEnumActivityType2["UNKNOWN_ACTIVITY_TYPE"] = "UNKNOWN_ACTIVITY_TYPE";
  OrderActivityTypeEnumActivityType2["ORDER_PLACED"] = "ORDER_PLACED";
  OrderActivityTypeEnumActivityType2["ORDER_PAID"] = "ORDER_PAID";
  OrderActivityTypeEnumActivityType2["ORDER_FULFILLED"] = "ORDER_FULFILLED";
  OrderActivityTypeEnumActivityType2["ORDER_NOT_FULFILLED"] = "ORDER_NOT_FULFILLED";
  OrderActivityTypeEnumActivityType2["ORDER_CANCELED"] = "ORDER_CANCELED";
  OrderActivityTypeEnumActivityType2["DOWNLOAD_LINK_SENT"] = "DOWNLOAD_LINK_SENT";
  OrderActivityTypeEnumActivityType2["TRACKING_NUMBER_ADDED"] = "TRACKING_NUMBER_ADDED";
  OrderActivityTypeEnumActivityType2["TRACKING_NUMBER_EDITED"] = "TRACKING_NUMBER_EDITED";
  OrderActivityTypeEnumActivityType2["TRACKING_LINK_ADDED"] = "TRACKING_LINK_ADDED";
  OrderActivityTypeEnumActivityType2["SHIPPING_CONFIRMATION_EMAIL_SENT"] = "SHIPPING_CONFIRMATION_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["INVOICE_ADDED"] = "INVOICE_ADDED";
  OrderActivityTypeEnumActivityType2["INVOICE_REMOVED"] = "INVOICE_REMOVED";
  OrderActivityTypeEnumActivityType2["INVOICE_SENT"] = "INVOICE_SENT";
  OrderActivityTypeEnumActivityType2["FULFILLER_EMAIL_SENT"] = "FULFILLER_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["SHIPPING_ADDRESS_EDITED"] = "SHIPPING_ADDRESS_EDITED";
  OrderActivityTypeEnumActivityType2["EMAIL_EDITED"] = "EMAIL_EDITED";
  OrderActivityTypeEnumActivityType2["PICKUP_READY_EMAIL_SENT"] = "PICKUP_READY_EMAIL_SENT";
  OrderActivityTypeEnumActivityType2["CUSTOM_ACTIVITY"] = "CUSTOM_ACTIVITY";
  OrderActivityTypeEnumActivityType2["MERCHANT_COMMENT"] = "MERCHANT_COMMENT";
  OrderActivityTypeEnumActivityType2["ORDER_CREATED_FROM_EXCHANGE"] = "ORDER_CREATED_FROM_EXCHANGE";
  OrderActivityTypeEnumActivityType2["NEW_EXCHANGE_ORDER_CREATED"] = "NEW_EXCHANGE_ORDER_CREATED";
  OrderActivityTypeEnumActivityType2["ORDER_PARTIALLY_PAID"] = "ORDER_PARTIALLY_PAID";
  OrderActivityTypeEnumActivityType2["DRAFT_ORDER_CHANGES_APPLIED"] = "DRAFT_ORDER_CHANGES_APPLIED";
  OrderActivityTypeEnumActivityType2["SAVED_PAYMENT_METHOD"] = "SAVED_PAYMENT_METHOD";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_CREATED"] = "AUTHORIZED_PAYMENT_CREATED";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_CAPTURED"] = "AUTHORIZED_PAYMENT_CAPTURED";
  OrderActivityTypeEnumActivityType2["AUTHORIZED_PAYMENT_VOIDED"] = "AUTHORIZED_PAYMENT_VOIDED";
  OrderActivityTypeEnumActivityType2["REFUND_INITIATED"] = "REFUND_INITIATED";
  OrderActivityTypeEnumActivityType2["PAYMENT_REFUNDED"] = "PAYMENT_REFUNDED";
  OrderActivityTypeEnumActivityType2["PAYMENT_REFUND_FAILED"] = "PAYMENT_REFUND_FAILED";
  OrderActivityTypeEnumActivityType2["REFUNDED_AS_STORE_CREDIT"] = "REFUNDED_AS_STORE_CREDIT";
  OrderActivityTypeEnumActivityType2["PAYMENT_PENDING"] = "PAYMENT_PENDING";
  OrderActivityTypeEnumActivityType2["PAYMENT_CANCELED"] = "PAYMENT_CANCELED";
  OrderActivityTypeEnumActivityType2["PAYMENT_DECLINED"] = "PAYMENT_DECLINED";
  OrderActivityTypeEnumActivityType2["ORDER_PENDING"] = "ORDER_PENDING";
  OrderActivityTypeEnumActivityType2["ORDER_REJECTED"] = "ORDER_REJECTED";
  OrderActivityTypeEnumActivityType2["RECEIPT_CREATED"] = "RECEIPT_CREATED";
  OrderActivityTypeEnumActivityType2["RECEIPT_SENT"] = "RECEIPT_SENT";
  OrderActivityTypeEnumActivityType2["CHARGEBACK_CREATED"] = "CHARGEBACK_CREATED";
  OrderActivityTypeEnumActivityType2["CHARGEBACK_REVERSED"] = "CHARGEBACK_REVERSED";
  OrderActivityTypeEnumActivityType2["ORDER_REFUNDED"] = "ORDER_REFUNDED";
  return OrderActivityTypeEnumActivityType2;
})(OrderActivityTypeEnumActivityType || {});
var AttributionSource = /* @__PURE__ */ ((AttributionSource2) => {
  AttributionSource2["UNSPECIFIED"] = "UNSPECIFIED";
  AttributionSource2["FACEBOOK_ADS"] = "FACEBOOK_ADS";
  return AttributionSource2;
})(AttributionSource || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["UNSPECIFIED"] = "UNSPECIFIED";
  ChannelType2["WEB"] = "WEB";
  ChannelType2["POS"] = "POS";
  ChannelType2["EBAY"] = "EBAY";
  ChannelType2["AMAZON"] = "AMAZON";
  ChannelType2["OTHER_PLATFORM"] = "OTHER_PLATFORM";
  ChannelType2["WIX_APP_STORE"] = "WIX_APP_STORE";
  ChannelType2["WIX_INVOICES"] = "WIX_INVOICES";
  ChannelType2["BACKOFFICE_MERCHANT"] = "BACKOFFICE_MERCHANT";
  ChannelType2["WISH"] = "WISH";
  ChannelType2["CLASS_PASS"] = "CLASS_PASS";
  ChannelType2["GLOBAL_E"] = "GLOBAL_E";
  ChannelType2["FACEBOOK"] = "FACEBOOK";
  ChannelType2["ETSY"] = "ETSY";
  ChannelType2["TIKTOK"] = "TIKTOK";
  ChannelType2["FAIRE_COM"] = "FAIRE_COM";
  ChannelType2["PAYPAL_AGENTIC_CHECKOUT"] = "PAYPAL_AGENTIC_CHECKOUT";
  return ChannelType2;
})(ChannelType || {});
var AdditionalFeeSource = /* @__PURE__ */ ((AdditionalFeeSource2) => {
  AdditionalFeeSource2["UNKNOWN_ADDITIONAL_FEE_SOURCE"] = "UNKNOWN_ADDITIONAL_FEE_SOURCE";
  AdditionalFeeSource2["SERVICE_PLUGIN"] = "SERVICE_PLUGIN";
  AdditionalFeeSource2["ITEM"] = "ITEM";
  AdditionalFeeSource2["MANUAL"] = "MANUAL";
  AdditionalFeeSource2["SHIPPING"] = "SHIPPING";
  AdditionalFeeSource2["PLATFORM"] = "PLATFORM";
  return AdditionalFeeSource2;
})(AdditionalFeeSource || {});
var OrderActionType = /* @__PURE__ */ ((OrderActionType2) => {
  OrderActionType2["EDIT"] = "EDIT";
  OrderActionType2["UPDATE"] = "UPDATE";
  OrderActionType2["REFUND"] = "REFUND";
  OrderActionType2["APPROVE"] = "APPROVE";
  OrderActionType2["CANCEL"] = "CANCEL";
  OrderActionType2["FULFILL"] = "FULFILL";
  OrderActionType2["COLLECT_PAYMENTS"] = "COLLECT_PAYMENTS";
  OrderActionType2["SEND_NOTIFICATIONS"] = "SEND_NOTIFICATIONS";
  OrderActionType2["ADD_MERCHANT_ACTIVITY"] = "ADD_MERCHANT_ACTIVITY";
  OrderActionType2["ADD_ACTIVITY"] = "ADD_ACTIVITY";
  OrderActionType2["CREATE_INVOICE"] = "CREATE_INVOICE";
  OrderActionType2["CREATE_RECEIPTS"] = "CREATE_RECEIPTS";
  OrderActionType2["ASSIGN_TAG"] = "ASSIGN_TAG";
  OrderActionType2["PRINT_PACKAGING_SLIP"] = "PRINT_PACKAGING_SLIP";
  OrderActionType2["MODIFY_INVENTORY"] = "MODIFY_INVENTORY";
  return OrderActionType2;
})(OrderActionType || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["UNKNOWN_CHARGE_TYPE"] = "UNKNOWN_CHARGE_TYPE";
  ChargeType2["PASS_ON"] = "PASS_ON";
  ChargeType2["ABSORBED"] = "ABSORBED";
  return ChargeType2;
})(ChargeType || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var PreviewEmailType = /* @__PURE__ */ ((PreviewEmailType2) => {
  PreviewEmailType2["ORDER_PLACED"] = "ORDER_PLACED";
  PreviewEmailType2["DOWNLOAD_LINKS"] = "DOWNLOAD_LINKS";
  PreviewEmailType2["ORDER_SHIPPED"] = "ORDER_SHIPPED";
  PreviewEmailType2["ORDER_READY_FOR_PICKUP"] = "ORDER_READY_FOR_PICKUP";
  return PreviewEmailType2;
})(PreviewEmailType || {});
var ScheduledAction = /* @__PURE__ */ ((ScheduledAction2) => {
  ScheduledAction2["UNSPECIFIED"] = "UNSPECIFIED";
  ScheduledAction2["VOID"] = "VOID";
  ScheduledAction2["CAPTURE"] = "CAPTURE";
  return ScheduledAction2;
})(ScheduledAction || {});
var DurationUnit = /* @__PURE__ */ ((DurationUnit2) => {
  DurationUnit2["UNKNOWN_DURATION_UNIT"] = "UNKNOWN_DURATION_UNIT";
  DurationUnit2["MINUTES"] = "MINUTES";
  DurationUnit2["HOURS"] = "HOURS";
  DurationUnit2["DAYS"] = "DAYS";
  return DurationUnit2;
})(DurationUnit || {});
var PaymentCollectabilityStatus = /* @__PURE__ */ ((PaymentCollectabilityStatus2) => {
  PaymentCollectabilityStatus2["UNKNOWN"] = "UNKNOWN";
  PaymentCollectabilityStatus2["COLLECTABLE"] = "COLLECTABLE";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_IS_CANCELLED"] = "NONCOLLECTABLE_ORDER_IS_CANCELLED";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_IS_PAID"] = "NONCOLLECTABLE_ORDER_IS_PAID";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_MISSING_PAYMENT_METHOD"] = "NONCOLLECTABLE_MISSING_PAYMENT_METHOD";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_IS_PENDING"] = "NONCOLLECTABLE_ORDER_IS_PENDING";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_IS_REJECTED"] = "NONCOLLECTABLE_ORDER_IS_REJECTED";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_HAS_SUBSCRIPTIONS"] = "NONCOLLECTABLE_ORDER_HAS_SUBSCRIPTIONS";
  PaymentCollectabilityStatus2["NONCOLLECTABLE_ORDER_HAS_CHARGEBACKS"] = "NONCOLLECTABLE_ORDER_HAS_CHARGEBACKS";
  return PaymentCollectabilityStatus2;
})(PaymentCollectabilityStatus || {});
var RefundableStatus = /* @__PURE__ */ ((RefundableStatus2) => {
  RefundableStatus2["NOT_REFUNDABLE"] = "NOT_REFUNDABLE";
  RefundableStatus2["MANUAL"] = "MANUAL";
  RefundableStatus2["REFUNDABLE"] = "REFUNDABLE";
  return RefundableStatus2;
})(RefundableStatus || {});
var NonRefundableReason = /* @__PURE__ */ ((NonRefundableReason2) => {
  NonRefundableReason2["NONE"] = "NONE";
  NonRefundableReason2["ALREADY_REFUNDED"] = "ALREADY_REFUNDED";
  NonRefundableReason2["PROVIDER_IS_DOWN"] = "PROVIDER_IS_DOWN";
  NonRefundableReason2["INTERNAL_ERROR"] = "INTERNAL_ERROR";
  NonRefundableReason2["NOT_PAID"] = "NOT_PAID";
  NonRefundableReason2["ACCESS_DENIED"] = "ACCESS_DENIED";
  NonRefundableReason2["ZERO_PRICE"] = "ZERO_PRICE";
  NonRefundableReason2["DISABLED_BY_PROVIDER"] = "DISABLED_BY_PROVIDER";
  NonRefundableReason2["PENDING_REFUND"] = "PENDING_REFUND";
  NonRefundableReason2["FORBIDDEN"] = "FORBIDDEN";
  NonRefundableReason2["TRANSACTION_NOT_FOUND"] = "TRANSACTION_NOT_FOUND";
  NonRefundableReason2["ORDER_IS_PENDING"] = "ORDER_IS_PENDING";
  NonRefundableReason2["ORDER_IS_REJECTED"] = "ORDER_IS_REJECTED";
  return NonRefundableReason2;
})(NonRefundableReason || {});
var ManuallyRefundableReason = /* @__PURE__ */ ((ManuallyRefundableReason2) => {
  ManuallyRefundableReason2["EXPIRED"] = "EXPIRED";
  ManuallyRefundableReason2["NOT_SUPPORTED"] = "NOT_SUPPORTED";
  ManuallyRefundableReason2["OFFLINE"] = "OFFLINE";
  ManuallyRefundableReason2["REQUIRES_CARD_READER"] = "REQUIRES_CARD_READER";
  return ManuallyRefundableReason2;
})(ManuallyRefundableReason || {});
var RestockType = /* @__PURE__ */ ((RestockType2) => {
  RestockType2["NO_ITEMS"] = "NO_ITEMS";
  RestockType2["ALL_ITEMS"] = "ALL_ITEMS";
  RestockType2["SOME_ITEMS"] = "SOME_ITEMS";
  return RestockType2;
})(RestockType || {});
var TransactionStatus = /* @__PURE__ */ ((TransactionStatus2) => {
  TransactionStatus2["UNDEFINED"] = "UNDEFINED";
  TransactionStatus2["APPROVED"] = "APPROVED";
  TransactionStatus2["PENDING"] = "PENDING";
  TransactionStatus2["PENDING_MERCHANT"] = "PENDING_MERCHANT";
  TransactionStatus2["CANCELED"] = "CANCELED";
  TransactionStatus2["DECLINED"] = "DECLINED";
  TransactionStatus2["REFUNDED"] = "REFUNDED";
  TransactionStatus2["PARTIALLY_REFUNDED"] = "PARTIALLY_REFUNDED";
  TransactionStatus2["AUTHORIZED"] = "AUTHORIZED";
  TransactionStatus2["VOIDED"] = "VOIDED";
  return TransactionStatus2;
})(TransactionStatus || {});
var AuthorizationCaptureStatus = /* @__PURE__ */ ((AuthorizationCaptureStatus2) => {
  AuthorizationCaptureStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  AuthorizationCaptureStatus2["PENDING"] = "PENDING";
  AuthorizationCaptureStatus2["SUCCEEDED"] = "SUCCEEDED";
  AuthorizationCaptureStatus2["FAILED"] = "FAILED";
  return AuthorizationCaptureStatus2;
})(AuthorizationCaptureStatus || {});
var AuthorizationVoidStatus = /* @__PURE__ */ ((AuthorizationVoidStatus2) => {
  AuthorizationVoidStatus2["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
  AuthorizationVoidStatus2["PENDING"] = "PENDING";
  AuthorizationVoidStatus2["SUCCEEDED"] = "SUCCEEDED";
  AuthorizationVoidStatus2["FAILED"] = "FAILED";
  return AuthorizationVoidStatus2;
})(AuthorizationVoidStatus || {});
var Reason = /* @__PURE__ */ ((Reason2) => {
  Reason2["UNKNOWN_REASON"] = "UNKNOWN_REASON";
  Reason2["MANUAL"] = "MANUAL";
  Reason2["SCHEDULED"] = "SCHEDULED";
  return Reason2;
})(Reason || {});
var ActionType = /* @__PURE__ */ ((ActionType2) => {
  ActionType2["UNKNOWN_ACTION_TYPE"] = "UNKNOWN_ACTION_TYPE";
  ActionType2["VOID"] = "VOID";
  ActionType2["CAPTURE"] = "CAPTURE";
  return ActionType2;
})(ActionType || {});
var ChargebackStatus = /* @__PURE__ */ ((ChargebackStatus2) => {
  ChargebackStatus2["UNSPECIFIED"] = "UNSPECIFIED";
  ChargebackStatus2["APPROVED"] = "APPROVED";
  ChargebackStatus2["REVERSED"] = "REVERSED";
  return ChargebackStatus2;
})(ChargebackStatus || {});
var MembershipPaymentStatus = /* @__PURE__ */ ((MembershipPaymentStatus2) => {
  MembershipPaymentStatus2["CHARGED"] = "CHARGED";
  MembershipPaymentStatus2["CHARGE_FAILED"] = "CHARGE_FAILED";
  return MembershipPaymentStatus2;
})(MembershipPaymentStatus || {});
var RefundStatus = /* @__PURE__ */ ((RefundStatus2) => {
  RefundStatus2["PENDING"] = "PENDING";
  RefundStatus2["SUCCEEDED"] = "SUCCEEDED";
  RefundStatus2["FAILED"] = "FAILED";
  RefundStatus2["SCHEDULED"] = "SCHEDULED";
  RefundStatus2["STARTED"] = "STARTED";
  return RefundStatus2;
})(RefundStatus || {});
var VersioningMode = /* @__PURE__ */ ((VersioningMode2) => {
  VersioningMode2["DEFAULT"] = "DEFAULT";
  VersioningMode2["GREATER_THAN"] = "GREATER_THAN";
  VersioningMode2["GREATER_OR_EQUAL"] = "GREATER_OR_EQUAL";
  return VersioningMode2;
})(VersioningMode || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var OrderApprovalStrategy = /* @__PURE__ */ ((OrderApprovalStrategy2) => {
  OrderApprovalStrategy2["DEFAULT"] = "DEFAULT";
  OrderApprovalStrategy2["PAYMENT_RECEIVED"] = "PAYMENT_RECEIVED";
  OrderApprovalStrategy2["PAYMENT_METHOD_SAVED"] = "PAYMENT_METHOD_SAVED";
  return OrderApprovalStrategy2;
})(OrderApprovalStrategy || {});
var DeltaPaymentOptionType = /* @__PURE__ */ ((DeltaPaymentOptionType2) => {
  DeltaPaymentOptionType2["UNKNOWN_PAYMENT_OPTION"] = "UNKNOWN_PAYMENT_OPTION";
  DeltaPaymentOptionType2["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
  DeltaPaymentOptionType2["MEMBERSHIP_OFFLINE"] = "MEMBERSHIP_OFFLINE";
  return DeltaPaymentOptionType2;
})(DeltaPaymentOptionType || {});
var InventoryAction = /* @__PURE__ */ ((InventoryAction2) => {
  InventoryAction2["RESTOCK"] = "RESTOCK";
  InventoryAction2["DECREASE"] = "DECREASE";
  return InventoryAction2;
})(InventoryAction || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["Unknown"] = "Unknown";
  Placement2["Replace"] = "Replace";
  Placement2["Before"] = "Before";
  Placement2["After"] = "After";
  return Placement2;
})(Placement || {});
var SubdivisionType = /* @__PURE__ */ ((SubdivisionType2) => {
  SubdivisionType2["UNKNOWN_SUBDIVISION_TYPE"] = "UNKNOWN_SUBDIVISION_TYPE";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_1"] = "ADMINISTRATIVE_AREA_LEVEL_1";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_2"] = "ADMINISTRATIVE_AREA_LEVEL_2";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_3"] = "ADMINISTRATIVE_AREA_LEVEL_3";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_4"] = "ADMINISTRATIVE_AREA_LEVEL_4";
  SubdivisionType2["ADMINISTRATIVE_AREA_LEVEL_5"] = "ADMINISTRATIVE_AREA_LEVEL_5";
  SubdivisionType2["COUNTRY"] = "COUNTRY";
  return SubdivisionType2;
})(SubdivisionType || {});
var SourceType = /* @__PURE__ */ ((SourceType2) => {
  SourceType2["UNKNOWN_SOURCE_TYPE"] = "UNKNOWN_SOURCE_TYPE";
  SourceType2["WIX_APP"] = "WIX_APP";
  SourceType2["EXTERNAL"] = "EXTERNAL";
  SourceType2["ADMIN"] = "ADMIN";
  SourceType2["OTHER"] = "OTHER";
  return SourceType2;
})(SourceType || {});
var CustomFieldGroup = /* @__PURE__ */ ((CustomFieldGroup2) => {
  CustomFieldGroup2["UNKNOWN_CUSTOM_FIELD_GROUP"] = "UNKNOWN_CUSTOM_FIELD_GROUP";
  CustomFieldGroup2["BUSINESS_DETAILS"] = "BUSINESS_DETAILS";
  CustomFieldGroup2["CUSTOMER_DETAILS"] = "CUSTOMER_DETAILS";
  CustomFieldGroup2["DOCUMENT"] = "DOCUMENT";
  CustomFieldGroup2["FOOTER"] = "FOOTER";
  CustomFieldGroup2["OTHER"] = "OTHER";
  return CustomFieldGroup2;
})(CustomFieldGroup || {});
var ValueType = /* @__PURE__ */ ((ValueType2) => {
  ValueType2["UNKNOWN_VALUE_TYPE"] = "UNKNOWN_VALUE_TYPE";
  ValueType2["STRING"] = "STRING";
  ValueType2["DATE"] = "DATE";
  ValueType2["BOOLEAN"] = "BOOLEAN";
  ValueType2["NUMBER"] = "NUMBER";
  return ValueType2;
})(ValueType || {});
var DepositType = /* @__PURE__ */ ((DepositType2) => {
  DepositType2["UNKNOWN"] = "UNKNOWN";
  DepositType2["FLAT"] = "FLAT";
  DepositType2["PERCENTAGE"] = "PERCENTAGE";
  return DepositType2;
})(DepositType || {});
var InvoiceStatus = /* @__PURE__ */ ((InvoiceStatus2) => {
  InvoiceStatus2["Draft"] = "Draft";
  InvoiceStatus2["Sent"] = "Sent";
  InvoiceStatus2["Processing"] = "Processing";
  InvoiceStatus2["Paid"] = "Paid";
  InvoiceStatus2["Overdue"] = "Overdue";
  InvoiceStatus2["Void"] = "Void";
  InvoiceStatus2["Deleted"] = "Deleted";
  InvoiceStatus2["PartiallyPaid"] = "PartiallyPaid";
  InvoiceStatus2["PartialAndOverdue"] = "PartialAndOverdue";
  return InvoiceStatus2;
})(InvoiceStatus || {});
async function importOrder2(order) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ order }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "order.lineItems.image" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [
          { path: "order.billingInfo.address" },
          { path: "order.shippingInfo.logistics.shippingDestination.address" },
          { path: "order.shippingInfo.logistics.pickupDetails.address" },
          { path: "order.recipientInfo.address" }
        ]
      }
    ]
  );
  const reqOpts = importOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { order: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["order"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setOrderNumberCounter2(counter, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    counter,
    allowLowerCounter: options?.allowLowerCounter
  });
  const reqOpts = setOrderNumberCounter(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          counter: "$[0]",
          allowLowerCounter: "$[1].allowLowerCounter"
        },
        singleArgumentUnchanged: false
      },
      ["counter", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteImportedOrders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderIds: options?.orderIds
  });
  const reqOpts = bulkDeleteImportedOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { orderIds: "$[0].orderIds" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function preparePaymentCollection2(ecomOrderId, amount, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    amount,
    paymentGatewayOrderId: options?.paymentGatewayOrderId,
    delayedCapture: options?.delayedCapture,
    merchantInitiated: options?.merchantInitiated,
    redirectUrls: options?.redirectUrls,
    delayedCaptureSettings: options?.delayedCaptureSettings
  });
  const reqOpts = preparePaymentCollection(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ecomOrderId: "$[0]",
          amount: "$[1]",
          paymentGatewayOrderId: "$[2].paymentGatewayOrderId",
          delayedCapture: "$[2].delayedCapture",
          merchantInitiated: "$[2].merchantInitiated",
          redirectUrls: "$[2].redirectUrls",
          delayedCaptureSettings: "$[2].delayedCaptureSettings"
        },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "amount", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getPaymentCollectabilityStatus2(ecomOrderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId
  });
  const reqOpts = getPaymentCollectabilityStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function recordManuallyCollectedPayment2(orderId, amount) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderId,
    amount
  });
  const reqOpts = recordManuallyCollectedPayment(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { orderId: "$[0]", amount: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["orderId", "amount"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function paymentCollectionMarkOrderAsPaid2(ecomOrderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId
  });
  const reqOpts = paymentCollectionMarkOrderAsPaid(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function paymentCollectionBulkMarkOrdersAsPaid2(ecomOrderIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderIds
  });
  const reqOpts = paymentCollectionBulkMarkOrdersAsPaid(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "results.item.billingInfo.address" },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address"
            },
            { path: "results.item.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRefundabilityStatus2(ecomOrderId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId
  });
  const reqOpts = getRefundabilityStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function paymentCollectionCreatePaymentGatewayOrder2(ecomOrderId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    chargedBy: options?.chargedBy
  });
  const reqOpts = paymentCollectionCreatePaymentGatewayOrder(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ecomOrderId: "$[0]",
          chargedBy: "$[1].chargedBy"
        },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function chargeMemberships2(ecomOrderId, memberId, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    memberId,
    membershipCharges: options?.membershipCharges
  });
  const reqOpts = chargeMemberships(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ecomOrderId: "$[0]",
          memberId: "$[1]",
          membershipCharges: "$[2].membershipCharges"
        },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function triggerRefund2(ecomOrderId, payments, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    payments,
    details: options?.details,
    sideEffects: options?.sideEffects
  });
  const reqOpts = triggerRefund(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          ecomOrderId: "$[0]",
          payments: "$[1]",
          details: "$[2].details",
          sideEffects: "$[2].sideEffects"
        },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "payments", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function voidAuthorizedPayments2(ecomOrderId, paymentIds) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    paymentIds
  });
  const reqOpts = voidAuthorizedPayments(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderId: "$[0]", paymentIds: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "paymentIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function captureAuthorizedPayments2(ecomOrderId, payments) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    ecomOrderId,
    payments
  });
  const reqOpts = captureAuthorizedPayments(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { ecomOrderId: "$[0]", payments: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["ecomOrderId", "payments"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getOrder2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = getOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    )?.order;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function searchOrders2(search) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ search });
  const reqOpts = searchOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "orders.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "orders.billingInfo.address" },
            {
              path: "orders.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "orders.shippingInfo.logistics.pickupDetails.address" },
            { path: "orders.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { search: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["search"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function createOrder2(order, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      order,
      settings: options?.settings
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "order.lineItems.image" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [
          { path: "order.billingInfo.address" },
          { path: "order.shippingInfo.logistics.shippingDestination.address" },
          { path: "order.shippingInfo.logistics.pickupDetails.address" },
          { path: "order.recipientInfo.address" }
        ]
      }
    ]
  );
  const reqOpts = createOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    )?.order;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { order: "$[0]", settings: "$[1].settings" },
        singleArgumentUnchanged: false
      },
      ["order", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrder2(_id, order) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ order: { ...order, id: _id } }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "order.lineItems.image" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [
          { path: "order.billingInfo.address" },
          { path: "order.shippingInfo.logistics.shippingDestination.address" },
          { path: "order.shippingInfo.logistics.pickupDetails.address" },
          { path: "order.recipientInfo.address" }
        ]
      }
    ]
  );
  const reqOpts = updateOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    )?.order;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { order: "$[1]" },
        explicitPathsToArguments: { "order.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "order"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOrders2(orders, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      orders,
      returnEntity: options?.returnEntity
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "orders.order.lineItems.image" }]
      },
      {
        transformFn: import_address.transformSDKAddressToRESTAddress,
        paths: [
          { path: "orders.order.billingInfo.address" },
          {
            path: "orders.order.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "orders.order.shippingInfo.logistics.pickupDetails.address" },
          { path: "orders.order.recipientInfo.address" }
        ]
      }
    ]
  );
  const reqOpts = bulkUpdateOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "results.item.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "results.item.billingInfo.address" },
            {
              path: "results.item.shippingInfo.logistics.shippingDestination.address"
            },
            {
              path: "results.item.shippingInfo.logistics.pickupDetails.address"
            },
            { path: "results.item.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orders: "$[0]",
          returnEntity: "$[1].returnEntity"
        },
        singleArgumentUnchanged: false
      },
      ["orders", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrderLineItem2(identifiers, lineItem) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
      id: identifiers?._id,
      lineItem: { ...lineItem, id: identifiers?.lineItemId }
    }),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [{ path: "lineItem.image" }]
      }
    ]
  );
  const reqOpts = updateOrderLineItem(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { lineItem: "$[1]" },
        explicitPathsToArguments: {
          id: "$[0]._id",
          "lineItem.id": "$[0].lineItemId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "lineItem"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addActivity2(_id, activity) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    activity
  });
  const reqOpts = addActivity(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]", activity: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["_id", "activity"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function addActivities2(orderId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderId,
    orderActivities: options?.orderActivities
  });
  const reqOpts = addActivities(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orderId: "$[0]",
          orderActivities: "$[1].orderActivities"
        },
        singleArgumentUnchanged: false
      },
      ["orderId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateActivity2(identifiers, activity) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: identifiers?._id,
    activityId: identifiers?.activityId,
    activity
  });
  const reqOpts = updateActivity(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]._id",
          activityId: "$[0].activityId",
          activity: "$[1]"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers", "activity"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteActivity2(identifiers) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: identifiers?._id,
    activityId: identifiers?.activityId
  });
  const reqOpts = deleteActivity(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]._id",
          activityId: "$[0].activityId"
        },
        singleArgumentUnchanged: false
      },
      ["identifiers"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelOrder2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    sendOrderCanceledEmail: options?.sendOrderCanceledEmail,
    customMessage: options?.customMessage,
    restockAllItems: options?.restockAllItems
  });
  const reqOpts = cancelOrder(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          sendOrderCanceledEmail: "$[1].sendOrderCanceledEmail",
          customMessage: "$[1].customMessage",
          restockAllItems: "$[1].restockAllItems"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateOrderStatus2(orderId, status) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderId,
    status
  });
  const reqOpts = updateOrderStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [{ path: "order.lineItems.image" }]
        },
        {
          transformFn: import_address2.transformRESTAddressToSDKAddress,
          paths: [
            { path: "order.billingInfo.address" },
            {
              path: "order.shippingInfo.logistics.shippingDestination.address"
            },
            { path: "order.shippingInfo.logistics.pickupDetails.address" },
            { path: "order.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { orderId: "$[0]", status: "$[1]" },
        singleArgumentUnchanged: false
      },
      ["orderId", "status"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function aggregateOrders2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    filter: options?.filter,
    aggregation: options?.aggregation,
    hierarchySeparatorOverride: options?.hierarchySeparatorOverride
  });
  const reqOpts = aggregateOrders(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          filter: "$[0].filter",
          aggregation: "$[0].aggregation",
          hierarchySeparatorOverride: "$[0].hierarchySeparatorOverride"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateOrderTags2(orderIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    orderIds,
    assignTags: options?.assignTags,
    unassignTags: options?.unassignTags
  });
  const reqOpts = bulkUpdateOrderTags(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          orderIds: "$[0]",
          assignTags: "$[1].assignTags",
          unassignTags: "$[1].unassignTags"
        },
        singleArgumentUnchanged: false
      },
      ["orderIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-v1-order-orders.public.ts
function importOrder3(httpClient) {
  return (order) => importOrder2(
    order,
    // @ts-ignore
    { httpClient }
  );
}
function setOrderNumberCounter3(httpClient) {
  return (counter, options) => setOrderNumberCounter2(
    counter,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkDeleteImportedOrders3(httpClient) {
  return (options) => bulkDeleteImportedOrders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function preparePaymentCollection3(httpClient) {
  return (ecomOrderId, amount, options) => preparePaymentCollection2(
    ecomOrderId,
    amount,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getPaymentCollectabilityStatus3(httpClient) {
  return (ecomOrderId) => getPaymentCollectabilityStatus2(
    ecomOrderId,
    // @ts-ignore
    { httpClient }
  );
}
function recordManuallyCollectedPayment3(httpClient) {
  return (orderId, amount) => recordManuallyCollectedPayment2(
    orderId,
    amount,
    // @ts-ignore
    { httpClient }
  );
}
function paymentCollectionMarkOrderAsPaid3(httpClient) {
  return (ecomOrderId) => paymentCollectionMarkOrderAsPaid2(
    ecomOrderId,
    // @ts-ignore
    { httpClient }
  );
}
function paymentCollectionBulkMarkOrdersAsPaid3(httpClient) {
  return (ecomOrderIds) => paymentCollectionBulkMarkOrdersAsPaid2(
    ecomOrderIds,
    // @ts-ignore
    { httpClient }
  );
}
function getRefundabilityStatus3(httpClient) {
  return (ecomOrderId) => getRefundabilityStatus2(
    ecomOrderId,
    // @ts-ignore
    { httpClient }
  );
}
function paymentCollectionCreatePaymentGatewayOrder3(httpClient) {
  return (ecomOrderId, options) => paymentCollectionCreatePaymentGatewayOrder2(
    ecomOrderId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function chargeMemberships3(httpClient) {
  return (ecomOrderId, memberId, options) => chargeMemberships2(
    ecomOrderId,
    memberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function triggerRefund3(httpClient) {
  return (ecomOrderId, payments, options) => triggerRefund2(
    ecomOrderId,
    payments,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function voidAuthorizedPayments3(httpClient) {
  return (ecomOrderId, paymentIds) => voidAuthorizedPayments2(
    ecomOrderId,
    paymentIds,
    // @ts-ignore
    { httpClient }
  );
}
function captureAuthorizedPayments3(httpClient) {
  return (ecomOrderId, payments) => captureAuthorizedPayments2(
    ecomOrderId,
    payments,
    // @ts-ignore
    { httpClient }
  );
}
function getOrder3(httpClient) {
  return (_id) => getOrder2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function searchOrders3(httpClient) {
  return (search) => searchOrders2(
    search,
    // @ts-ignore
    { httpClient }
  );
}
function createOrder3(httpClient) {
  return (order, options) => createOrder2(
    order,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateOrder3(httpClient) {
  return (_id, order) => updateOrder2(
    _id,
    order,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOrders3(httpClient) {
  return (orders, options) => bulkUpdateOrders2(
    orders,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateOrderLineItem3(httpClient) {
  return (identifiers, lineItem) => updateOrderLineItem2(
    identifiers,
    lineItem,
    // @ts-ignore
    { httpClient }
  );
}
function addActivity3(httpClient) {
  return (_id, activity) => addActivity2(
    _id,
    activity,
    // @ts-ignore
    { httpClient }
  );
}
function addActivities3(httpClient) {
  return (orderId, options) => addActivities2(
    orderId,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateActivity3(httpClient) {
  return (identifiers, activity) => updateActivity2(
    identifiers,
    activity,
    // @ts-ignore
    { httpClient }
  );
}
function deleteActivity3(httpClient) {
  return (identifiers) => deleteActivity2(
    identifiers,
    // @ts-ignore
    { httpClient }
  );
}
function cancelOrder3(httpClient) {
  return (_id, options) => cancelOrder2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}
function updateOrderStatus3(httpClient) {
  return (orderId, status) => updateOrderStatus2(
    orderId,
    status,
    // @ts-ignore
    { httpClient }
  );
}
function aggregateOrders3(httpClient) {
  return (options) => aggregateOrders2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function bulkUpdateOrderTags3(httpClient) {
  return (orderIds, options) => bulkUpdateOrderTags2(
    orderIds,
    options,
    // @ts-ignore
    { httpClient }
  );
}
var onOrderApproved = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.order_approved",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.purchasedDate" },
          { path: "data.order.lineItems.digitalFile.expirationDate" },
          {
            path: "data.order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "data.order.shippingInfo.logistics.deliverByDate" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "data.order.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "data.order.lineItems.image" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "data.order.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [
          { path: "data.order.billingInfo.address" },
          {
            path: "data.order.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "data.order.shippingInfo.logistics.pickupDetails.address" },
          { path: "data.order.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onOrderUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.order_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.purchasedDate" },
          { path: "entity.lineItems.digitalFile.expirationDate" },
          {
            path: "entity.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "entity.shippingInfo.logistics.deliverByDate" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "entity.activities.createdDate" },
          { path: "metadata.eventTime" },
          { path: "modifiedFields.createdDate" },
          { path: "modifiedFields.updatedDate" },
          { path: "modifiedFields.purchasedDate" },
          { path: "modifiedFields.lineItems.digitalFile.expirationDate" },
          {
            path: "modifiedFields.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "modifiedFields.shippingInfo.logistics.deliverByDate" },
          {
            path: "modifiedFields.shippingInfo.logistics.deliveryTimeSlot.from"
          },
          {
            path: "modifiedFields.shippingInfo.logistics.deliveryTimeSlot.to"
          },
          { path: "modifiedFields.activities.createdDate" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [
          { path: "entity.lineItems.image" },
          { path: "modifiedFields.lineItems.image" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.lineItems.physicalProperties.weight" },
          { path: "modifiedFields.lineItems.physicalProperties.weight" }
        ]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [
          { path: "entity.billingInfo.address" },
          {
            path: "entity.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "entity.shippingInfo.logistics.pickupDetails.address" },
          { path: "entity.recipientInfo.address" },
          { path: "modifiedFields.billingInfo.address" },
          {
            path: "modifiedFields.shippingInfo.logistics.shippingDestination.address"
          },
          {
            path: "modifiedFields.shippingInfo.logistics.pickupDetails.address"
          },
          { path: "modifiedFields.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onOrderCanceled = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.order_canceled",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.purchasedDate" },
          { path: "data.order.lineItems.digitalFile.expirationDate" },
          {
            path: "data.order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "data.order.shippingInfo.logistics.deliverByDate" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "data.order.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "data.order.lineItems.image" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "data.order.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [
          { path: "data.order.billingInfo.address" },
          {
            path: "data.order.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "data.order.shippingInfo.logistics.pickupDetails.address" },
          { path: "data.order.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onOrderCreated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.order_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.purchasedDate" },
          { path: "entity.lineItems.digitalFile.expirationDate" },
          {
            path: "entity.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "entity.shippingInfo.logistics.deliverByDate" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "entity.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "entity.lineItems.image" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "entity.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [
          { path: "entity.billingInfo.address" },
          {
            path: "entity.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "entity.shippingInfo.logistics.pickupDetails.address" },
          { path: "entity.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onOrderPaymentStatusUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.v1.order_payment_status_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.order.createdDate" },
          { path: "data.order.updatedDate" },
          { path: "data.order.purchasedDate" },
          { path: "data.order.lineItems.digitalFile.expirationDate" },
          {
            path: "data.order.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "data.order.shippingInfo.logistics.deliverByDate" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "data.order.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "data.order.activities.createdDate" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image3.transformRESTImageToSDKImage,
        paths: [{ path: "data.order.lineItems.image" }]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "data.order.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address3.transformRESTAddressToSDKAddress,
        paths: [
          { path: "data.order.billingInfo.address" },
          {
            path: "data.order.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "data.order.shippingInfo.logistics.pickupDetails.address" },
          { path: "data.order.recipientInfo.address" }
        ]
      }
    ])
  )
)();

// src/ecom-v1-order-orders.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var importOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(importOrder3);
var setOrderNumberCounter4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(setOrderNumberCounter3);
var bulkDeleteImportedOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkDeleteImportedOrders3);
var preparePaymentCollection4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(preparePaymentCollection3);
var getPaymentCollectabilityStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getPaymentCollectabilityStatus3);
var recordManuallyCollectedPayment4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(recordManuallyCollectedPayment3);
var paymentCollectionMarkOrderAsPaid4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(paymentCollectionMarkOrderAsPaid3);
var paymentCollectionBulkMarkOrdersAsPaid4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(paymentCollectionBulkMarkOrdersAsPaid3);
var getRefundabilityStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRefundabilityStatus3);
var paymentCollectionCreatePaymentGatewayOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(
  paymentCollectionCreatePaymentGatewayOrder3
);
var chargeMemberships4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(chargeMemberships3);
var triggerRefund4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(triggerRefund3);
var voidAuthorizedPayments4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(voidAuthorizedPayments3);
var captureAuthorizedPayments4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(captureAuthorizedPayments3);
var getOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getOrder3);
var searchOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(searchOrders3);
var createOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createOrder3);
var updateOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOrder3);
var bulkUpdateOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateOrders3);
var updateOrderLineItem4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOrderLineItem3);
var addActivity4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(addActivity3);
var addActivities4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(addActivities3);
var updateActivity4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateActivity3);
var deleteActivity4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteActivity3);
var cancelOrder4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(cancelOrder3);
var updateOrderStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateOrderStatus3);
var aggregateOrders4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(aggregateOrders3);
var bulkUpdateOrderTags4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(bulkUpdateOrderTags3);
var onOrderApproved2 = (0, import_event_definition_modules.createEventModule)(onOrderApproved);
var onOrderUpdated2 = (0, import_event_definition_modules.createEventModule)(onOrderUpdated);
var onOrderCanceled2 = (0, import_event_definition_modules.createEventModule)(onOrderCanceled);
var onOrderCreated2 = (0, import_event_definition_modules.createEventModule)(onOrderCreated);
var onOrderPaymentStatusUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onOrderPaymentStatusUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ActionType,
  ActivityType,
  AdditionalFeeSource,
  AdjustmentType,
  AttributionSource,
  AuthorizationCaptureStatus,
  AuthorizationVoidStatus,
  ChannelType,
  ChargeType,
  ChargebackStatus,
  CustomFieldGroup,
  DeltaPaymentOptionType,
  DepositType,
  DescriptionLineType,
  DiscountReason,
  DiscountType,
  DurationUnit,
  FulfillmentStatus,
  InventoryAction,
  InvoiceStatus,
  ItemTypePreset,
  JurisdictionType,
  LineItemQuantityChangeType,
  ManuallyRefundableReason,
  MembershipPaymentStatus,
  NonRefundableReason,
  OrderActionType,
  OrderActivityTypeEnumActivityType,
  OrderApprovalStrategy,
  OrderStatus,
  PaymentCollectabilityStatus,
  PaymentOptionType,
  PaymentStatus,
  PickupMethod,
  Placement,
  PreviewEmailType,
  Reason,
  RefundStatus,
  RefundableStatus,
  RestockType,
  ScheduledAction,
  SortOrder,
  SourceType,
  SubdivisionType,
  SubscriptionFrequency,
  TaxableAddressType,
  TransactionStatus,
  ValueType,
  VatType,
  VersioningMode,
  WebhookIdentityType,
  WeightUnit,
  addActivities,
  addActivity,
  aggregateOrders,
  bulkDeleteImportedOrders,
  bulkUpdateOrderTags,
  bulkUpdateOrders,
  cancelOrder,
  captureAuthorizedPayments,
  chargeMemberships,
  createOrder,
  deleteActivity,
  getOrder,
  getPaymentCollectabilityStatus,
  getRefundabilityStatus,
  importOrder,
  onOrderApproved,
  onOrderCanceled,
  onOrderCreated,
  onOrderPaymentStatusUpdated,
  onOrderUpdated,
  paymentCollectionBulkMarkOrdersAsPaid,
  paymentCollectionCreatePaymentGatewayOrder,
  paymentCollectionMarkOrderAsPaid,
  preparePaymentCollection,
  recordManuallyCollectedPayment,
  searchOrders,
  setOrderNumberCounter,
  triggerRefund,
  updateActivity,
  updateOrder,
  updateOrderLineItem,
  updateOrderStatus,
  voidAuthorizedPayments
});
