// @wix/auto_sdk_stores_info-sections-v-3/build/es/src/stores-catalog-v3-info-section-info-sections-v-3.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformSDKFloatToRESTFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformSDKFieldMaskToRESTFieldMask } from '@wix/sdk-runtime/transformations/field-mask';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl(opts) {
    const domainToMappings = {
        'www.wixapis.com': [
            {
                srcPath: '/stores/v3/info-sections',
                destPath: '/v3/info-sections',
            },
            {
                srcPath: '/stores/v3/bulk/info-sections',
                destPath: '/v3/bulk/info-sections',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/info-sections',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/info-section',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/info-section',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/info-section',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/info-section',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/info-section',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_info-sections-v-3';
/**
 * Creates an info section.
 *
 * To assign the info section to a product, include the `infoSection.id` or `infoSection.uniqueName`
 * when [creating](https://dev.wix.com/docs/rest/business-solutions/stores/catalog-v3/products-v3/create-product) or
 * [updating](https://dev.wix.com/docs/rest/business-solutions/stores/catalog-v3/products-v3/update-product) a product.
 */
export function createInfoSection(payload) {
    function __createInfoSection({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSection.createdDate' },
                    { path: 'infoSection.updatedDate' },
                    { path: 'infoSection.description.metadata.createdTimestamp' },
                    { path: 'infoSection.description.metadata.updatedTimestamp' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'infoSection.description.nodes.imageData.image.duration' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'infoSection.description.nodes.appEmbedData.image.duration' },
                    { path: 'infoSection.description.nodes.videoData.video.duration' },
                    {
                        path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                    },
                    { path: 'infoSection.description.nodes.audioData.audio.duration' },
                    {
                        path: 'infoSection.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.CreateInfoSection',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'infoSection.createdDate' },
                        { path: 'infoSection.updatedDate' },
                        { path: 'infoSection.description.metadata.createdTimestamp' },
                        { path: 'infoSection.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'infoSection.description.nodes.imageData.image.duration',
                        },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createInfoSection;
}
/** Retrieves an info section. */
export function getInfoSection(payload) {
    function __getInfoSection({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'GET',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.GetInfoSection',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections/{infoSectionId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'infoSection.createdDate' },
                        { path: 'infoSection.updatedDate' },
                        { path: 'infoSection.description.metadata.createdTimestamp' },
                        { path: 'infoSection.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'infoSection.description.nodes.imageData.image.duration',
                        },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getInfoSection;
}
/**
 * Retrieves an info section by ID or `uniqueName`, or creates an info section if one with the passed `uniqueName` doesn't exist.
 *
 * > **Note:** If an info section with the passed `uniqueName` doesn't exist, the `uniqueName` and `title` fields are required to create a new info section.
 */
export function getOrCreateInfoSection(payload) {
    function __getOrCreateInfoSection({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'infoSection.description.nodes.imageData.image.duration' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'infoSection.description.nodes.appEmbedData.image.duration' },
                    { path: 'infoSection.description.nodes.videoData.video.duration' },
                    {
                        path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                    },
                    { path: 'infoSection.description.nodes.audioData.audio.duration' },
                    {
                        path: 'infoSection.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSection.description.metadata.createdTimestamp' },
                    { path: 'infoSection.description.metadata.updatedTimestamp' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.GetOrCreateInfoSection',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections/get-or-create',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'infoSection.createdDate' },
                        { path: 'infoSection.updatedDate' },
                        { path: 'infoSection.description.metadata.createdTimestamp' },
                        { path: 'infoSection.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'infoSection.description.nodes.imageData.image.duration',
                        },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getOrCreateInfoSection;
}
/**
 * Retrieves multiple info sections by ID or `uniqueName`, or creates multiple info sections if those with the passed `uniqueName` don't exist.
 *
 * > **Note:** If an info section with the passed `uniqueName` doesn't exist, the `uniqueName` and `title` fields are required to create a new info section.
 */
export function bulkGetOrCreateInfoSections(payload) {
    function __bulkGetOrCreateInfoSections({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSections.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'infoSections.description.nodes.imageData.image.duration' },
                    { path: 'infoSections.description.nodes.mapData.mapSettings.lat' },
                    { path: 'infoSections.description.nodes.mapData.mapSettings.lng' },
                    {
                        path: 'infoSections.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.appEmbedData.image.duration',
                    },
                    { path: 'infoSections.description.nodes.videoData.video.duration' },
                    {
                        path: 'infoSections.description.nodes.videoData.thumbnail.duration',
                    },
                    { path: 'infoSections.description.nodes.audioData.audio.duration' },
                    {
                        path: 'infoSections.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSections.description.metadata.createdTimestamp' },
                    { path: 'infoSections.description.metadata.updatedTimestamp' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.BulkGetOrCreateInfoSections',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/bulk/info-sections/get-or-create',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.updatedDate' },
                        { path: 'results.item.description.metadata.createdTimestamp' },
                        { path: 'results.item.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'results.item.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.description.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkGetOrCreateInfoSections;
}
/**
 * Updates an info section.
 *
 *
 * Each time the info section is updated, `revision` increments by 1.
 * The current `revision` must be passed when updating the info section.
 * This ensures you're working with the latest info section and prevents unintended overwrites.
 */
export function updateInfoSection(payload) {
    function __updateInfoSection({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSection.createdDate' },
                    { path: 'infoSection.updatedDate' },
                    { path: 'infoSection.description.metadata.createdTimestamp' },
                    { path: 'infoSection.description.metadata.updatedTimestamp' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'infoSection.description.nodes.imageData.image.duration' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                    { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'infoSection.description.nodes.appEmbedData.image.duration' },
                    { path: 'infoSection.description.nodes.videoData.video.duration' },
                    {
                        path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                    },
                    { path: 'infoSection.description.nodes.audioData.audio.duration' },
                    {
                        path: 'infoSection.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'PATCH',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.UpdateInfoSection',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections/{infoSection.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'infoSection.createdDate' },
                        { path: 'infoSection.updatedDate' },
                        { path: 'infoSection.description.metadata.createdTimestamp' },
                        { path: 'infoSection.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'infoSection.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'infoSection.description.nodes.imageData.image.duration',
                        },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lat' },
                        { path: 'infoSection.description.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'infoSection.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateInfoSection;
}
/**
 * Deletes an info section.
 *
 * > **Note:** Deleting an info section will also remove it from all products it is assigned to.
 */
export function deleteInfoSection(payload) {
    function __deleteInfoSection({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'DELETE',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.DeleteInfoSection',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections/{infoSectionId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __deleteInfoSection;
}
/**
 * Retrieves a list of up to 100 info sections, given the provided filtering, sorting, and cursor paging.
 * Pass supported values to the `fields` array in the request to include those fields in the response.
 *
 *
 * Query Info Sections runs with these defaults, which you can override:
 *
 * - `createdDate` is sorted in `DESC` order
 * - `cursorPaging.limit` is `100`
 *
 * For field support for filters and sorting,
 * see [Info Sections: Supported Filters and Sorting](https://dev.wix.com/docs/rest/business-solutions/stores/catalog-v3/info-sections-v3/supported-filters-and-sorting).
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/docs/rest/articles/getting-started/api-query-language),
 * and [Sorting and Paging](https://dev.wix.com/docs/rest/articles/getting-started/sorting-and-paging).
 */
export function queryInfoSections(payload) {
    function __queryInfoSections({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.QueryInfoSections',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/info-sections/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'infoSections.createdDate' },
                        { path: 'infoSections.updatedDate' },
                        { path: 'infoSections.description.metadata.createdTimestamp' },
                        { path: 'infoSections.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'infoSections.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'infoSections.description.nodes.imageData.image.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'infoSections.description.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'infoSections.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'infoSections.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryInfoSections;
}
/** Creates multiple info sections. */
export function bulkCreateInfoSections(payload) {
    function __bulkCreateInfoSections({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSections.createdDate' },
                    { path: 'infoSections.updatedDate' },
                    { path: 'infoSections.description.metadata.createdTimestamp' },
                    { path: 'infoSections.description.metadata.updatedTimestamp' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSections.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'infoSections.description.nodes.imageData.image.duration' },
                    { path: 'infoSections.description.nodes.mapData.mapSettings.lat' },
                    { path: 'infoSections.description.nodes.mapData.mapSettings.lng' },
                    {
                        path: 'infoSections.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    {
                        path: 'infoSections.description.nodes.appEmbedData.image.duration',
                    },
                    { path: 'infoSections.description.nodes.videoData.video.duration' },
                    {
                        path: 'infoSections.description.nodes.videoData.thumbnail.duration',
                    },
                    { path: 'infoSections.description.nodes.audioData.audio.duration' },
                    {
                        path: 'infoSections.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.BulkCreateInfoSections',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/bulk/info-sections/create',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.updatedDate' },
                        { path: 'results.item.description.metadata.createdTimestamp' },
                        { path: 'results.item.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'results.item.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.description.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkCreateInfoSections;
}
/**
 * Updates multiple info sections.
 *
 * Each time an info section is updated, `revision` increments by 1.
 * The current `revision` must be passed when updating an info section.
 * This ensures you're working with the latest info section and prevents unintended overwrites.
 */
export function bulkUpdateInfoSections(payload) {
    function __bulkUpdateInfoSections({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'infoSections.fieldMask' }],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'infoSections.infoSection.createdDate' },
                    { path: 'infoSections.infoSection.updatedDate' },
                    {
                        path: 'infoSections.infoSection.description.metadata.createdTimestamp',
                    },
                    {
                        path: 'infoSections.infoSection.description.metadata.updatedTimestamp',
                    },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'infoSections.infoSection.description.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.galleryData.options.item.ratio',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.imageData.image.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.mapData.mapSettings.lat',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.mapData.mapSettings.lng',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.pollData.design.poll.background.image.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.appEmbedData.image.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.videoData.video.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.videoData.thumbnail.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.audioData.audio.duration',
                    },
                    {
                        path: 'infoSections.infoSection.description.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.BulkUpdateInfoSections',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/bulk/info-sections/update',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.updatedDate' },
                        { path: 'results.item.description.metadata.createdTimestamp' },
                        { path: 'results.item.description.metadata.updatedTimestamp' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'results.item.description.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.description.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.description.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.description.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.description.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkUpdateInfoSections;
}
/** Deletes multiple info sections. */
export function bulkDeleteInfoSections(payload) {
    function __bulkDeleteInfoSections({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v3.info_section',
            method: 'POST',
            methodFqn: 'com.wix.stores.catalog.info.section.v3.InfoSectionsService.BulkDeleteInfoSections',
            packageName: PACKAGE_NAME,
            url: resolveComWixStoresCatalogInfoSectionV3InfoSectionsServiceUrl({
                protoPath: '/v3/bulk/info-sections/delete',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __bulkDeleteInfoSections;
}
