// @wix/auto_sdk_stores_info-sections-v-3/build/es/src/stores-catalog-v3-info-section-info-sections-v-3.meta.js
import * as ambassadorWixStoresCatalogV3InfoSection from './stores-catalog-v3-info-section-info-sections-v-3.http.js';
export function createInfoSection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.createInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getInfoSection() {
    const payload = { infoSectionId: ':infoSectionId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.getInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/info-sections/{infoSectionId}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrCreateInfoSection() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.getOrCreateInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGetOrCreateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkGetOrCreateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateInfoSection() {
    const payload = { infoSection: { id: ':infoSectionId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.updateInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/info-sections/{infoSection.id}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteInfoSection() {
    const payload = { infoSectionId: ':infoSectionId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.deleteInfoSection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/info-sections/{infoSectionId}',
        pathParams: { infoSectionId: 'infoSectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.queryInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/info-sections/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkCreateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkUpdateInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteInfoSections() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3InfoSection.bulkDeleteInfoSections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/info-sections/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
