// @wix/auto_sdk_stores_collections/build/internal/cjs/src/stores-catalog-v1-collection-collections.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCollectionBySlug = exports.getCollection = exports.queryCollections = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCatalogApiV2CollectionReadApiUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/stores-reader/v2/collections',
                destPath: '/v2/collections',
            },
            {
                srcPath: '/stores-reader/v2/inventoryItems',
                destPath: '/v2/inventoryItems',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/wix-ecommerce-catalog-reader-web',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/stores-reader/v2/inventoryItems',
                destPath: '/v2/inventoryItems',
            },
            {
                srcPath: '/stores-reader/v2/collections',
                destPath: '/v2/collections',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_stores_collections';
/**
 * Retrieves a list of up to 100 collections, given the provided paging, sorting and filtering.
 * See [Stores Pagination](https://dev.wix.com/api/rest/wix-stores/pagination) for more information.
 */
function queryCollections(payload) {
    function __queryCollections({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v1.collection',
            method: 'POST',
            methodFqn: 'wix.catalog.api.v2.CollectionReadApi.QueryCollections',
            packageName: PACKAGE_NAME,
            url: resolveWixCatalogApiV2CollectionReadApiUrl({
                protoPath: '/v2/collections/query',
                data: payload,
                host,
            }),
            data: payload,
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixCatalogApiV2CollectionReadApiUrl({
                        protoPath: '/v2/collections/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __queryCollections;
}
exports.queryCollections = queryCollections;
/** Retrieves a collection with the provided ID. */
function getCollection(payload) {
    function __getCollection({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v1.collection',
            method: 'GET',
            methodFqn: 'wix.catalog.api.v2.CollectionReadApi.GetCollection',
            packageName: PACKAGE_NAME,
            url: resolveWixCatalogApiV2CollectionReadApiUrl({
                protoPath: '/v2/collections/{id}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getCollection;
}
exports.getCollection = getCollection;
/** Retrieves a collection with the provided slug. */
function getCollectionBySlug(payload) {
    function __getCollectionBySlug({ host }) {
        const metadata = {
            entityFqdn: 'wix.stores.catalog.v1.collection',
            method: 'GET',
            methodFqn: 'wix.catalog.api.v2.CollectionReadApi.GetCollectionBySlug',
            packageName: PACKAGE_NAME,
            url: resolveWixCatalogApiV2CollectionReadApiUrl({
                protoPath: '/v2/collections/slug/{slug}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getCollectionBySlug;
}
exports.getCollectionBySlug = getCollectionBySlug;
