// @wix/auto_sdk_stores_inventory-items-v-3/build/cjs/src/stores-catalog-v3-inventory-item-inventory-items-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortType = exports.SortOrder = exports.SortDirection = exports.ScalarType = exports.ReasonType = exports.NestedAggregationType = exports.Mode = exports.MissingValues = exports.Interval = exports.AvailabilityStatus = exports.AggregationType = exports.onInventoryItemUpdatedWithReason = exports.onInventoryItemUpdated = exports.onInventoryItemStockStatusUpdated = exports.onInventoryItemDeleted = exports.onInventoryItemCreated = exports.bulkSetOrAdjustInventoryItemsByFilter = exports.bulkIncrementInventoryItemsByVariantAndLocation = exports.bulkDecrementInventoryItemsByVariantAndLocation = exports.bulkIncrementInventoryItems = exports.bulkDecrementInventoryItems = exports.searchInventoryItems = exports.queryInventoryItems = exports.bulkDeleteInventoryItems = exports.deleteInventoryItem = exports.bulkUpdateInventoryItemsByFilter = exports.bulkUpdateInventoryItems = exports.updateInventoryItem = exports.getInventoryItem = exports.bulkCreateInventoryItems = exports.createInventoryItem = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1 = require("./stores-catalog-v3-inventory-item-inventory-items-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createInventoryItem(httpClient) {
    return (inventoryItem) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.createInventoryItem)(inventoryItem, 
    // @ts-ignore
    { httpClient });
}
exports.createInventoryItem = createInventoryItem;
function bulkCreateInventoryItems(httpClient) {
    return (inventoryItems, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkCreateInventoryItems)(inventoryItems, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateInventoryItems = bulkCreateInventoryItems;
function getInventoryItem(httpClient) {
    return (inventoryItemId) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.getInventoryItem)(inventoryItemId, 
    // @ts-ignore
    { httpClient });
}
exports.getInventoryItem = getInventoryItem;
function updateInventoryItem(httpClient) {
    return (_id, inventoryItem, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.updateInventoryItem)(_id, inventoryItem, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateInventoryItem = updateInventoryItem;
function bulkUpdateInventoryItems(httpClient) {
    return (inventoryItems, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkUpdateInventoryItems)(inventoryItems, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateInventoryItems = bulkUpdateInventoryItems;
function bulkUpdateInventoryItemsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkUpdateInventoryItemsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateInventoryItemsByFilter = bulkUpdateInventoryItemsByFilter;
function deleteInventoryItem(httpClient) {
    return (inventoryItemId) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.deleteInventoryItem)(inventoryItemId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteInventoryItem = deleteInventoryItem;
function bulkDeleteInventoryItems(httpClient) {
    return (inventoryItemIds) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkDeleteInventoryItems)(inventoryItemIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteInventoryItems = bulkDeleteInventoryItems;
function queryInventoryItems(httpClient) {
    return () => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.queryInventoryItems)(
    // @ts-ignore
    { httpClient });
}
exports.queryInventoryItems = queryInventoryItems;
function searchInventoryItems(httpClient) {
    return (options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.searchInventoryItems)(options, 
    // @ts-ignore
    { httpClient });
}
exports.searchInventoryItems = searchInventoryItems;
function bulkDecrementInventoryItems(httpClient) {
    return (decrementData, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkDecrementInventoryItems)(decrementData, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDecrementInventoryItems = bulkDecrementInventoryItems;
function bulkIncrementInventoryItems(httpClient) {
    return (incrementData, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkIncrementInventoryItems)(incrementData, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkIncrementInventoryItems = bulkIncrementInventoryItems;
function bulkDecrementInventoryItemsByVariantAndLocation(httpClient) {
    return (decrementData, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkDecrementInventoryItemsByVariantAndLocation)(decrementData, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDecrementInventoryItemsByVariantAndLocation = bulkDecrementInventoryItemsByVariantAndLocation;
function bulkIncrementInventoryItemsByVariantAndLocation(httpClient) {
    return (incrementData, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkIncrementInventoryItemsByVariantAndLocation)(incrementData, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkIncrementInventoryItemsByVariantAndLocation = bulkIncrementInventoryItemsByVariantAndLocation;
/** @internal */
function bulkSetOrAdjustInventoryItemsByFilter(httpClient) {
    return (filter, options) => (0, stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_1.bulkSetOrAdjustInventoryItemsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkSetOrAdjustInventoryItemsByFilter = bulkSetOrAdjustInventoryItemsByFilter;
exports.onInventoryItemCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.inventory_item_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onInventoryItemDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.inventory_item_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onInventoryItemStockStatusUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.inventory_item_stock_status_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
])))();
exports.onInventoryItemUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.inventory_item_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onInventoryItemUpdatedWithReason = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.inventory_item_updated_with_reason', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.currentEntity.createdDate' },
            { path: 'data.currentEntity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2 = require("./stores-catalog-v3-inventory-item-inventory-items-v-3.universal.js");
Object.defineProperty(exports, "AggregationType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.AggregationType; } });
Object.defineProperty(exports, "AvailabilityStatus", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.AvailabilityStatus; } });
Object.defineProperty(exports, "Interval", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.Interval; } });
Object.defineProperty(exports, "MissingValues", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.MissingValues; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.Mode; } });
Object.defineProperty(exports, "NestedAggregationType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.NestedAggregationType; } });
Object.defineProperty(exports, "ReasonType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.ReasonType; } });
Object.defineProperty(exports, "ScalarType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.ScalarType; } });
Object.defineProperty(exports, "SortDirection", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.SortDirection; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SortType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.SortType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_inventory_item_inventory_items_v_3_universal_js_2.WebhookIdentityType; } });
