// @wix/auto_sdk_stores_ribbons-v-3/build/cjs/src/stores-catalog-v3-ribbon-ribbons-v-3.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkDeleteRibbons = exports.bulkGetOrCreateRibbons = exports.getOrCreateRibbon = exports.bulkUpdateRibbons = exports.bulkCreateRibbons = exports.queryRibbons = exports.deleteRibbon = exports.updateRibbon = exports.getRibbon = exports.createRibbon = void 0;
const ambassadorWixStoresCatalogV3Ribbon = __importStar(require("./stores-catalog-v3-ribbon-ribbons-v-3.http.js"));
function createRibbon() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.createRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createRibbon = createRibbon;
function getRibbon() {
    const payload = { ribbonId: ':ribbonId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.getRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/ribbons/{ribbonId}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getRibbon = getRibbon;
function updateRibbon() {
    const payload = { ribbon: { id: ':ribbonId' } };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.updateRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/ribbons/{ribbon.id}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateRibbon = updateRibbon;
function deleteRibbon() {
    const payload = { ribbonId: ':ribbonId' };
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.deleteRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/ribbons/{ribbonId}',
        pathParams: { ribbonId: 'ribbonId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteRibbon = deleteRibbon;
function queryRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.queryRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryRibbons = queryRibbons;
function bulkCreateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkCreateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateRibbons = bulkCreateRibbons;
function bulkUpdateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkUpdateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateRibbons = bulkUpdateRibbons;
function getOrCreateRibbon() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.getOrCreateRibbon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/ribbons/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrCreateRibbon = getOrCreateRibbon;
function bulkGetOrCreateRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkGetOrCreateRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/get-or-create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGetOrCreateRibbons = bulkGetOrCreateRibbons;
function bulkDeleteRibbons() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresCatalogV3Ribbon.bulkDeleteRibbons(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/ribbons/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteRibbons = bulkDeleteRibbons;
