// @wix/auto_sdk_stores_ribbons-v-3/build/cjs/src/stores-catalog-v3-ribbon-ribbons-v-3.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.RequestedFields = exports.onRibbonUpdated = exports.onRibbonDeleted = exports.onRibbonCreated = exports.bulkDeleteRibbons = exports.bulkGetOrCreateRibbons = exports.getOrCreateRibbon = exports.bulkUpdateRibbons = exports.bulkCreateRibbons = exports.queryRibbons = exports.deleteRibbon = exports.updateRibbon = exports.getRibbon = exports.createRibbon = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1 = require("./stores-catalog-v3-ribbon-ribbons-v-3.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/stores' };
function createRibbon(httpClient) {
    return (ribbon) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.createRibbon)(ribbon, 
    // @ts-ignore
    { httpClient });
}
exports.createRibbon = createRibbon;
function getRibbon(httpClient) {
    return (ribbonId, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.getRibbon)(ribbonId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getRibbon = getRibbon;
function updateRibbon(httpClient) {
    return (_id, ribbon, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.updateRibbon)(_id, ribbon, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateRibbon = updateRibbon;
function deleteRibbon(httpClient) {
    return (ribbonId) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.deleteRibbon)(ribbonId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteRibbon = deleteRibbon;
function queryRibbons(httpClient) {
    return (options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.queryRibbons)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryRibbons = queryRibbons;
function bulkCreateRibbons(httpClient) {
    return (ribbons, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.bulkCreateRibbons)(ribbons, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateRibbons = bulkCreateRibbons;
function bulkUpdateRibbons(httpClient) {
    return (ribbons, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.bulkUpdateRibbons)(ribbons, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateRibbons = bulkUpdateRibbons;
function getOrCreateRibbon(httpClient) {
    return (ribbonName, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.getOrCreateRibbon)(ribbonName, options, 
    // @ts-ignore
    { httpClient });
}
exports.getOrCreateRibbon = getOrCreateRibbon;
function bulkGetOrCreateRibbons(httpClient) {
    return (ribbonNames, options) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.bulkGetOrCreateRibbons)(ribbonNames, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkGetOrCreateRibbons = bulkGetOrCreateRibbons;
function bulkDeleteRibbons(httpClient) {
    return (ribbonIds) => (0, stores_catalog_v3_ribbon_ribbons_v_3_universal_js_1.bulkDeleteRibbons)(ribbonIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteRibbons = bulkDeleteRibbons;
exports.onRibbonCreated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.ribbon_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onRibbonDeleted = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.ribbon_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onRibbonUpdated = (0, sdk_types_1.EventDefinition)('wix.stores.catalog.v3.ribbon_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var stores_catalog_v3_ribbon_ribbons_v_3_universal_js_2 = require("./stores-catalog-v3-ribbon-ribbons-v-3.universal.js");
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return stores_catalog_v3_ribbon_ribbons_v_3_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return stores_catalog_v3_ribbon_ribbons_v_3_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return stores_catalog_v3_ribbon_ribbons_v_3_universal_js_2.WebhookIdentityType; } });
