// @wix/auto_sdk_stores_wishlist/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  WebhookIdentityType: () => WebhookIdentityType,
  getWishlistById: () => getWishlistById4
});
module.exports = __toCommonJS(index_exports);

// src/platform-v1-wishlist-wishlist.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/platform-v1-wishlist-wishlist.http.ts
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressWishlistWishlistUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "ecom._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: "/api"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/wishlists",
        destPath: "/v1/wishlists"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_wishlist";
function getWishlistById(payload) {
  function __getWishlistById({ host }) {
    const metadata = {
      entityFqdn: "wix.platform.v1.wishlist",
      method: "POST",
      methodFqn: "com.wixpress.wishlist.Wishlist.GetWishlistById",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressWishlistWishlistUrl({
        protoPath: "/v1/wishlists/get",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "wishlist.items.dateAdded" }]
        }
      ])
    };
    return metadata;
  }
  return __getWishlistById;
}

// src/platform-v1-wishlist-wishlist.universal.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getWishlistById2(_id, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    id: _id,
    limit: options?.limit,
    offset: options?.offset,
    kind: options?.kind
  });
  const reqOpts = getWishlistById(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.wishlist;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          id: "$[0]",
          limit: "$[1].limit",
          offset: "$[1].offset",
          kind: "$[1].kind"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/platform-v1-wishlist-wishlist.public.ts
function getWishlistById3(httpClient) {
  return (_id, options) => getWishlistById2(
    _id,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/platform-v1-wishlist-wishlist.context.ts
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
var getWishlistById4 = /* @__PURE__ */ (0, import_rest_modules2.createRESTModule)(getWishlistById3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityType,
  getWishlistById
});
