// @wix/auto_sdk_stores_wishlist/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  getWishlistById: () => getWishlistById2
});
module.exports = __toCommonJS(meta_exports);

// src/platform-v1-wishlist-wishlist.http.ts
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressWishlistWishlistUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/wishlist-server",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "ecom._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: "/api"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/stores/v1/wishlists",
        destPath: "/v1/wishlists"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/wishlist-server",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_stores_wishlist";
function getWishlistById(payload) {
  function __getWishlistById({ host }) {
    const metadata = {
      entityFqdn: "wix.platform.v1.wishlist",
      method: "POST",
      methodFqn: "com.wixpress.wishlist.Wishlist.GetWishlistById",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressWishlistWishlistUrl({
        protoPath: "/v1/wishlists/get",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "wishlist.items.dateAdded" }]
        }
      ])
    };
    return metadata;
  }
  return __getWishlistById;
}

// src/platform-v1-wishlist-wishlist.types.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/platform-v1-wishlist-wishlist.meta.ts
function getWishlistById2() {
  const payload = {};
  const getRequestOptions = getWishlistById(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/wishlists/get",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityTypeOriginal,
  getWishlistById
});
