// @wix/auto_sdk_ecom_subscription-contracts/build/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AdditionalFeeSource: () => AdditionalFeeSource,
  AdjustmentType: () => AdjustmentType,
  ChannelType: () => ChannelType,
  ChargeType: () => ChargeType,
  DescriptionLineType: () => DescriptionLineType,
  DiscountReason: () => DiscountReason,
  DiscountType: () => DiscountType,
  ItemTypePreset: () => ItemTypePreset,
  JurisdictionType: () => JurisdictionType,
  PaymentOptionType: () => PaymentOptionType,
  PickupMethod: () => PickupMethod,
  SortOrder: () => SortOrder,
  SubscriptionFrequency: () => SubscriptionFrequency,
  TaxableAddressType: () => TaxableAddressType,
  V2SubscriptionFrequency: () => V2SubscriptionFrequency,
  VatType: () => VatType,
  WebhookIdentityType: () => WebhookIdentityType,
  WeightUnit: () => WeightUnit,
  getSubscriptionContract: () => getSubscriptionContract4,
  onSubscriptionContractCreated: () => onSubscriptionContractCreated2,
  onSubscriptionContractDeleted: () => onSubscriptionContractDeleted2,
  onSubscriptionContractUpdated: () => onSubscriptionContractUpdated2,
  querySubscriptionContracts: () => querySubscriptionContracts4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_address2 = require("@wix/sdk-runtime/transformations/address");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/ecom/subscription-contracts/v1/subscription-contracts",
        destPath: "/v1/subscription-contracts"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/subscription-contracts/v1/subscription-contracts",
        destPath: "/v1/subscription-contracts"
      }
    ],
    _: [
      {
        srcPath: "/ecom/subscription-contracts/v1/subscription-contracts",
        destPath: "/v1/subscription-contracts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_subscription-contracts";
function getSubscriptionContract(payload) {
  function __getSubscriptionContract({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.subscription_contracts.v1.subscription_contract",
      method: "GET",
      methodFqn: "wix.ecom.subscription_contracts.v1.SubscriptionContracts.GetSubscriptionContract",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
        protoPath: "/v1/subscription-contracts/{subscriptionContractId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "subscriptionContract.createdDate" },
            { path: "subscriptionContract.updatedDate" },
            { path: "subscriptionContract.subscriptionSettings.startDate" },
            {
              path: "subscriptionContract.lineItems.image.urlExpirationDate"
            },
            {
              path: "subscriptionContract.lineItems.digitalFile.expirationDate"
            },
            {
              path: "subscriptionContract.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "subscriptionContract.lineItems.image.focalPoint.x" },
            { path: "subscriptionContract.lineItems.image.focalPoint.y" },
            {
              path: "subscriptionContract.lineItems.physicalProperties.weight"
            },
            {
              path: "subscriptionContract.billingInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.billingInfo.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "subscriptionContract.recipientInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContract.recipientInfo.address.geocode.longitude"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getSubscriptionContract;
}
function querySubscriptionContracts(payload) {
  function __querySubscriptionContracts({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.subscription_contracts.v1.subscription_contract",
      method: "GET",
      methodFqn: "wix.ecom.subscription_contracts.v1.SubscriptionContracts.QuerySubscriptionContracts",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
        protoPath: "/v1/subscription-contracts/query",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "subscriptionContracts.createdDate" },
            { path: "subscriptionContracts.updatedDate" },
            { path: "subscriptionContracts.subscriptionSettings.startDate" },
            {
              path: "subscriptionContracts.lineItems.image.urlExpirationDate"
            },
            {
              path: "subscriptionContracts.lineItems.digitalFile.expirationDate"
            },
            {
              path: "subscriptionContracts.lineItems.subscriptionInfo.subscriptionSettings.startDate"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliverByDate"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.from"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.to"
            }
          ]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "subscriptionContracts.lineItems.image.focalPoint.x" },
            { path: "subscriptionContracts.lineItems.image.focalPoint.y" },
            {
              path: "subscriptionContracts.lineItems.physicalProperties.weight"
            },
            {
              path: "subscriptionContracts.billingInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.billingInfo.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.longitude"
            },
            {
              path: "subscriptionContracts.recipientInfo.address.geocode.latitude"
            },
            {
              path: "subscriptionContracts.recipientInfo.address.geocode.longitude"
            }
          ]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
            protoPath: "/v1/subscription-contracts/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __querySubscriptionContracts;
}

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.universal.ts
var import_address = require("@wix/sdk-runtime/transformations/address");
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var SubscriptionFrequency = /* @__PURE__ */ ((SubscriptionFrequency2) => {
  SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  SubscriptionFrequency2["DAY"] = "DAY";
  SubscriptionFrequency2["WEEK"] = "WEEK";
  SubscriptionFrequency2["MONTH"] = "MONTH";
  SubscriptionFrequency2["YEAR"] = "YEAR";
  return SubscriptionFrequency2;
})(SubscriptionFrequency || {});
var DescriptionLineType = /* @__PURE__ */ ((DescriptionLineType2) => {
  DescriptionLineType2["UNRECOGNISED"] = "UNRECOGNISED";
  DescriptionLineType2["PLAIN_TEXT"] = "PLAIN_TEXT";
  DescriptionLineType2["COLOR"] = "COLOR";
  return DescriptionLineType2;
})(DescriptionLineType || {});
var ItemTypePreset = /* @__PURE__ */ ((ItemTypePreset2) => {
  ItemTypePreset2["UNRECOGNISED"] = "UNRECOGNISED";
  ItemTypePreset2["PHYSICAL"] = "PHYSICAL";
  ItemTypePreset2["DIGITAL"] = "DIGITAL";
  ItemTypePreset2["GIFT_CARD"] = "GIFT_CARD";
  ItemTypePreset2["SERVICE"] = "SERVICE";
  return ItemTypePreset2;
})(ItemTypePreset || {});
var PaymentOptionType = /* @__PURE__ */ ((PaymentOptionType2) => {
  PaymentOptionType2["FULL_PAYMENT_ONLINE"] = "FULL_PAYMENT_ONLINE";
  PaymentOptionType2["FULL_PAYMENT_OFFLINE"] = "FULL_PAYMENT_OFFLINE";
  PaymentOptionType2["MEMBERSHIP"] = "MEMBERSHIP";
  PaymentOptionType2["DEPOSIT_ONLINE"] = "DEPOSIT_ONLINE";
  PaymentOptionType2["MEMBERSHIP_OFFLINE"] = "MEMBERSHIP_OFFLINE";
  PaymentOptionType2["MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER"] = "MEMBERSHIP_ONLINE_WITH_OFFLINE_REMAINDER";
  return PaymentOptionType2;
})(PaymentOptionType || {});
var JurisdictionType = /* @__PURE__ */ ((JurisdictionType2) => {
  JurisdictionType2["UNDEFINED"] = "UNDEFINED";
  JurisdictionType2["COUNTRY"] = "COUNTRY";
  JurisdictionType2["STATE"] = "STATE";
  JurisdictionType2["COUNTY"] = "COUNTY";
  JurisdictionType2["CITY"] = "CITY";
  JurisdictionType2["SPECIAL"] = "SPECIAL";
  return JurisdictionType2;
})(JurisdictionType || {});
var V2SubscriptionFrequency = /* @__PURE__ */ ((V2SubscriptionFrequency2) => {
  V2SubscriptionFrequency2["UNDEFINED"] = "UNDEFINED";
  V2SubscriptionFrequency2["DAY"] = "DAY";
  V2SubscriptionFrequency2["WEEK"] = "WEEK";
  V2SubscriptionFrequency2["MONTH"] = "MONTH";
  V2SubscriptionFrequency2["YEAR"] = "YEAR";
  return V2SubscriptionFrequency2;
})(V2SubscriptionFrequency || {});
var AdjustmentType = /* @__PURE__ */ ((AdjustmentType2) => {
  AdjustmentType2["UNKNOWN_ADJUSTMENT_TYPE"] = "UNKNOWN_ADJUSTMENT_TYPE";
  AdjustmentType2["EXTRA_CHARGE"] = "EXTRA_CHARGE";
  AdjustmentType2["CREDIT"] = "CREDIT";
  return AdjustmentType2;
})(AdjustmentType || {});
var TaxableAddressType = /* @__PURE__ */ ((TaxableAddressType2) => {
  TaxableAddressType2["UNKNOWN_TAXABLE_ADDRESS"] = "UNKNOWN_TAXABLE_ADDRESS";
  TaxableAddressType2["BUSINESS"] = "BUSINESS";
  TaxableAddressType2["BILLING"] = "BILLING";
  TaxableAddressType2["SHIPPING"] = "SHIPPING";
  return TaxableAddressType2;
})(TaxableAddressType || {});
var WeightUnit = /* @__PURE__ */ ((WeightUnit2) => {
  WeightUnit2["UNSPECIFIED_WEIGHT_UNIT"] = "UNSPECIFIED_WEIGHT_UNIT";
  WeightUnit2["KG"] = "KG";
  WeightUnit2["LB"] = "LB";
  return WeightUnit2;
})(WeightUnit || {});
var ChargeType = /* @__PURE__ */ ((ChargeType2) => {
  ChargeType2["UNKNOWN_CHARGE_TYPE"] = "UNKNOWN_CHARGE_TYPE";
  ChargeType2["PASS_ON"] = "PASS_ON";
  ChargeType2["ABSORBED"] = "ABSORBED";
  return ChargeType2;
})(ChargeType || {});
var VatType = /* @__PURE__ */ ((VatType2) => {
  VatType2["UNSPECIFIED"] = "UNSPECIFIED";
  VatType2["CPF"] = "CPF";
  VatType2["CNPJ"] = "CNPJ";
  return VatType2;
})(VatType || {});
var PickupMethod = /* @__PURE__ */ ((PickupMethod2) => {
  PickupMethod2["UNKNOWN_METHOD"] = "UNKNOWN_METHOD";
  PickupMethod2["STORE_PICKUP"] = "STORE_PICKUP";
  PickupMethod2["PICKUP_POINT"] = "PICKUP_POINT";
  return PickupMethod2;
})(PickupMethod || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["GLOBAL"] = "GLOBAL";
  DiscountType2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  DiscountType2["SHIPPING"] = "SHIPPING";
  return DiscountType2;
})(DiscountType || {});
var DiscountReason = /* @__PURE__ */ ((DiscountReason2) => {
  DiscountReason2["UNSPECIFIED"] = "UNSPECIFIED";
  DiscountReason2["EXCHANGED_ITEMS"] = "EXCHANGED_ITEMS";
  DiscountReason2["BILLING_ADJUSTMENT"] = "BILLING_ADJUSTMENT";
  return DiscountReason2;
})(DiscountReason || {});
var ChannelType = /* @__PURE__ */ ((ChannelType2) => {
  ChannelType2["UNSPECIFIED"] = "UNSPECIFIED";
  ChannelType2["WEB"] = "WEB";
  ChannelType2["POS"] = "POS";
  ChannelType2["EBAY"] = "EBAY";
  ChannelType2["AMAZON"] = "AMAZON";
  ChannelType2["OTHER_PLATFORM"] = "OTHER_PLATFORM";
  ChannelType2["WIX_APP_STORE"] = "WIX_APP_STORE";
  ChannelType2["WIX_INVOICES"] = "WIX_INVOICES";
  ChannelType2["BACKOFFICE_MERCHANT"] = "BACKOFFICE_MERCHANT";
  ChannelType2["WISH"] = "WISH";
  ChannelType2["CLASS_PASS"] = "CLASS_PASS";
  ChannelType2["GLOBAL_E"] = "GLOBAL_E";
  ChannelType2["FACEBOOK"] = "FACEBOOK";
  ChannelType2["ETSY"] = "ETSY";
  ChannelType2["TIKTOK"] = "TIKTOK";
  ChannelType2["FAIRE_COM"] = "FAIRE_COM";
  ChannelType2["PAYPAL_AGENTIC_CHECKOUT"] = "PAYPAL_AGENTIC_CHECKOUT";
  return ChannelType2;
})(ChannelType || {});
var AdditionalFeeSource = /* @__PURE__ */ ((AdditionalFeeSource2) => {
  AdditionalFeeSource2["UNKNOWN_ADDITIONAL_FEE_SOURCE"] = "UNKNOWN_ADDITIONAL_FEE_SOURCE";
  AdditionalFeeSource2["SERVICE_PLUGIN"] = "SERVICE_PLUGIN";
  AdditionalFeeSource2["ITEM"] = "ITEM";
  AdditionalFeeSource2["MANUAL"] = "MANUAL";
  AdditionalFeeSource2["SHIPPING"] = "SHIPPING";
  AdditionalFeeSource2["PLATFORM"] = "PLATFORM";
  return AdditionalFeeSource2;
})(AdditionalFeeSource || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function getSubscriptionContract2(subscriptionContractId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    subscriptionContractId
  });
  const reqOpts = getSubscriptionContract(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [{ path: "subscriptionContract.lineItems.image" }]
        },
        {
          transformFn: import_address.transformRESTAddressToSDKAddress,
          paths: [
            { path: "subscriptionContract.billingInfo.address" },
            {
              path: "subscriptionContract.shippingInfo.logistics.shippingDestination.address"
            },
            {
              path: "subscriptionContract.shippingInfo.logistics.pickupDetails.address"
            },
            { path: "subscriptionContract.recipientInfo.address" }
          ]
        }
      ])
    )?.subscriptionContract;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { subscriptionContractId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["subscriptionContractId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function querySubscriptionContracts2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = querySubscriptionContracts(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image.transformRESTImageToSDKImage,
            paths: [{ path: "subscriptionContracts.lineItems.image" }]
          },
          {
            transformFn: import_address.transformRESTAddressToSDKAddress,
            paths: [
              { path: "subscriptionContracts.billingInfo.address" },
              {
                path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address"
              },
              {
                path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address"
              },
              { path: "subscriptionContracts.recipientInfo.address" }
            ]
          }
        ])
      );
      return {
        items: transformedData?.subscriptionContracts,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function typedQuerySubscriptionContracts(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ query });
  const reqOpts = querySubscriptionContracts(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image.transformRESTImageToSDKImage,
          paths: [{ path: "subscriptionContracts.lineItems.image" }]
        },
        {
          transformFn: import_address.transformRESTAddressToSDKAddress,
          paths: [
            { path: "subscriptionContracts.billingInfo.address" },
            {
              path: "subscriptionContracts.shippingInfo.logistics.shippingDestination.address"
            },
            {
              path: "subscriptionContracts.shippingInfo.logistics.pickupDetails.address"
            },
            { path: "subscriptionContracts.recipientInfo.address" }
          ]
        }
      ])
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.public.ts
function getSubscriptionContract3(httpClient) {
  return (subscriptionContractId) => getSubscriptionContract2(
    subscriptionContractId,
    // @ts-ignore
    { httpClient }
  );
}
function querySubscriptionContracts3(httpClient) {
  return () => querySubscriptionContracts2(
    // @ts-ignore
    { httpClient }
  );
}
function typedQuerySubscriptionContracts2(httpClient) {
  return (query) => typedQuerySubscriptionContracts(
    query,
    // @ts-ignore
    { httpClient }
  );
}
var onSubscriptionContractCreated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.subscription_contracts.v1.subscription_contract_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.subscriptionSettings.startDate" },
          { path: "entity.lineItems.digitalFile.expirationDate" },
          {
            path: "entity.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "entity.shippingInfo.logistics.deliverByDate" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "entity.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [{ path: "entity.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address2.transformRESTAddressToSDKAddress,
        paths: [
          { path: "entity.billingInfo.address" },
          {
            path: "entity.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "entity.shippingInfo.logistics.pickupDetails.address" },
          { path: "entity.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onSubscriptionContractDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.subscription_contracts.v1.subscription_contract_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.subscriptionSettings.startDate" },
          { path: "undefined.lineItems.digitalFile.expirationDate" },
          {
            path: "undefined.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "undefined.shippingInfo.logistics.deliverByDate" },
          { path: "undefined.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "undefined.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [{ path: "undefined.lineItems.image" }]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [{ path: "undefined.lineItems.physicalProperties.weight" }]
      },
      {
        transformFn: import_address2.transformRESTAddressToSDKAddress,
        paths: [
          { path: "undefined.billingInfo.address" },
          {
            path: "undefined.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "undefined.shippingInfo.logistics.pickupDetails.address" },
          { path: "undefined.recipientInfo.address" }
        ]
      }
    ])
  )
)();
var onSubscriptionContractUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.subscription_contracts.v1.subscription_contract_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.subscriptionSettings.startDate" },
          { path: "entity.lineItems.digitalFile.expirationDate" },
          {
            path: "entity.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "entity.shippingInfo.logistics.deliverByDate" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.from" },
          { path: "entity.shippingInfo.logistics.deliveryTimeSlot.to" },
          { path: "metadata.eventTime" },
          { path: "modifiedFields.createdDate" },
          { path: "modifiedFields.updatedDate" },
          { path: "modifiedFields.subscriptionSettings.startDate" },
          { path: "modifiedFields.lineItems.digitalFile.expirationDate" },
          {
            path: "modifiedFields.lineItems.subscriptionInfo.subscriptionSettings.startDate"
          },
          { path: "modifiedFields.shippingInfo.logistics.deliverByDate" },
          {
            path: "modifiedFields.shippingInfo.logistics.deliveryTimeSlot.from"
          },
          {
            path: "modifiedFields.shippingInfo.logistics.deliveryTimeSlot.to"
          }
        ]
      },
      {
        transformFn: import_image2.transformRESTImageToSDKImage,
        paths: [
          { path: "entity.lineItems.image" },
          { path: "modifiedFields.lineItems.image" }
        ]
      },
      {
        transformFn: import_float2.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.lineItems.physicalProperties.weight" },
          { path: "modifiedFields.lineItems.physicalProperties.weight" }
        ]
      },
      {
        transformFn: import_address2.transformRESTAddressToSDKAddress,
        paths: [
          { path: "entity.billingInfo.address" },
          {
            path: "entity.shippingInfo.logistics.shippingDestination.address"
          },
          { path: "entity.shippingInfo.logistics.pickupDetails.address" },
          { path: "entity.recipientInfo.address" },
          { path: "modifiedFields.billingInfo.address" },
          {
            path: "modifiedFields.shippingInfo.logistics.shippingDestination.address"
          },
          {
            path: "modifiedFields.shippingInfo.logistics.pickupDetails.address"
          },
          { path: "modifiedFields.recipientInfo.address" }
        ]
      }
    ])
  )
)();

// src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var import_query_method_router = require("@wix/sdk-runtime/query-method-router");
function customQuerySubscriptionContracts(httpClient) {
  const router = (0, import_query_method_router.createQueryOverloadRouter)({
    builderQueryFunction: () => querySubscriptionContracts3(httpClient)(),
    typedQueryFunction: (query) => typedQuerySubscriptionContracts2(httpClient)(query),
    hasOptionsParameter: false
  });
  function overloadedQuery(query) {
    return router(...arguments);
  }
  return overloadedQuery;
}
var getSubscriptionContract4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getSubscriptionContract3);
var querySubscriptionContracts4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(customQuerySubscriptionContracts);
var onSubscriptionContractCreated2 = (0, import_event_definition_modules.createEventModule)(
  onSubscriptionContractCreated
);
var onSubscriptionContractDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onSubscriptionContractDeleted
);
var onSubscriptionContractUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onSubscriptionContractUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AdditionalFeeSource,
  AdjustmentType,
  ChannelType,
  ChargeType,
  DescriptionLineType,
  DiscountReason,
  DiscountType,
  ItemTypePreset,
  JurisdictionType,
  PaymentOptionType,
  PickupMethod,
  SortOrder,
  SubscriptionFrequency,
  TaxableAddressType,
  V2SubscriptionFrequency,
  VatType,
  WebhookIdentityType,
  WeightUnit,
  getSubscriptionContract,
  onSubscriptionContractCreated,
  onSubscriptionContractDeleted,
  onSubscriptionContractUpdated,
  querySubscriptionContracts
});
