// @wix/auto_sdk_stores_info-sections-v-3/build/internal/es/src/stores-catalog-v3-info-section-info-sections-v-3.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { bulkCreateInfoSections as universalBulkCreateInfoSections, bulkDeleteInfoSections as universalBulkDeleteInfoSections, bulkGetOrCreateInfoSections as universalBulkGetOrCreateInfoSections, bulkUpdateInfoSections as universalBulkUpdateInfoSections, createInfoSection as universalCreateInfoSection, deleteInfoSection as universalDeleteInfoSection, getInfoSection as universalGetInfoSection, getOrCreateInfoSection as universalGetOrCreateInfoSection, queryInfoSections as universalQueryInfoSections, updateInfoSection as universalUpdateInfoSection, } from './stores-catalog-v3-info-section-info-sections-v-3.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/stores' };
export function createInfoSection(httpClient) {
    return (infoSection) => universalCreateInfoSection(infoSection, 
    // @ts-ignore
    { httpClient });
}
export function getInfoSection(httpClient) {
    return (infoSectionId, options) => universalGetInfoSection(infoSectionId, options, 
    // @ts-ignore
    { httpClient });
}
export function getOrCreateInfoSection(httpClient) {
    return (options) => universalGetOrCreateInfoSection(options, 
    // @ts-ignore
    { httpClient });
}
export function bulkGetOrCreateInfoSections(httpClient) {
    return (options) => universalBulkGetOrCreateInfoSections(options, 
    // @ts-ignore
    { httpClient });
}
export function updateInfoSection(httpClient) {
    return (_id, infoSection, options) => universalUpdateInfoSection(_id, infoSection, options, 
    // @ts-ignore
    { httpClient });
}
export function deleteInfoSection(httpClient) {
    return (infoSectionId) => universalDeleteInfoSection(infoSectionId, 
    // @ts-ignore
    { httpClient });
}
export function queryInfoSections(httpClient) {
    return (options) => universalQueryInfoSections(options, 
    // @ts-ignore
    { httpClient });
}
export function bulkCreateInfoSections(httpClient) {
    return (infoSections, options) => universalBulkCreateInfoSections(infoSections, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdateInfoSections(httpClient) {
    return (infoSections, options) => universalBulkUpdateInfoSections(infoSections, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkDeleteInfoSections(httpClient) {
    return (infoSectionIds) => universalBulkDeleteInfoSections(infoSectionIds, 
    // @ts-ignore
    { httpClient });
}
export const onInfoSectionCreated = EventDefinition('wix.stores.catalog.v3.info_section_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
        ],
    },
])))();
export const onInfoSectionDeleted = EventDefinition('wix.stores.catalog.v3.info_section_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'undefined.description.metadata.createdTimestamp' },
            { path: 'undefined.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'undefined.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            {
                path: 'undefined.description.nodes.galleryData.options.item.ratio',
            },
            { path: 'undefined.description.nodes.imageData.image.duration' },
            { path: 'undefined.description.nodes.mapData.mapSettings.lat' },
            { path: 'undefined.description.nodes.mapData.mapSettings.lng' },
            {
                path: 'undefined.description.nodes.pollData.poll.image.duration',
            },
            {
                path: 'undefined.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'undefined.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'undefined.description.nodes.appEmbedData.image.duration' },
            { path: 'undefined.description.nodes.videoData.video.duration' },
            {
                path: 'undefined.description.nodes.videoData.thumbnail.duration',
            },
            { path: 'undefined.description.nodes.audioData.audio.duration' },
            {
                path: 'undefined.description.nodes.audioData.coverImage.duration',
            },
        ],
    },
])))();
export const onInfoSectionUpdated = EventDefinition('wix.stores.catalog.v3.info_section_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.description.metadata.createdTimestamp' },
            { path: 'entity.description.metadata.updatedTimestamp' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.description.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.description.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.description.nodes.galleryData.options.item.ratio' },
            { path: 'entity.description.nodes.imageData.image.duration' },
            { path: 'entity.description.nodes.mapData.mapSettings.lat' },
            { path: 'entity.description.nodes.mapData.mapSettings.lng' },
            { path: 'entity.description.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.description.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.description.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.description.nodes.appEmbedData.image.duration' },
            { path: 'entity.description.nodes.videoData.video.duration' },
            { path: 'entity.description.nodes.videoData.thumbnail.duration' },
            { path: 'entity.description.nodes.audioData.audio.duration' },
            { path: 'entity.description.nodes.audioData.coverImage.duration' },
        ],
    },
])))();
export { Alignment, AppType, BackgroundType, Crop, DecorationType, Direction, FontType, GIFType, InitialExpandedItems, LayoutType, LineStyle, MapType, NodeType, NullValue, Orientation, PluginContainerDataAlignment, PollLayoutDirection, PollLayoutType, RequestedFields, SortOrder, Source, Target, TextAlignment, ThumbnailsAlignment, Type, VerticalAlignment, ViewMode, ViewRole, VoteRole, WebhookIdentityType, Width, WidthType, } from './stores-catalog-v3-info-section-info-sections-v-3.universal.js';
