// @wix/stores/build/cjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.inventory = exports.subscriptionOptions = exports.abandonedCarts = exports.storesLocationsV3 = exports.ribbonsV3 = exports.catalogVersioning = exports.productsV3 = exports.inventoryItemsV3 = exports.infoSectionsV3 = exports.customizationsV3 = exports.brandsV3 = exports.products = exports.collections = exports.wishlist = void 0;
const wishlist = __importStar(require("@wix/auto_sdk_stores_wishlist"));
exports.wishlist = wishlist;
const collections = __importStar(require("@wix/auto_sdk_stores_collections"));
exports.collections = collections;
const products = __importStar(require("@wix/auto_sdk_stores_products"));
exports.products = products;
const brandsV3 = __importStar(require("@wix/auto_sdk_stores_brands-v-3"));
exports.brandsV3 = brandsV3;
const customizationsV3 = __importStar(require("@wix/auto_sdk_stores_customizations-v-3"));
exports.customizationsV3 = customizationsV3;
const infoSectionsV3 = __importStar(require("@wix/auto_sdk_stores_info-sections-v-3"));
exports.infoSectionsV3 = infoSectionsV3;
const inventoryItemsV3 = __importStar(require("@wix/auto_sdk_stores_inventory-items-v-3"));
exports.inventoryItemsV3 = inventoryItemsV3;
const productsV3 = __importStar(require("@wix/auto_sdk_stores_products-v-3"));
exports.productsV3 = productsV3;
const catalogVersioning = __importStar(require("@wix/auto_sdk_stores_catalog-versioning"));
exports.catalogVersioning = catalogVersioning;
const ribbonsV3 = __importStar(require("@wix/auto_sdk_stores_ribbons-v-3"));
exports.ribbonsV3 = ribbonsV3;
const storesLocationsV3 = __importStar(require("@wix/auto_sdk_stores_stores-locations-v-3"));
exports.storesLocationsV3 = storesLocationsV3;
const abandonedCarts = __importStar(require("@wix/auto_sdk_stores_abandoned-carts"));
exports.abandonedCarts = abandonedCarts;
const subscriptionOptions = __importStar(require("@wix/auto_sdk_stores_subscription-options"));
exports.subscriptionOptions = subscriptionOptions;
const inventory = __importStar(require("@wix/auto_sdk_stores_inventory"));
exports.inventory = inventory;
