// @wix/auto_sdk_ecom_discount-rules/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AppliedSubjectType: () => AppliedSubjectType,
  DayOfWeek: () => DayOfWeek,
  DiscountType: () => DiscountType,
  EligibilityType: () => EligibilityType,
  ScopeType: () => ScopeType,
  SortOrder: () => SortOrder,
  Status: () => Status,
  TriggerType: () => TriggerType,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  createDiscountRule: () => createDiscountRule4,
  deleteDiscountRule: () => deleteDiscountRule4,
  getDiscountRule: () => getDiscountRule4,
  onDiscountRuleCreated: () => onDiscountRuleCreated2,
  onDiscountRuleDeleted: () => onDiscountRuleDeleted2,
  onDiscountRuleUpdated: () => onDiscountRuleUpdated2,
  queryDiscountRules: () => queryDiscountRules4,
  updateDiscountRule: () => updateDiscountRule4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-discounts-v1-discount-rule-discount-rules.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_float3 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths3 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/ecom-discounts-v1-discount-rule-discount-rules.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-discounts-v1-discount-rule-discount-rules.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomDiscountsDiscountRuleServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/discount-rules-service",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/discount-rules",
        destPath: "/v1/discount-rules"
      },
      {
        srcPath: "/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/discounts/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ],
    _: [
      {
        srcPath: "/ecom/discounts/v1/discount-rules",
        destPath: "/v1/discount-rules"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_discount-rules";
function createDiscountRule(payload) {
  function __createDiscountRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "discountRule.createdDate" },
          { path: "discountRule.updatedDate" },
          { path: "discountRule.activeTimeInfo.start" },
          { path: "discountRule.activeTimeInfo.end" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "discountRule.discounts.values.percentage" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "POST",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.CreateDiscountRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __createDiscountRule;
}
function getDiscountRule(payload) {
  function __getDiscountRule({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "GET",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.GetDiscountRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRuleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __getDiscountRule;
}
function updateDiscountRule(payload) {
  function __updateDiscountRule({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "mask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "discountRule.createdDate" },
          { path: "discountRule.updatedDate" },
          { path: "discountRule.activeTimeInfo.start" },
          { path: "discountRule.activeTimeInfo.end" }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [{ path: "discountRule.discounts.values.percentage" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "PATCH",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.UpdateDiscountRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRule.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRule.createdDate" },
            { path: "discountRule.updatedDate" },
            { path: "discountRule.activeTimeInfo.start" },
            { path: "discountRule.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRule.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __updateDiscountRule;
}
function deleteDiscountRule(payload) {
  function __deleteDiscountRule({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "DELETE",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.DeleteDiscountRule",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/{discountRuleId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDiscountRule;
}
function queryDiscountRules(payload) {
  function __queryDiscountRules({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.discounts.v1.discount_rule",
      method: "POST",
      methodFqn: "com.wix.ecom.discounts.DiscountRuleService.QueryDiscountRules",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
        protoPath: "/v1/discount-rules/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "discountRules.createdDate" },
            { path: "discountRules.updatedDate" },
            { path: "discountRules.activeTimeInfo.start" },
            { path: "discountRules.activeTimeInfo.end" }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [{ path: "discountRules.discounts.values.percentage" }]
        }
      ])
    };
    return metadata;
  }
  return __queryDiscountRules;
}

// src/ecom-discounts-v1-discount-rule-discount-rules.universal.ts
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var ScopeType = /* @__PURE__ */ ((ScopeType2) => {
  ScopeType2["UNDEFINED_SCOPE"] = "UNDEFINED_SCOPE";
  ScopeType2["CATALOG_ITEM"] = "CATALOG_ITEM";
  ScopeType2["CUSTOM_FILTER"] = "CUSTOM_FILTER";
  return ScopeType2;
})(ScopeType || {});
var EligibilityType = /* @__PURE__ */ ((EligibilityType2) => {
  EligibilityType2["UNKNOWN_ELIGIBILITY_TYPE"] = "UNKNOWN_ELIGIBILITY_TYPE";
  EligibilityType2["INDIVIDUAL_MEMBERS"] = "INDIVIDUAL_MEMBERS";
  return EligibilityType2;
})(EligibilityType || {});
var TriggerType = /* @__PURE__ */ ((TriggerType2) => {
  TriggerType2["UNDEFINED"] = "UNDEFINED";
  TriggerType2["AND"] = "AND";
  TriggerType2["SUBTOTAL_RANGE"] = "SUBTOTAL_RANGE";
  TriggerType2["ITEM_QUANTITY_RANGE"] = "ITEM_QUANTITY_RANGE";
  TriggerType2["CUSTOM"] = "CUSTOM";
  TriggerType2["OR"] = "OR";
  return TriggerType2;
})(TriggerType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["UNKOWN_DAY_OF_WEEK"] = "UNKOWN_DAY_OF_WEEK";
  DayOfWeek2["MON"] = "MON";
  DayOfWeek2["TUE"] = "TUE";
  DayOfWeek2["WED"] = "WED";
  DayOfWeek2["THU"] = "THU";
  DayOfWeek2["FRI"] = "FRI";
  DayOfWeek2["SAT"] = "SAT";
  DayOfWeek2["SUN"] = "SUN";
  return DayOfWeek2;
})(DayOfWeek || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNDEFINED"] = "UNDEFINED";
  Type2["SPECIFIC_ITEMS"] = "SPECIFIC_ITEMS";
  return Type2;
})(Type || {});
var DiscountType = /* @__PURE__ */ ((DiscountType2) => {
  DiscountType2["UNDEFINED"] = "UNDEFINED";
  DiscountType2["PERCENTAGE"] = "PERCENTAGE";
  DiscountType2["FIXED_AMOUNT"] = "FIXED_AMOUNT";
  DiscountType2["FIXED_PRICE"] = "FIXED_PRICE";
  DiscountType2["QUANTITY_BASED"] = "QUANTITY_BASED";
  return DiscountType2;
})(DiscountType || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNDEFINED"] = "UNDEFINED";
  Status2["LIVE"] = "LIVE";
  Status2["EXPIRED"] = "EXPIRED";
  Status2["PENDING"] = "PENDING";
  return Status2;
})(Status || {});
var AppliedSubjectType = /* @__PURE__ */ ((AppliedSubjectType2) => {
  AppliedSubjectType2["UNDEFINED"] = "UNDEFINED";
  AppliedSubjectType2["ALL_ITEMS"] = "ALL_ITEMS";
  AppliedSubjectType2["LOWEST_PRICED_ITEM"] = "LOWEST_PRICED_ITEM";
  return AppliedSubjectType2;
})(AppliedSubjectType || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createDiscountRule2(discountRule) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    discountRule
  });
  const reqOpts = createDiscountRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discountRule;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { discountRule: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["discountRule"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDiscountRule2(discountRuleId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    discountRuleId
  });
  const reqOpts = getDiscountRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discountRule;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { discountRuleId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["discountRuleId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateDiscountRule2(_id, discountRule) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    discountRule: { ...discountRule, id: _id }
  });
  const reqOpts = updateDiscountRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.discountRule;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { discountRule: "$[1]" },
        explicitPathsToArguments: { "discountRule.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "discountRule"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteDiscountRule2(discountRuleId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    discountRuleId
  });
  const reqOpts = deleteDiscountRule(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { discountRuleId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["discountRuleId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDiscountRules2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryDiscountRules(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [])
      );
      return {
        items: transformedData?.discountRules,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function typedQueryDiscountRules(query) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ query });
  const reqOpts = queryDiscountRules(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["query"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-discounts-v1-discount-rule-discount-rules.public.ts
function createDiscountRule3(httpClient) {
  return (discountRule) => createDiscountRule2(
    discountRule,
    // @ts-ignore
    { httpClient }
  );
}
function getDiscountRule3(httpClient) {
  return (discountRuleId) => getDiscountRule2(
    discountRuleId,
    // @ts-ignore
    { httpClient }
  );
}
function updateDiscountRule3(httpClient) {
  return (_id, discountRule) => updateDiscountRule2(
    _id,
    discountRule,
    // @ts-ignore
    { httpClient }
  );
}
function deleteDiscountRule3(httpClient) {
  return (discountRuleId) => deleteDiscountRule2(
    discountRuleId,
    // @ts-ignore
    { httpClient }
  );
}
function queryDiscountRules3(httpClient) {
  return () => queryDiscountRules2(
    // @ts-ignore
    { httpClient }
  );
}
function typedQueryDiscountRules2(httpClient) {
  return (query) => typedQueryDiscountRules(
    query,
    // @ts-ignore
    { httpClient }
  );
}
var onDiscountRuleCreated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.discounts.v1.discount_rule_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.activeTimeInfo.start" },
          { path: "entity.activeTimeInfo.end" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "entity.discounts.values.percentage" }]
      }
    ])
  )
)();
var onDiscountRuleDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.discounts.v1.discount_rule_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "undefined.updatedDate" },
          { path: "undefined.activeTimeInfo.start" },
          { path: "undefined.activeTimeInfo.end" },
          { path: "metadata.eventTime" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [{ path: "undefined.discounts.values.percentage" }]
      }
    ])
  )
)();
var onDiscountRuleUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.ecom.discounts.v1.discount_rule_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths3.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "entity.updatedDate" },
          { path: "entity.activeTimeInfo.start" },
          { path: "entity.activeTimeInfo.end" },
          { path: "metadata.eventTime" },
          { path: "modifiedFields.createdDate" },
          { path: "modifiedFields.updatedDate" },
          { path: "modifiedFields.activeTimeInfo.start" },
          { path: "modifiedFields.activeTimeInfo.end" }
        ]
      },
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          { path: "entity.discounts.values.percentage" },
          { path: "modifiedFields.discounts.values.percentage" }
        ]
      }
    ])
  )
)();

// src/ecom-discounts-v1-discount-rule-discount-rules.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var import_query_method_router = require("@wix/sdk-runtime/query-method-router");
function customQueryDiscountRules(httpClient) {
  const router = (0, import_query_method_router.createQueryOverloadRouter)({
    builderQueryFunction: () => queryDiscountRules3(httpClient)(),
    typedQueryFunction: (query) => typedQueryDiscountRules2(httpClient)(query),
    hasOptionsParameter: false
  });
  function overloadedQuery(query) {
    return router(...arguments);
  }
  return overloadedQuery;
}
var createDiscountRule4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createDiscountRule3);
var getDiscountRule4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getDiscountRule3);
var updateDiscountRule4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateDiscountRule3);
var deleteDiscountRule4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteDiscountRule3);
var queryDiscountRules4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(customQueryDiscountRules);
var onDiscountRuleCreated2 = (0, import_event_definition_modules.createEventModule)(
  onDiscountRuleCreated
);
var onDiscountRuleDeleted2 = (0, import_event_definition_modules.createEventModule)(
  onDiscountRuleDeleted
);
var onDiscountRuleUpdated2 = (0, import_event_definition_modules.createEventModule)(
  onDiscountRuleUpdated
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppliedSubjectType,
  DayOfWeek,
  DiscountType,
  EligibilityType,
  ScopeType,
  SortOrder,
  Status,
  TriggerType,
  Type,
  WebhookIdentityType,
  createDiscountRule,
  deleteDiscountRule,
  getDiscountRule,
  onDiscountRuleCreated,
  onDiscountRuleDeleted,
  onDiscountRuleUpdated,
  queryDiscountRules,
  updateDiscountRule
});
