// @wix/auto_sdk_ecom_recommendations/build/internal/cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AlgorithmType: () => AlgorithmType,
  getRecommendation: () => getRecommendation4,
  listAvailableAlgorithms: () => listAvailableAlgorithms4
});
module.exports = __toCommonJS(index_exports);

// src/ecom-recommendations-v1-recommendation-recommendations.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/ecom-recommendations-v1-recommendation-recommendations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomRecommendationsV1RecommendationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      },
      {
        srcPath: "/ecom/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    _: [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_recommendations";
function listAvailableAlgorithms(payload) {
  function __listAvailableAlgorithms({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "GET",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.ListAvailableAlgorithms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/algorithms",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listAvailableAlgorithms;
}
function getRecommendation(payload) {
  function __getRecommendation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "POST",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.GetRecommendation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/get",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getRecommendation;
}

// src/ecom-recommendations-v1-recommendation-recommendations.universal.ts
var AlgorithmType = /* @__PURE__ */ ((AlgorithmType2) => {
  AlgorithmType2["UNSPECIFIED"] = "UNSPECIFIED";
  AlgorithmType2["RELATED_ITEMS"] = "RELATED_ITEMS";
  AlgorithmType2["GLOBAL"] = "GLOBAL";
  return AlgorithmType2;
})(AlgorithmType || {});
async function listAvailableAlgorithms2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = listAvailableAlgorithms(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRecommendation2(algorithms, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    algorithms,
    items: options?.items,
    minimumRecommendedItems: options?.minimumRecommendedItems
  });
  const reqOpts = getRecommendation(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          algorithms: "$[0]",
          items: "$[1].items",
          minimumRecommendedItems: "$[1].minimumRecommendedItems"
        },
        singleArgumentUnchanged: false
      },
      ["algorithms", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/ecom-recommendations-v1-recommendation-recommendations.public.ts
function listAvailableAlgorithms3(httpClient) {
  return () => listAvailableAlgorithms2(
    // @ts-ignore
    { httpClient }
  );
}
function getRecommendation3(httpClient) {
  return (algorithms, options) => getRecommendation2(
    algorithms,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/ecom-recommendations-v1-recommendation-recommendations.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var listAvailableAlgorithms4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listAvailableAlgorithms3);
var getRecommendation4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRecommendation3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlgorithmType,
  getRecommendation,
  listAvailableAlgorithms
});
