// @wix/auto_sdk_ecom_recommendations/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  AlgorithmTypeOriginal: () => AlgorithmType,
  getRecommendation: () => getRecommendation2,
  listAvailableAlgorithms: () => listAvailableAlgorithms2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-recommendations-v1-recommendation-recommendations.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomRecommendationsV1RecommendationsServiceUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      },
      {
        srcPath: "/ecom/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    _: [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/v1/recommendations",
        destPath: "/v1/recommendations"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_recommendations";
function listAvailableAlgorithms(payload) {
  function __listAvailableAlgorithms({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "GET",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.ListAvailableAlgorithms",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/algorithms",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listAvailableAlgorithms;
}
function getRecommendation(payload) {
  function __getRecommendation({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.recommendations.v1.recommendation",
      method: "POST",
      methodFqn: "com.wix.ecom.recommendations.v1.RecommendationsService.GetRecommendation",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomRecommendationsV1RecommendationsServiceUrl({
        protoPath: "/v1/recommendations/get",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __getRecommendation;
}

// src/ecom-recommendations-v1-recommendation-recommendations.types.ts
var AlgorithmType = /* @__PURE__ */ ((AlgorithmType2) => {
  AlgorithmType2["UNSPECIFIED"] = "UNSPECIFIED";
  AlgorithmType2["RELATED_ITEMS"] = "RELATED_ITEMS";
  AlgorithmType2["GLOBAL"] = "GLOBAL";
  return AlgorithmType2;
})(AlgorithmType || {});

// src/ecom-recommendations-v1-recommendation-recommendations.meta.ts
function listAvailableAlgorithms2() {
  const payload = {};
  const getRequestOptions = listAvailableAlgorithms(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/recommendations/algorithms",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRecommendation2() {
  const payload = {};
  const getRequestOptions = getRecommendation(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/recommendations/get",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlgorithmTypeOriginal,
  getRecommendation,
  listAvailableAlgorithms
});
