// @wix/auto_sdk_stores_inventory/build/internal/es/src/stores-v2-inventory-inventory.meta.js
import * as ambassadorWixStoresV2Inventory from './stores-v2-inventory-inventory.http.js';
export function getInventoryVariants() {
    const payload = { inventoryId: ':inventoryId' };
    const getRequestOptions = ambassadorWixStoresV2Inventory.getInventoryVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/{inventoryId}/getVariants',
        pathParams: { inventoryId: 'inventoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.queryInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateInventoryVariants() {
    const payload = {
        inventoryItem: { productId: ':inventoryItemProductId' },
    };
    const getRequestOptions = ambassadorWixStoresV2Inventory.updateInventoryVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/inventoryItems/product/{inventoryItem.productId}',
        pathParams: { inventoryItemProductId: 'inventoryItemProductId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function decrementInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.decrementInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/decrement',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function incrementInventory() {
    const payload = {};
    const getRequestOptions = ambassadorWixStoresV2Inventory.incrementInventory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/inventoryItems/increment',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
