// @wix/auto_sdk_stores_inventory/build/internal/es/src/stores-v2-inventory-inventory.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTFieldMaskToSDKFieldMask } from '@wix/sdk-runtime/transformations/field-mask';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { decrementInventory as universalDecrementInventory, getInventoryVariants as universalGetInventoryVariants, incrementInventory as universalIncrementInventory, queryInventory as universalQueryInventory, updateInventoryVariants as universalUpdateInventoryVariants, } from './stores-v2-inventory-inventory.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/stores' };
export function getInventoryVariants(httpClient) {
    return (inventoryId, options) => universalGetInventoryVariants(inventoryId, options, 
    // @ts-ignore
    { httpClient });
}
export function queryInventory(httpClient) {
    return (options) => universalQueryInventory(options, 
    // @ts-ignore
    { httpClient });
}
export function updateInventoryVariants(httpClient) {
    return (productId, inventoryItem) => universalUpdateInventoryVariants(productId, inventoryItem, 
    // @ts-ignore
    { httpClient });
}
export function decrementInventory(httpClient) {
    return (decrementData) => universalDecrementInventory(decrementData, 
    // @ts-ignore
    { httpClient });
}
export function incrementInventory(httpClient) {
    return (incrementData) => universalIncrementInventory(incrementData, 
    // @ts-ignore
    { httpClient });
}
export const onInventoryItemChanged = EventDefinition('com.wix.ecommerce.inventory.api.v1.InventoryItemChanged', false, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTFieldMaskToSDKFieldMask,
        paths: [{ path: 'data.fieldMask' }],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.updatedDate' }],
    },
])))();
export const onInventoryVariantsChanged = EventDefinition('com.wix.ecommerce.inventory.api.v1.InventoryVariantsChanged', false, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.updatedDate' }],
    },
])))();
export { ReasonType, WebhookIdentityType, } from './stores-v2-inventory-inventory.universal.js';
