// @wix/auto_sdk_ecom_order-invoices/build/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  WebhookIdentityTypeOriginal: () => WebhookIdentityType,
  addInvoiceToOrder: () => addInvoiceToOrder2,
  bulkGenerateInvoices: () => bulkGenerateInvoices2,
  generateInvoice: () => generateInvoice2,
  listInvoicesForMultipleOrders: () => listInvoicesForMultipleOrders2
});
module.exports = __toCommonJS(meta_exports);

// src/ecom-orders-v1-invoice-order-invoices.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomOrdersInvoicesV1InvoicesUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/ecom/v1/bulk/invoices",
        destPath: "/v1/bulk/invoices"
      },
      {
        srcPath: "/ecom/v1/invoices",
        destPath: "/v1/invoices"
      },
      {
        srcPath: "/ecom/v1/ep-invoices",
        destPath: "/v1/invoices"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/ecom/v1/ep-invoices",
        destPath: "/v1/invoices"
      },
      {
        srcPath: "/_api/ecom-payments/v1/invoices",
        destPath: "/v1/invoices"
      }
    ],
    _: [
      {
        srcPath: "/v1/invoices",
        destPath: "/v1/invoices"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/ecom/v1/invoices",
        destPath: "/v1/invoices"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_ecom_order-invoices";
function listInvoicesForMultipleOrders(payload) {
  function __listInvoicesForMultipleOrders({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.ListInvoicesForMultipleOrders",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/list-by-ids",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __listInvoicesForMultipleOrders;
}
function generateInvoice(payload) {
  function __generateInvoice({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.GenerateInvoice",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/orders/{orderId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __generateInvoice;
}
function bulkGenerateInvoices(payload) {
  function __bulkGenerateInvoices({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.BulkGenerateInvoices",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/bulk/invoices/generate",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkGenerateInvoices;
}
function addInvoiceToOrder(payload) {
  function __addInvoiceToOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.ecom.orders.v1.invoice",
      method: "POST",
      methodFqn: "com.wix.ecom.orders.invoices.v1.Invoices.AddInvoiceToOrder",
      packageName: PACKAGE_NAME,
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
        protoPath: "/v1/invoices/orders/{orderId}/add-invoice",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __addInvoiceToOrder;
}

// src/ecom-orders-v1-invoice-order-invoices.types.ts
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});

// src/ecom-orders-v1-invoice-order-invoices.meta.ts
function listInvoicesForMultipleOrders2() {
  const payload = {};
  const getRequestOptions = listInvoicesForMultipleOrders(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/list-by-ids",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function generateInvoice2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = generateInvoice(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/orders/{orderId}",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkGenerateInvoices2() {
  const payload = {};
  const getRequestOptions = bulkGenerateInvoices(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/bulk/invoices/generate",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function addInvoiceToOrder2() {
  const payload = { orderId: ":orderId" };
  const getRequestOptions = addInvoiceToOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/invoices/orders/{orderId}/add-invoice",
    pathParams: { orderId: "orderId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebhookIdentityTypeOriginal,
  addInvoiceToOrder,
  bulkGenerateInvoices,
  generateInvoice,
  listInvoicesForMultipleOrders
});
